<!-- MHonArc v2.6.10 -->
<!--X-Subject: Re: Redesign of the relax data model: 3. Molecules, residues,	and spins -->
<!--X-From-R13: "Sqjneq q'Ohiretar" <rqjneq.qnhiretarNtznvy.pbz> -->
<!--X-Date: Mon, 15 Jan 2007 09:57:28 +0100 -->
<!--X-Message-Id: 7f080ed10701150056n30ccda79xb45b60ae93439828@mail.gmail.com -->
<!--X-Content-Type: text/plain -->
<!--X-Reference: f001463a0701071417w6bd7927cp8fdd052e698575ec@mail.gmail.com -->
<!--X-Head-End-->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<title>Re: Redesign of the relax data model: 3. Molecules, residues,	and spins -- January 15, 2007 - 09:57</title>
<link rel="stylesheet" type="text/css" href="/archives-color-gna.css"> 
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
</head>
<body>
<!--X-Body-Begin-->
<!--X-User-Header-->
<!--X-User-Header-End-->
<!--X-TopPNI-->
<h2><img src="https://gna.org/images/gna.theme/mail.orig.png" width="48" height="48"
alt="mail" class="pageicon" />Re: Redesign of the relax data model: 3. Molecules, residues,	and spins</h2>
<br />
<div class="topmenu">
<a href="../" class="tabs">Others Months</a> | <a href="index.html#00031" class="tabs">Index by Date</a> | <a href="threads.html#00031" class="tabs">Thread Index</a><br />
<span class="smaller">&gt;&gt;&nbsp;&nbsp;
[<a href="msg00030.html">Date Prev</a>] [<a href="msg00032.html">Date Next</a>] [<a href="msg00030.html">Thread Prev</a>] [<a href="msg00015.html">Thread Next</a>]
</div>

<!--X-TopPNI-End-->
<!--X-MsgBody-->
<!--X-Subject-Header-Begin-->
<h3><a name="header" href="#header">Header</a></h3>
<!--X-Subject-Header-End-->
<!--X-Head-of-Message-->
<ul class="headdata">
<li class="menuitem">
<em>To</em>: &quot;gary thompson&quot; &lt;garyt@xxxxxxxxxxxxxxx&gt;</li>
<li class="menuitem">
<em>Date</em>: Mon, 15 Jan 2007 19:56:14 +1100</li>
<li class="menuitem">
<em>Cc</em>: relax-devel@xxxxxxx</li>
<li class="menuitem">
<em>Message-id</em>: &lt;<a href="msg00031.html">7f080ed10701150056n30ccda79xb45b60ae93439828@mail.gmail.com</a>&gt;</li>
<li class="menuitem">
<em>References</em>: &lt;<a href="msg00014.html">f001463a0701071417w6bd7927cp8fdd052e698575ec@mail.gmail.com</a>&gt;</li>
</ul>
<!--X-Head-of-Message-End-->
<!--X-Head-Body-Sep-Begin-->
</div><!-- end headdata -->
<br />
<h3><a name="content" href="#content">Content</a></h3>
<div class="postedby">Posted by <strong>Edward d'Auvergne</strong> on January 15, 2007 - 09:57:</div>
<div class="msgdata">
<!--X-Head-Body-Sep-End-->
<!--X-Body-of-Message-->
<blockquote class="blockquote"><tt>&gt;   3.5  Molecule, sequence, and spin user function classes
</tt></blockquote><pre style="margin: 0em;"><br>[snip]</pre><br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   3.5.1  The 'molecule' user function class
&gt;
&gt;   This user function class could contain functions such as:
&gt;
&gt;   molecule.add()
&gt;   molecule.copy()    # Copy the molecule information (name and num) from
&gt;   another pipe.
&gt;   molecule.delete()
&gt;   molecule.info()   # Print the molecule info.
&gt;   molecule.sort()
&gt;
&gt;   Other functions could be created to enable associations between the
&gt;   'data.mol[index]' data structure and the Scientific Python PDB data
&gt;   structure.  This will allow the 'vectors()' user function to correctly
&gt;   extract XH bond vectors from the PDB data structure.  The 'pdb' user
&gt;   function class could also be renamed to 'structure' to enable other 3D
&gt;   molecular structure files to be transparently supported by the
&gt;   'molecule' user functions (e.g. CIF).  The mapping of the structure to
&gt;   the molecule-residue-spin framework could be quite complex, especially
&gt;   if the Scientific PDB format is not the only format handled.  Would any
&gt;   one have ideas of how to map multiple molecules in the PDB file to the
&gt;   molecule name and number in the proposed molecule-residue-spin data
&gt;   model?</pre><br>
<pre style="margin: 0em;"><br>&gt; NMR models in a PDB are easy to handle as these are treated as
&gt;   different structures by the Scientific Python PDB reader, each can be
&gt;   isolated by model number.  If a specific model is chosen, the model
&gt;   number could become 'data.mol[index].num'.  If the average of all models
&gt;   is chosen, then 'data.mol[index].num' could be None.  A number of other
&gt;   questions are:</pre><br>
<pre style="margin: 0em;">As an aside I wrote a parser for pdb files a while back that dealt
with multiple molecules with either chain ids or seg id or no ids and
I had to prioriise along the lines of segid&gt; chain-id&gt; nothing.
However, I had to add special cases for nmr structure files and
actually had the idea of an ensemble of structures as well as a single
structure
&gt;
&gt;   How would you expect to handle chain IDs and segment IDs?  Should this
&gt;   be the 'mol_name'?</pre><br>
<pre style="margin: 0em;">    I would just give each molecule a name if you don't have a segment
id then use the chain id if there is no chain id and  use a generic
name like mol_1, mol_2  etc
</pre></blockquote><pre style="margin: 0em;"><br>If there is no segment ID or chain ID, would it be safe to set the
molecule name to None?  In this case, surely there is only a single
macromolecule in the PDB file?</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   Are there other molecule identifiers in the PDB or other structure file
&gt;   formats (CIF, etc.)?
&gt;
&gt;   How do you think a multi-chain molecule should be handled when a segment
&gt;   of the polypeptide or nucleotide chain has been removed (e.g. insulin)?</pre><br>
<pre style="margin: 0em;">I am not following this one
&gt;
&gt;   If two chains 'A' and 'B' are part of the same polypeptide or RNA
&gt;   transcript and have sequentially numbered residue numbers, should both
&gt;   chains be handled as the same molecule in data.mol?  Or should
&gt;   data.mol[0].name = 'A' and data.mol[1].name = 'B'?
&gt;
I would call them a and b and start numbering  residues at each break
of the chain id numbering of all things in pdb files can be a
nightmare (they can even be negative and in the wrong order...) and
many programs just renumber residues &amp; atoms  in the order they are
read
</pre></blockquote><pre style="margin: 0em;"><br>So if the one molecule has two chains, A and B, we should treat the
structure as containing two molecules A and B.  Does this have the
potential to confuse the user?</pre><br>
<pre style="margin: 0em;">As the spin specific data structures will all be implemented as list
types (arrays), there is no need to renumber the residues of the
protein, nucleic acid, etc.  Their name and number will be stored
separately from the array indecies.</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   Do you think that the molecule-residue-spin data model is sufficient to
&gt;   handle all standard variants of the PDB (or other 3D structure formats)?
&gt;
&gt;   Would a molecule-chain-residue-spin data model be better?</pre><br>
<pre style="margin: 0em;">no definitely not! trying to identify which chains belong to which
molecules is generally a hopeless operation for most of these files
and what would it gain us?</pre><br>
<pre style="margin: 0em;">&gt;
&gt;   For identifying a molecule, are the data structures data.mol[0].name and
&gt;   data.mol[0].num sufficient?</pre><br>
<pre style="margin: 0em;">yes. does the number need to be explicit? what happens if you delete a
molecule or add one are the molecule numbers increased monotonically
on reading and never reused or should the number just be the number of
the molecule in the current ordering
</pre></blockquote><pre style="margin: 0em;"><br>I was unsure about the 'data.mol[index].num' variable.  This probably
has no use.</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   Are there other data structures which could be placed in data.mol[0]?
&gt;
&gt;   Which of these issues do you think should be handled by the user rather
&gt;   than internally by relax?  I would prefer that relax does most of the
&gt;   work.
&gt;
&gt;   What other user functions do you think would be useful to add to the
&gt;   'molecule' class?
&gt;
&gt;   And finally what other 3D structure files do you think should be
&gt;   supported?  These must have parsers available as Python modules.
&gt;</pre><br>
<pre style="margin: 0em;">I think we should support import/export via plugin  classes (which can
be loaded using the same code as is used in the unit_test) and then
provide a pdb reader as a starting place. The plugins would be based
around an interface class much in the same way that sax is structured
<a  href="http://www.saxproject.org/">http://www.saxproject.org/</a></pre><br>
<pre style="margin: 0em;">both readers and writers would then be subclasses of the same class:</pre><br>
<pre style="margin: 0em;"><br>class stucture_interface():
   def start_molecule(name,number)
        pass
   def end_molecule()
        pass
   def start_residue(type,number)
        pass
   def end_residue()
        pass
   def add_atom(number,name,x,y,z,b_factor=None)
        pass</pre><br>
<pre style="margin: 0em;">consider a simple molecule</pre><br>
<pre style="margin: 0em;">A.TYR.HA
A.TYR.CO
A.GAL.HA
A.GAL.CO</pre><br>
<pre style="margin: 0em;">B.HIS.HA
B.HIS.CO</pre><br>
<pre style="margin: 0em;"><br>a reader would get called as follows</pre><br>
<pre style="margin: 0em;">start_molecule('A',1)
start_residue('TYR',1)
add_atom(1,'HA',1.0,1.0,1.0)
add_atom(2,'CO',1.0,1.0,1.0)
end_residue()
start_residue('GAL',1)
add_atom(1,'HA',1.0,1.0,1.0)
add_atom(2,'CO',1.0,1.0,1.0)
end_residue()
end_molecule()</pre><br>
<pre style="margin: 0em;">start_molecule('B',2)
start_residue('HIS',1)
add_atom(1,'HA',1.0,1.0,1.0)
add_atom(2, 'CO',1.0,1.0,1.0)
end_residue()
end_molecule()</pre><br>
<pre style="margin: 0em;"><br>writing is done in the inverse manner</pre><br>
<pre style="margin: 0em;">This design produces a simple interface to work with and allows for
future additions such as ensembles to be caterered for with relative
ease. It is simple to learn and code and is symmetric.</pre><br>
<pre style="margin: 0em;">as the plugins would be classes which are identified at run time they
could take extra arguments during instantiation to allow for e.g
selection of chain-id in preference to segid
</pre></blockquote><pre style="margin: 0em;"><br>This is another great idea!  The ScientificPython, scipy, or other
pre-written PDB parsers could be easily supported as well as
readers/writers coded into relax.  I would suggest that this idea be
implemented separately from the proposed redesign.</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   3.6  The input and output files</pre><br>
<tt>i presume this refers to the relax run files...
</tt></blockquote><pre style="margin: 0em;"><br>This refers to the 'results' files and the value writing/displaying
user functions (for the relaxation data, etc).</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt;   Up to 6 columns could be used to identify spin-specific data (for both
&gt;   input and output).  These could correspond to the six spin identifiers
&gt;   'mol_num', 'mol_name', 'res_num', 'res_name', 'atom_num', and
&gt;   'atom_name'.  If any of these are set to None for all spins, the column
&gt;   could be dropped.  For example if no molecule info exists, these two
&gt;   columns can be dropped.  If no residues exist, these can be dropped as
&gt;   well.</pre><br>
<pre style="margin: 0em;">I would leave them in and give them null values dropping things just
increases code complexity
</pre></blockquote><pre style="margin: 0em;"><br>I don't mind complexity on our side if it mean simplicity for the
user.  For example if you are working on a protein studying solely the
backbone N relaxation, the molecule name, atom number, and atom name
need not be placed in the file containing the R1 relaxation data.  In
that case all you need is the residue number and name and the
relaxation values and errors.  If you are an organic chemist working
on a non-polymeric molecule, the residue number and name are
inconsequential and could be dropped from the output.  Parsing these
files using the names in the header line should be straight forward.</pre><br>
<br>
<blockquote class="blockquote"><pre style="margin: 0em;">&gt; For protein NH data of a single molecule, the data could appear
&gt;   as:
&gt;
&gt;   res_num res_name    atom_num    atom_name   ...
&gt;   1       GLY         1           N           ...
&gt;   2       PRO         11          N           ...
&gt;   3       LEU         28          N           ...
&gt;
&gt;   For RNA, the data could appear as:
&gt;
&gt;   res_num res_name    atom_num    atom_name   ...
&gt;   1       G           23          N1          ...
&gt;   1       G           18          C8          ...
&gt;   2       U           38          N3          ...
&gt;   2       U           52          C5          ...
&gt;   2       U           46          C6          ...
&gt;
&gt;   For a non-polymeric organic molecule, the data could appear as:
&gt;
&gt;   atom_num    atom_name   ...
&gt;   1           C1          ...
&gt;   16          C16         ...
&gt;   23          C23         ...
&gt;
&gt;   Are there any other standard ways of representing this data in a
&gt;   columnar format?  These formats may not be the best solution.
</pre></blockquote><pre style="margin: 0em;"><br>Cheers,</pre><br>
<pre style="margin: 0em;">Edward</pre><br>
<br>

<!--X-Body-of-Message-End-->
<!--X-MsgBody-End-->
<!--X-Follow-Ups-->
</div><!-- end msgdata -->
<br />
<h3><a name="related" href="#related">Related Messages</a></h3>
<div class="relateddata">
<!--X-Follow-Ups-End-->
<!--X-References-->
<ul><li><strong>References</strong>:
<ul>
<li><strong><a name="00014" href="msg00014.html">re: Redesign of the relax data model: 3. Molecules, residues,	and spins</a></strong>
<ul><li><em>From:</em> gary thompson</li></ul></li>
</ul></li></ul>
<!--X-References-End-->
<!--X-BotPNI-->
</div><!-- end relateddata -->
<!-- NoBotLinksApartFromRelatedMessages -->

<!--X-BotPNI-End-->
<!--X-User-Footer-->
<!--X-User-Footer-End-->
<div class="footer">You are on the <a href="http://gna.org">Gna!</a> mail server.</div><br />
<div class="right">Powered by <a href="http://www.mhonarc.org">MHonArc</a>, Updated Mon Jan 15 11:00:24 2007</div>  
</body>
</html>
