<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>extern.scientific_python.IO.PDB</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://nmr-relax.com">relax</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="extern-module.html">Package&nbsp;extern</a> ::
        <a href="extern.scientific_python-module.html">Package&nbsp;scientific_python</a> ::
        <a href="extern.scientific_python.IO-module.html">Package&nbsp;IO</a> ::
        Module&nbsp;PDB
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="extern.scientific_python.IO.PDB-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="extern.scientific_python.IO.PDB-module.html">Module extern.scientific_python.IO.PDB</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># This module handles input and output of PDB files.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># Written by Konrad Hinsen &lt;hinsen@cnrs-orleans.fr&gt;</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># Last revision: 2008-8-19</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># </tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">Parsing and writing of Protein Data Bank (PDB) files</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">This module provides classes that represent PDB (Protein Data Bank)</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">files and configurations contained in PDB files. It provides access to</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">PDB files on two levels: low-level (line by line) and high-level</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">(chains, residues, and atoms).</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">Caution: The PDB file format has been heavily abused, and it is</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">probably impossible to write code that can deal with all variants</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">correctly. This modules tries to read the widest possible range of PDB</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">files, but gives priority to a correct interpretation of the PDB</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">format as defined by the Brookhaven National Laboratory.</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">A special problem are atom names. The PDB file format specifies that</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">the first two letters contain the right-justified chemical element</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">name. A later modification allowed the initial space in hydrogen names</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">to be replaced by a digit. Many programs ignore all this and treat the</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">name as an arbitrary left-justified four-character name. This makes it</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">difficult to extract the chemical element accurately; most programs</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">write the '"CA"' for C_alpha in such a way that it actually stands for</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">a calcium atom. For this reason a special element field has been added</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">later, but only few files use it. In the absence of an element field,</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">the code in this module attempts to guess the element using all information</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring">available.</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">The low-level routines in this module do not try to deal with the atom</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">name problem; they return and expect four-character atom names</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">including spaces in the correct positions. The high-level routines use</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">atom names without leading or trailing spaces, but provide and use the</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring">element field whenever possible. For output, they use the element</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">field to place the atom name correctly, and for input, they construct</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">the element field content from the atom name if no explicit element</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">field is found in the file.</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">Except where indicated, numerical values use the same units and</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">conventions as specified in the PDB format description.</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">Example::</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">  &gt;&gt;&gt;conf = Structure('example.pdb')</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">  &gt;&gt;&gt;print conf</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">  &gt;&gt;&gt;for residue in conf.residues:</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">  &gt;&gt;&gt;    for atom in residue:</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">  &gt;&gt;&gt;        print atom</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: atom_format</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: anisou_format</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: conect_format</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: ter_format</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: model_format</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: header_format</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: cryst1_format</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: scalen_format</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: mtrixn_format</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: generic_format</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: export_filters</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">@undocumented: DummyChain</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package extern=extern-module.html"><a title="extern" class="py-name" href="#" onclick="return doclink('link-0', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package extern.scientific_python=extern.scientific_python-module.html"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-1', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Package extern.scientific_python.IO=extern.scientific_python.IO-module.html"><a title="extern.scientific_python.IO" class="py-name" href="#" onclick="return doclink('link-2', 'IO', 'link-2');">IO</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Module extern.scientific_python.IO.TextFile=extern.scientific_python.IO.TextFile-module.html,Class extern.scientific_python.IO.TextFile.TextFile=extern.scientific_python.IO.TextFile.TextFile-class.html"><a title="extern.scientific_python.IO.TextFile
extern.scientific_python.IO.TextFile.TextFile" class="py-name" href="#" onclick="return doclink('link-3', 'TextFile', 'link-3');">TextFile</a></tt> <tt class="py-keyword">import</tt> <tt id="link-4" class="py-name"><a title="extern.scientific_python.IO.TextFile
extern.scientific_python.IO.TextFile.TextFile" class="py-name" href="#" onclick="return doclink('link-4', 'TextFile', 'link-3');">TextFile</a></tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-5', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-6', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="extern.scientific_python.IO" class="py-name" href="#" onclick="return doclink('link-7', 'IO', 'link-2');">IO</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module extern.scientific_python.IO.FortranFormat=extern.scientific_python.IO.FortranFormat-module.html,Class extern.scientific_python.IO.FortranFormat.FortranFormat=extern.scientific_python.IO.FortranFormat.FortranFormat-class.html"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-8', 'FortranFormat', 'link-8');">FortranFormat</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-9', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class extern.scientific_python.IO.FortranFormat.FortranLine=extern.scientific_python.IO.FortranFormat.FortranLine-class.html"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-10', 'FortranLine', 'link-10');">FortranLine</a></tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-11', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-12', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package extern.scientific_python.Geometry=extern.scientific_python.Geometry-module.html"><a title="extern.scientific_python.Geometry" class="py-name" href="#" onclick="return doclink('link-13', 'Geometry', 'link-13');">Geometry</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">Vector</tt><tt class="py-op">,</tt> <tt id="link-14" class="py-name" targets="Class bmrblib.structure.tensor.Tensor=bmrblib.structure.tensor.Tensor-class.html"><a title="bmrblib.structure.tensor.Tensor" class="py-name" href="#" onclick="return doclink('link-14', 'Tensor', 'link-14');">Tensor</a></tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-15" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-15', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-16', 'scientific_python', 'link-1');">scientific_python</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">N</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">PDBExportFilters</tt> <tt class="py-keyword">import</tt> <tt class="py-name">export_filters</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-17" class="py-name" targets="Function generic_fns.align_tensor.copy()=generic_fns.align_tensor-module.html#copy,Function generic_fns.diffusion_tensor.copy()=generic_fns.diffusion_tensor-module.html#copy,Function generic_fns.pipes.copy()=generic_fns.pipes-module.html#copy,Function generic_fns.relax_data.copy()=generic_fns.relax_data-module.html#copy,Function generic_fns.sequence.copy()=generic_fns.sequence-module.html#copy,Function generic_fns.value.copy()=generic_fns.value-module.html#copy,Method gui.user_functions.molecule.Molecule.copy()=gui.user_functions.molecule.Molecule-class.html#copy,Method gui.user_functions.pipe.Pipe.copy()=gui.user_functions.pipe.Pipe-class.html#copy,Method gui.user_functions.residue.Residue.copy()=gui.user_functions.residue.Residue-class.html#copy,Method gui.user_functions.sequence.Sequence.copy()=gui.user_functions.sequence.Sequence-class.html#copy,Method gui.user_functions.spin.Spin.copy()=gui.user_functions.spin.Spin-class.html#copy,Method prompt.align_tensor.Align_tensor.copy()=prompt.align_tensor.Align_tensor-class.html#copy,Method prompt.diffusion_tensor.Diffusion_tensor.copy()=prompt.diffusion_tensor.Diffusion_tensor-class.html#copy,Method prompt.molecule.Molecule.copy()=prompt.molecule.Molecule-class.html#copy,Method prompt.pcs.PCS.copy()=prompt.pcs.PCS-class.html#copy,Method prompt.pipe.Pipe.copy()=prompt.pipe.Pipe-class.html#copy,Method prompt.rdc.RDC.copy()=prompt.rdc.RDC-class.html#copy,Method prompt.relax_data.Relax_data.copy()=prompt.relax_data.Relax_data-class.html#copy,Method prompt.residue.Residue.copy()=prompt.residue.Residue-class.html#copy,Method prompt.sequence.Sequence.copy()=prompt.sequence.Sequence-class.html#copy,Method prompt.spin.Spin.copy()=prompt.spin.Spin-class.html#copy,Method prompt.value.Value.copy()=prompt.value.Value-class.html#copy"><a title="generic_fns.align_tensor.copy
generic_fns.diffusion_tensor.copy
generic_fns.pipes.copy
generic_fns.relax_data.copy
generic_fns.sequence.copy
generic_fns.value.copy
gui.user_functions.molecule.Molecule.copy
gui.user_functions.pipe.Pipe.copy
gui.user_functions.residue.Residue.copy
gui.user_functions.sequence.Sequence.copy
gui.user_functions.spin.Spin.copy
prompt.align_tensor.Align_tensor.copy
prompt.diffusion_tensor.Diffusion_tensor.copy
prompt.molecule.Molecule.copy
prompt.pcs.PCS.copy
prompt.pipe.Pipe.copy
prompt.rdc.RDC.copy
prompt.relax_data.Relax_data.copy
prompt.residue.Residue.copy
prompt.sequence.Sequence.copy
prompt.spin.Spin.copy
prompt.value.Value.copy" class="py-name" href="#" onclick="return doclink('link-17', 'copy', 'link-17');">copy</a></tt><tt class="py-op">,</tt> <tt class="py-name">string</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-comment"># Fortran formats for PDB entries</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-name">atom_format</tt> <tt class="py-op">=</tt> <tt id="link-18" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-18', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,I5,1X,A4,A1,A4,A1,I4,A1,3X,3F8.3,2F6.2,'</tt> <tt class="py-op">+</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">                            <tt class="py-string">'6X,A4,2A2'</tt><tt class="py-op">)</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-name">anisou_format</tt> <tt class="py-op">=</tt> <tt id="link-19" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-19', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,I5,1X,A4,A1,A4,A1,I4,A1,1X,6I7,2X,A4,2A2'</tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-name">conect_format</tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-20', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,11I5'</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-name">ter_format</tt> <tt class="py-op">=</tt> <tt id="link-21" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-21', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,I5,6X,A4,A1,I4,A1'</tt><tt class="py-op">)</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-name">model_format</tt> <tt class="py-op">=</tt> <tt id="link-22" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-22', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,4X,I4'</tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-name">header_format</tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-23', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,4X,A40,A9,3X,A4'</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-name">cryst1_format</tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-24', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,3F9.3,3F7.2,1X,A11,I4'</tt><tt class="py-op">)</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-name">scalen_format</tt> <tt class="py-op">=</tt> <tt id="link-25" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-25', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,4X,3F10.6,5X,F10.5'</tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-name">mtrixn_format</tt> <tt class="py-op">=</tt> <tt id="link-26" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-26', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,1X,I3,3F10.6,5X,F10.5,4X,I1'</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-name">generic_format</tt> <tt class="py-op">=</tt> <tt id="link-27" class="py-name"><a title="extern.scientific_python.IO.FortranFormat
extern.scientific_python.IO.FortranFormat.FortranFormat" class="py-name" href="#" onclick="return doclink('link-27', 'FortranFormat', 'link-8');">FortranFormat</a></tt><tt class="py-op">(</tt><tt class="py-string">'A6,A74'</tt><tt class="py-op">)</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-comment"># Amino acid and nucleic acid residues</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt id="link-28" class="py-name" targets="Variable extern.scientific_python.IO.PDB.amino_acids=extern.scientific_python.IO.PDB-module.html#amino_acids"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-28', 'amino_acids', 'link-28');">amino_acids</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'ALA'</tt><tt class="py-op">,</tt> <tt class="py-string">'ARG'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASN'</tt><tt class="py-op">,</tt> <tt class="py-string">'ASP'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'CYX'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLN'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLU'</tt><tt class="py-op">,</tt> <tt class="py-string">'GLY'</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line">               <tt class="py-string">'HIS'</tt><tt class="py-op">,</tt> <tt class="py-string">'HID'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HIP'</tt><tt class="py-op">,</tt> <tt class="py-string">'HSD'</tt><tt class="py-op">,</tt> <tt class="py-string">'HSE'</tt><tt class="py-op">,</tt> <tt class="py-string">'HSP'</tt><tt class="py-op">,</tt> <tt class="py-string">'ILE'</tt><tt class="py-op">,</tt> <tt class="py-string">'LEU'</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">               <tt class="py-string">'LYS'</tt><tt class="py-op">,</tt> <tt class="py-string">'MET'</tt><tt class="py-op">,</tt> <tt class="py-string">'PHE'</tt><tt class="py-op">,</tt> <tt class="py-string">'PRO'</tt><tt class="py-op">,</tt> <tt class="py-string">'SER'</tt><tt class="py-op">,</tt> <tt class="py-string">'THR'</tt><tt class="py-op">,</tt> <tt class="py-string">'TRP'</tt><tt class="py-op">,</tt> <tt class="py-string">'TYR'</tt><tt class="py-op">,</tt> <tt class="py-string">'VAL'</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">               <tt class="py-string">'ACE'</tt><tt class="py-op">,</tt> <tt class="py-string">'NME'</tt><tt class="py-op">,</tt> <tt class="py-string">'NHE'</tt><tt class="py-op">]</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt id="link-29" class="py-name" targets="Variable extern.scientific_python.IO.PDB.nucleic_acids=extern.scientific_python.IO.PDB-module.html#nucleic_acids"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-29', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> <tt class="py-string">'A'</tt><tt class="py-op">,</tt>  <tt class="py-string">'C'</tt><tt class="py-op">,</tt>  <tt class="py-string">'G'</tt><tt class="py-op">,</tt>  <tt class="py-string">'I'</tt><tt class="py-op">,</tt>  <tt class="py-string">'T'</tt><tt class="py-op">,</tt>  <tt class="py-string">'U'</tt><tt class="py-op">,</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">                 <tt class="py-string">'+A'</tt><tt class="py-op">,</tt> <tt class="py-string">'+C'</tt><tt class="py-op">,</tt> <tt class="py-string">'+G'</tt><tt class="py-op">,</tt> <tt class="py-string">'+I'</tt><tt class="py-op">,</tt> <tt class="py-string">'+T'</tt><tt class="py-op">,</tt> <tt class="py-string">'+U'</tt><tt class="py-op">,</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">                  <tt class="py-string">'RA'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RC'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RG'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RU'</tt><tt class="py-op">,</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line">                  <tt class="py-string">'DA'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DC'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DG'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DT'</tt><tt class="py-op">,</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line">                  <tt class="py-string">'RA5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RC5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RG5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RU5'</tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">                  <tt class="py-string">'DA5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DC5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DG5'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DT5'</tt><tt class="py-op">,</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">                  <tt class="py-string">'RA3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RC3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RG3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RU3'</tt><tt class="py-op">,</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line">                  <tt class="py-string">'DA3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DC3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DG3'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DT3'</tt><tt class="py-op">,</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">                  <tt class="py-string">'RAN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RCN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RGN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'RUN'</tt><tt class="py-op">,</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">                  <tt class="py-string">'DAN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DCN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DGN'</tt><tt class="py-op">,</tt>  <tt class="py-string">'DTN'</tt><tt class="py-op">,</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">                  <tt class="py-op">]</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"> </tt>
<a name="defineAminoAcidResidue"></a><div id="defineAminoAcidResidue-def"><a name="L109"></a><tt class="py-lineno"> 109</tt> <a class="py-toggle" href="#" id="defineAminoAcidResidue-toggle" onclick="return toggle('defineAminoAcidResidue');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB-module.html#defineAminoAcidResidue">defineAminoAcidResidue</a><tt class="py-op">(</tt><tt class="py-param">symbol</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="defineAminoAcidResidue-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="defineAminoAcidResidue-expanded"><a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">    Make the parser recognize a particular residue type as an amino</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring">    acid residue</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">    @param symbol: the three-letter code for an amino acid</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">    @type symbol: C{str}</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line">    <tt class="py-name">symbol</tt> <tt class="py-op">=</tt> <tt class="py-name">symbol</tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">symbol</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-30" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-30', 'amino_acids', 'link-28');">amino_acids</a></tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line">        <tt id="link-31" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-31', 'amino_acids', 'link-28');">amino_acids</a></tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method data.align_tensor.AlignTensorSimList.append()=data.align_tensor.AlignTensorSimList-class.html#append,Method data.diff_tensor.DiffTensorSimList.append()=data.diff_tensor.DiffTensorSimList-class.html#append"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-32', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">symbol</tt><tt class="py-op">)</tt> </tt>
</div><a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"> </tt>
<a name="defineNucleicAcidResidue"></a><div id="defineNucleicAcidResidue-def"><a name="L121"></a><tt class="py-lineno"> 121</tt> <a class="py-toggle" href="#" id="defineNucleicAcidResidue-toggle" onclick="return toggle('defineNucleicAcidResidue');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB-module.html#defineNucleicAcidResidue">defineNucleicAcidResidue</a><tt class="py-op">(</tt><tt class="py-param">symbol</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="defineNucleicAcidResidue-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="defineNucleicAcidResidue-expanded"><a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">    Make the parser recognize a particular residue type as an nucleic</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">    acid residue</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-docstring">    @param symbol: the one-letter code for a nucleic acid</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"><tt class="py-docstring">    @type symbol: C{str}</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">    <tt class="py-name">symbol</tt> <tt class="py-op">=</tt> <tt class="py-name">symbol</tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">symbol</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-33" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-33', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt><tt class="py-op">:</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">        <tt id="link-34" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-34', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-35', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">symbol</tt><tt class="py-op">)</tt> </tt>
</div><a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line"> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"><tt class="py-comment"># Low-level file object. It represents line contents as Python dictionaries.</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"><tt class="py-comment"># For output, there are additional methods that generate sequence numbers</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-comment"># for everything.</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="PDBFile"></a><div id="PDBFile-def"><a name="L139"></a><tt class="py-lineno"> 139</tt> <a class="py-toggle" href="#" id="PDBFile-toggle" onclick="return toggle('PDBFile');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html">PDBFile</a><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PDBFile-expanded"><a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">    X{PDB} file with access at the record level</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">    The low-level file access is handled by the module</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">    L{extern.scientific_python.IO.TextFile}, therefore compressed files and URLs</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">    (for reading) can be used as well.</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.__init__"></a><div id="PDBFile.__init__-def"><a name="L149"></a><tt class="py-lineno"> 149</tt> <a class="py-toggle" href="#" id="PDBFile.__init__-toggle" onclick="return toggle('PDBFile.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file_or_filename</tt><tt class="py-op">,</tt> <tt class="py-param">mode</tt> <tt class="py-op">=</tt> <tt class="py-string">'r'</tt><tt class="py-op">,</tt> <tt class="py-param">subformat</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.__init__-expanded"><a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">        @param file_or_filename: the name of the PDB file, or a file object</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring">        @type file_or_filename: C{str} or C{file}</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">        @param mode: the file access mode, 'r' (read) or 'w' (write)</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">        @type mode: C{str}</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">        @param subformat: indicates a specific dialect of the PDB format.</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">                          Subformats are defined in</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring">                          L{extern.scientific_python.IO.PDBExportFilters}; they are used</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">                          only when writing.</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring">        @type subformat: C{str} or C{NoneType}</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">file_or_filename</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Variable dep_check.file=dep_check-module.html#file"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-36', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-37" class="py-name"><a title="extern.scientific_python.IO.TextFile
extern.scientific_python.IO.TextFile.TextFile" class="py-name" href="#" onclick="return doclink('link-37', 'TextFile', 'link-3');">TextFile</a></tt><tt class="py-op">(</tt><tt class="py-name">file_or_filename</tt><tt class="py-op">,</tt> <tt class="py-name">mode</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-38', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt class="py-name">file_or_filename</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">lower</tt><tt class="py-op">(</tt><tt class="py-name">mode</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'w'</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">subformat</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">            <tt class="py-name">export</tt> <tt class="py-op">=</tt> <tt class="py-name">export_filters</tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-name">subformat</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">export</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-op">=</tt> <tt class="py-name">export</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">open</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">output</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Package data=data-module.html"><a title="data" class="py-name" href="#" onclick="return doclink('link-39', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">                         <tt class="py-string">'residue_number'</tt><tt class="py-op">:</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">                         <tt class="py-string">'chain_id'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">                         <tt class="py-string">'segment_id'</tt><tt class="py-op">:</tt> <tt class="py-string">''</tt><tt class="py-op">}</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">het_flag</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_number</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
</div><a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.readLine"></a><div id="PDBFile.readLine-def"><a name="L180"></a><tt class="py-lineno"> 180</tt> <a class="py-toggle" href="#" id="PDBFile.readLine-toggle" onclick="return toggle('PDBFile.readLine');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#readLine">readLine</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.readLine-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.readLine-expanded"><a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring">        Return the contents of the next non-blank line (= record) The</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">        return value is a tuple whose first element (a string)</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"><tt class="py-docstring">        contains the record type. For supported record types (HEADER,</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line"><tt class="py-docstring">        CRYST1, SCALEn, MTRIXn, ATOM, HETATM, ANISOU, TERM, MODEL,</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"><tt class="py-docstring">        CONECT), the items from the remaining fields are put into a</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring">        dictionary which is returned as the second tuple element. Most</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">        dictionary elements are strings or numbers; atom positions are</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">        returned as a vector, and anisotropic temperature factors are</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring">        returned as a rank-2 tensor, already multiplied by 1.e-4.</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">        White space is stripped from all strings except for atom</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring">        names, whose correct interpretation can depend on an initial</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">        space. For unsupported record types, the second tuple element</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">        is a string containing the remaining part of the record.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the contents of one PDB record</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{tuple}</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-40', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Method extern.scientific_python.IO.TextFile.TextFile.readline()=extern.scientific_python.IO.TextFile.TextFile-class.html#readline"><a title="extern.scientific_python.IO.TextFile.TextFile.readline" class="py-name" href="#" onclick="return doclink('link-41', 'readline', 'link-41');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">'END'</tt><tt class="py-op">,</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'\n'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Function relax_io.strip()=relax_io-module.html#strip"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-42', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> <tt class="py-keyword">break</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">ljust</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-number">80</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">        <tt id="link-43" class="py-name" targets="Method extern.scientific_python.IO.PDB.Atom.type()=extern.scientific_python.IO.PDB.Atom-class.html#type,Method extern.scientific_python.IO.PDB.HetAtom.type()=extern.scientific_python.IO.PDB.HetAtom-class.html#type,Variable info.Bieri11.type=info.Bieri11-class.html#type,Variable info.Clore90.type=info.Clore90-class.html#type,Variable info.Delaglio95.type=info.Delaglio95-class.html#type,Variable info.GoddardKneller.type=info.GoddardKneller-class.html#type,Variable info.LipariSzabo82a.type=info.LipariSzabo82a-class.html#type,Variable info.LipariSzabo82b.type=info.LipariSzabo82b-class.html#type,Variable info.Ref.type=info.Ref-class.html#type,Variable info.dAuvergne06.type=info.dAuvergne06-class.html#type,Variable info.dAuvergneGooley03.type=info.dAuvergneGooley03-class.html#type,Variable info.dAuvergneGooley06.type=info.dAuvergneGooley06-class.html#type,Variable info.dAuvergneGooley07.type=info.dAuvergneGooley07-class.html#type,Variable info.dAuvergneGooley08a.type=info.dAuvergneGooley08a-class.html#type,Variable info.dAuvergneGooley08b.type=info.dAuvergneGooley08b-class.html#type,Method prompt.molecule.Molecule.type()=prompt.molecule.Molecule-class.html#type"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-43', 'type', 'link-43');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-44', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-45" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-45', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ATOM'</tt> <tt class="py-keyword">or</tt> <tt id="link-46" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-46', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HETATM'</tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-47" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-47', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">atom_format</tt><tt class="py-op">)</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">            <tt id="link-48" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-48', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">                    <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">                    <tt class="py-string">'alternate'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-49', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">                    <tt class="py-string">'residue_name'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-50', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">                    <tt class="py-string">'chain_id'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-51', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">                    <tt class="py-string">'residue_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">                    <tt class="py-string">'insertion_code'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-52', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">                    <tt class="py-string">'position'</tt><tt class="py-op">:</tt> <tt class="py-name">Vector</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">:</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">                    <tt class="py-string">'occupancy'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">                    <tt class="py-string">'temperature_factor'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">                    <tt class="py-string">'segment_id'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-53', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">                    <tt class="py-string">'element'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-54', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">                    <tt class="py-string">'charge'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-55', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-56" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-56', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-57" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-57', 'data', 'link-39');">data</a></tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-58" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-58', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ANISOU'</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-59" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-59', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">anisou_format</tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">            <tt id="link-60" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-60', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line">                    <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">                    <tt class="py-string">'alternate'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-61', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">                    <tt class="py-string">'residue_name'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-62', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">                    <tt class="py-string">'chain_id'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-63', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">                    <tt class="py-string">'residue_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">                    <tt class="py-string">'insertion_code'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-64', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">                    <tt class="py-string">'u'</tt><tt class="py-op">:</tt> <tt class="py-number">1.e-4</tt><tt class="py-op">*</tt><tt id="link-65" class="py-name"><a title="bmrblib.structure.tensor.Tensor" class="py-name" href="#" onclick="return doclink('link-65', 'Tensor', 'link-14');">Tensor</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">                                       <tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt> <tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">                                       <tt class="py-op">[</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">                    <tt class="py-string">'segment_id'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-66', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">                    <tt class="py-string">'element'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-67', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">                    <tt class="py-string">'charge'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-68', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">16</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-69" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-69', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-70" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-70', 'data', 'link-39');">data</a></tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-71" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-71', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'TER'</tt><tt class="py-op">:</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-72" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-72', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">ter_format</tt><tt class="py-op">)</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">            <tt id="link-73" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-73', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">                    <tt class="py-string">'residue_name'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-74', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">                    <tt class="py-string">'chain_id'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-75', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line">                    <tt class="py-string">'residue_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">                    <tt class="py-string">'insertion_code'</tt><tt class="py-op">:</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-76', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">}</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-77" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-77', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-78', 'data', 'link-39');">data</a></tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-79" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-79', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CONECT'</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-80" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-80', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">conect_format</tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">            <tt id="link-81" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-81', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">                    <tt class="py-string">'bonded'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt id="link-82" class="py-name" targets="Variable prompt.eliminate.Eliminate.i=prompt.eliminate.Eliminate-class.html#i,Variable prompt.minimisation.Minimisation.i=prompt.minimisation.Minimisation-class.html#i"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-82', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">for</tt> <tt id="link-83" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-83', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> <tt class="py-keyword">if</tt> <tt id="link-84" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-84', 'i', 'link-82');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">                    <tt class="py-string">'hydrogen_bonded'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt id="link-85" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-85', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">for</tt> <tt id="link-86" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-86', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-number">10</tt><tt class="py-op">]</tt> <tt class="py-keyword">if</tt> <tt id="link-87" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-87', 'i', 'link-82');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">                    <tt class="py-string">'salt_bridged'</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt><tt id="link-88" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-88', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">for</tt> <tt id="link-89" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-89', 'i', 'link-82');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-number">12</tt><tt class="py-op">]</tt> <tt class="py-keyword">if</tt> <tt id="link-90" class="py-name"><a title="prompt.eliminate.Eliminate.i
prompt.minimisation.Minimisation.i" class="py-name" href="#" onclick="return doclink('link-90', 'i', 'link-82');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-91" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-91', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-92" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-92', 'data', 'link-39');">data</a></tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-93" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-93', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'MODEL'</tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-94', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">model_format</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">            <tt id="link-95" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-95', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-96" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-96', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-97" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-97', 'data', 'link-39');">data</a></tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-98" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-98', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HEADER'</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-99" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-99', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">header_format</tt><tt class="py-op">)</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">            <tt id="link-100" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-100', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'compound'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">                    <tt class="py-string">'date'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">                    <tt class="py-string">'pdb_code'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-101" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-101', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-102" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-102', 'data', 'link-39');">data</a></tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-103" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-103', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CRYST1'</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-104" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-104', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">cryst1_format</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">            <tt id="link-105" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-105', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'a'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">                    <tt class="py-string">'b'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">                    <tt class="py-string">'c'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">                    <tt class="py-string">'alpha'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">                    <tt class="py-string">'beta'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">                    <tt class="py-string">'gamma'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">                    <tt class="py-string">'space_group'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">                    <tt class="py-string">'z'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-106" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-106', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-107" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-107', 'data', 'link-39');">data</a></tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-108" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-108', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'SCALE'</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-109', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">scalen_format</tt><tt class="py-op">)</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">            <tt id="link-110" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-110', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'s1'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">                    <tt class="py-string">'s2'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">                    <tt class="py-string">'s3'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">                    <tt class="py-string">'u'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-111" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-111', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-112" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-112', 'data', 'link-39');">data</a></tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-113" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-113', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'MTRIX'</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-114" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-114', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">mtrixn_format</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">            <tt id="link-115" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-115', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'serial'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">                    <tt class="py-string">'m1'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">                    <tt class="py-string">'m2'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">                    <tt class="py-string">'m3'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">                    <tt class="py-string">'v'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">                    <tt class="py-string">'given'</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">}</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-116" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-116', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-117" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-117', 'data', 'link-39');">data</a></tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-118" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-118', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
</div><a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.writeLine"></a><div id="PDBFile.writeLine-def"><a name="L294"></a><tt class="py-lineno"> 294</tt> <a class="py-toggle" href="#" id="PDBFile.writeLine-toggle" onclick="return toggle('PDBFile.writeLine');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#writeLine">writeLine</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">,</tt> <tt class="py-param">data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.writeLine-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.writeLine-expanded"><a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">        Write a line using record type and data dictionary in the</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"><tt class="py-docstring">        same format as returned by readLine(). Default values are</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"><tt class="py-docstring">        provided for non-essential information, so the data dictionary</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"><tt class="py-docstring">        need not contain all entries.</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"><tt class="py-docstring">        @param type: PDB record type</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"><tt class="py-docstring">        @type type: C{str}</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">        @param data: PDB record data</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring">        @type data: C{tuple}</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line">            <tt id="link-119" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-119', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-120" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-120', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt><tt class="py-op">.</tt><tt class="py-name">processLine</tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-121', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-122" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-122', 'data', 'link-39');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-123" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-123', 'type', 'link-43');">type</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-124" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-124', 'type', 'link-43');">type</a></tt><tt class="py-op">]</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-125" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-125', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ATOM'</tt> <tt class="py-keyword">or</tt> <tt id="link-126" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-126', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HETATM'</tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">            <tt id="link-127" class="py-name" targets="Function relax_warnings.format()=relax_warnings-module.html#format"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-127', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">atom_format</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">            <tt class="py-name">position</tt> <tt class="py-op">=</tt> <tt id="link-128" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-128', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'position'</tt><tt class="py-op">]</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-129" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-129', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">                           <tt id="link-130" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-130', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'name'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">                           <tt id="link-131" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-131', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'alternate'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">                           <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt id="link-132" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-132', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">                           <tt id="link-133" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-133', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">                           <tt id="link-134" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-134', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">                           <tt id="link-135" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-135', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">                           <tt class="py-name">position</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">position</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">position</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">                           <tt id="link-136" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-136', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'occupancy'</tt><tt class="py-op">,</tt> <tt class="py-number">0.</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">                           <tt id="link-137" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-137', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'temperature_factor'</tt><tt class="py-op">,</tt> <tt class="py-number">0.</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">                           <tt id="link-138" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-138', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                           <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-139', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'element'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">                           <tt id="link-140" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-140', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'charge'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-141" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-141', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ANISOU'</tt><tt class="py-op">:</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">            <tt id="link-142" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-142', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">anisou_format</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">            <tt class="py-name">u</tt> <tt class="py-op">=</tt> <tt class="py-number">1.e4</tt><tt class="py-op">*</tt><tt id="link-143" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-143', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">            <tt class="py-name">u</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">,</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">,</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">                 <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">,</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">,</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-144" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-144', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">                           <tt id="link-145" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-145', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'name'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">                           <tt id="link-146" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-146', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'alternate'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">                           <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-147', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">                           <tt id="link-148" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-148', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">                           <tt id="link-149" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-149', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">                           <tt id="link-150" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-150', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> \ </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">                        <tt class="py-op">+</tt> <tt class="py-name">u</tt> \ </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">                        <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-151" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-151', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">                           <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-152', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'element'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">                           <tt id="link-153" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-153', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'charge'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-154" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-154', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'TER'</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">            <tt id="link-155" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-155', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">ter_format</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-156" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-156', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">                           <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">rjust</tt><tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-157', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">                           <tt id="link-158" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-158', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">                           <tt id="link-159" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-159', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">                           <tt id="link-160" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-160', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-161" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-161', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CONECT'</tt><tt class="py-op">:</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">            <tt id="link-162" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-162', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">conect_format</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-163" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-163', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-164', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'bonded'</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">+</tt><tt class="py-number">4</tt><tt class="py-op">*</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-165" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-165', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'hydrogen_bonded'</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">+</tt><tt class="py-number">4</tt><tt class="py-op">*</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-166" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-166', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'salt_bridged'</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">+</tt><tt class="py-number">2</tt><tt class="py-op">*</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-167" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-167', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'MODEL'</tt><tt class="py-op">:</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">            <tt id="link-168" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-168', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">model_format</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-169" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-169', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-170" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-170', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CRYST1'</tt><tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">            <tt id="link-171" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-171', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cryst1_format</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-172" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-172', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'a'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-173', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'b'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-174" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-174', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'c'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">                           <tt id="link-175" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-175', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'alpha'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-176" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-176', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'beta'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">                           <tt id="link-177" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-177', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'gamma'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">                           <tt id="link-178" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-178', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'space_group'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">                           <tt id="link-179" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-179', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'z'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-180" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-180', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'SCALE'</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt id="link-181" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-181', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">scalen_format</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-182" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-182', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'s1'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-183" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-183', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'s2'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-184', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'s3'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">                           <tt id="link-185" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-185', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'u'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-186" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-186', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'MTRIX'</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">            <tt id="link-187" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-187', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">scalen_format</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-188" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-188', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'serial'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">                           <tt id="link-189" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-189', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'m1'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-190" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-190', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'m2'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-191" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-191', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'m3'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">                           <tt id="link-192" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-192', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'v'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-193" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-193', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'given'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-194" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-194', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HEADER'</tt><tt class="py-op">:</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">            <tt id="link-195" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-195', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">header_format</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-196" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-196', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'compound'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-197" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-197', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'date'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">                           <tt id="link-198" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-198', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'pdb_code'</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">            <tt id="link-199" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-199', 'format', 'link-127');">format</a></tt> <tt class="py-op">=</tt> <tt class="py-name">generic_format</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-200" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-200', 'data', 'link-39');">data</a></tt><tt class="py-op">]</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-201', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Method bmrblib.nmr_star_dict.NMR_STAR.write()=bmrblib.nmr_star_dict.NMR_STAR-class.html#write,Method bmrblib.pystarlib.File.File.write()=bmrblib.pystarlib.File.File-class.html#write,Method extern.scientific_python.IO.TextFile.TextFile.write()=extern.scientific_python.IO.TextFile.TextFile-class.html#write,Function generic_fns.bmrb.write()=generic_fns.bmrb-module.html#write,Function generic_fns.grace.write()=generic_fns.grace-module.html#write,Function generic_fns.pcs.write()=generic_fns.pcs-module.html#write,Function generic_fns.rdc.write()=generic_fns.rdc-module.html#write,Function generic_fns.relax_data.write()=generic_fns.relax_data-module.html#write,Function generic_fns.results.write()=generic_fns.results-module.html#write,Function generic_fns.sequence.write()=generic_fns.sequence-module.html#write,Function generic_fns.value.write()=generic_fns.value-module.html#write,Method gui.controller.LogCtrl.write()=gui.controller.LogCtrl-class.html#write,Method gui.controller.Redirect_text.write()=gui.controller.Redirect_text-class.html#write,Method gui.user_functions.grace.Grace.write()=gui.user_functions.grace.Grace-class.html#write,Method gui.user_functions.results.Results.write()=gui.user_functions.results.Results-class.html#write,Method gui.user_functions.sequence.Sequence.write()=gui.user_functions.sequence.Sequence-class.html#write,Method multi.processor_io.IO_filter.write()=multi.processor_io.IO_filter-class.html#write,Method multi.processor_io.PrependStringIO.write()=multi.processor_io.PrependStringIO-class.html#write,Method prompt.bmrb.BMRB.write()=prompt.bmrb.BMRB-class.html#write,Method prompt.grace.Grace.write()=prompt.grace.Grace-class.html#write,Method prompt.pcs.PCS.write()=prompt.pcs.PCS-class.html#write,Method prompt.rdc.RDC.write()=prompt.rdc.RDC-class.html#write,Method prompt.relax_data.Relax_data.write()=prompt.relax_data.Relax_data-class.html#write,Method prompt.results.Results.write()=prompt.results.Results-class.html#write,Method prompt.sequence.Sequence.write()=prompt.sequence.Sequence-class.html#write,Method prompt.value.Value.write()=prompt.value.Value-class.html#write,Method relax_io.DummyFileObject.write()=relax_io.DummyFileObject-class.html#write,Method relax_io.SplitIO.write()=relax_io.SplitIO-class.html#write"><a title="bmrblib.nmr_star_dict.NMR_STAR.write
bmrblib.pystarlib.File.File.write
extern.scientific_python.IO.TextFile.TextFile.write
generic_fns.bmrb.write
generic_fns.grace.write
generic_fns.pcs.write
generic_fns.rdc.write
generic_fns.relax_data.write
generic_fns.results.write
generic_fns.sequence.write
generic_fns.value.write
gui.controller.LogCtrl.write
gui.controller.Redirect_text.write
gui.user_functions.grace.Grace.write
gui.user_functions.results.Results.write
gui.user_functions.sequence.Sequence.write
multi.processor_io.IO_filter.write
multi.processor_io.PrependStringIO.write
prompt.bmrb.BMRB.write
prompt.grace.Grace.write
prompt.pcs.PCS.write
prompt.rdc.RDC.write
prompt.relax_data.Relax_data.write
prompt.results.Results.write
prompt.sequence.Sequence.write
prompt.value.Value.write
relax_io.DummyFileObject.write
relax_io.SplitIO.write" class="py-name" href="#" onclick="return doclink('link-202', 'write', 'link-202');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-203" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine" class="py-name" href="#" onclick="return doclink('link-203', 'FortranLine', 'link-10');">FortranLine</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt id="link-204" class="py-name"><a title="relax_warnings.format" class="py-name" href="#" onclick="return doclink('link-204', 'format', 'link-127');">format</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.writeComment"></a><div id="PDBFile.writeComment-def"><a name="L384"></a><tt class="py-lineno"> 384</tt> <a class="py-toggle" href="#" id="PDBFile.writeComment-toggle" onclick="return toggle('PDBFile.writeComment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#writeComment">writeComment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.writeComment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.writeComment-expanded"><a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line"><tt class="py-docstring">        Write text into one or several comment lines.</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-docstring">        Each line of the text is prefixed with 'REMARK' and written</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-docstring">        to the file.</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line"><tt class="py-docstring">        @param text: the comment contents</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line"><tt class="py-docstring">        @type text: C{str}</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt id="link-205" class="py-name" targets="Method gui.wizard.Wiz_page.text()=gui.wizard.Wiz_page-class.html#text,Variable prompt.help._Helper.text=prompt.help._Helper-class.html#text,Variable prompt.help._Helper_python.text=prompt.help._Helper_python-class.html#text"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-205', 'text', 'link-205');">text</a></tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">            <tt class="py-name">eol</tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Method gui.controller.LogCtrl.find()=gui.controller.LogCtrl-class.html#find"><a title="gui.controller.LogCtrl.find" class="py-name" href="#" onclick="return doclink('link-206', 'find', 'link-206');">find</a></tt><tt class="py-op">(</tt><tt id="link-207" class="py-name"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-207', 'text', 'link-205');">text</a></tt><tt class="py-op">,</tt><tt class="py-string">'\n'</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">eol</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">                <tt class="py-name">eol</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-208" class="py-name"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-208', 'text', 'link-205');">text</a></tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-209', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="bmrblib.nmr_star_dict.NMR_STAR.write
bmrblib.pystarlib.File.File.write
extern.scientific_python.IO.TextFile.TextFile.write
generic_fns.bmrb.write
generic_fns.grace.write
generic_fns.pcs.write
generic_fns.rdc.write
generic_fns.relax_data.write
generic_fns.results.write
generic_fns.sequence.write
generic_fns.value.write
gui.controller.LogCtrl.write
gui.controller.Redirect_text.write
gui.user_functions.grace.Grace.write
gui.user_functions.results.Results.write
gui.user_functions.sequence.Sequence.write
multi.processor_io.IO_filter.write
multi.processor_io.PrependStringIO.write
prompt.bmrb.BMRB.write
prompt.grace.Grace.write
prompt.pcs.PCS.write
prompt.rdc.RDC.write
prompt.relax_data.Relax_data.write
prompt.results.Results.write
prompt.sequence.Sequence.write
prompt.value.Value.write
relax_io.DummyFileObject.write
relax_io.SplitIO.write" class="py-name" href="#" onclick="return doclink('link-210', 'write', 'link-202');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'REMARK %s \n'</tt> <tt class="py-op">%</tt> <tt id="link-211" class="py-name"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-211', 'text', 'link-205');">text</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">eol</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">            <tt id="link-212" class="py-name"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-212', 'text', 'link-205');">text</a></tt> <tt class="py-op">=</tt> <tt id="link-213" class="py-name"><a title="gui.wizard.Wiz_page.text
prompt.help._Helper.text
prompt.help._Helper_python.text" class="py-name" href="#" onclick="return doclink('link-213', 'text', 'link-205');">text</a></tt><tt class="py-op">[</tt><tt class="py-name">eol</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
</div><a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.writeAtom"></a><div id="PDBFile.writeAtom-def"><a name="L400"></a><tt class="py-lineno"> 400</tt> <a class="py-toggle" href="#" id="PDBFile.writeAtom-toggle" onclick="return toggle('PDBFile.writeAtom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#writeAtom">writeAtom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">position</tt><tt class="py-op">,</tt> <tt class="py-param">occupancy</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt> <tt class="py-param">temperature_factor</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">                  <tt class="py-param">element</tt><tt class="py-op">=</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.writeAtom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.writeAtom-expanded"><a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line"><tt class="py-docstring">        Write an ATOM or HETATM record using the information supplied.</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"><tt class="py-docstring">        The residue and chain information is taken from the last calls to</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line"><tt class="py-docstring">        the methods L{nextResidue} and L{nextChain}.</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"><tt class="py-docstring">        @param name: the atom name</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-docstring">        @type name: C{str}</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"><tt class="py-docstring">        @param position: the atom position</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line"><tt class="py-docstring">        @type position: L{extern.scientific_python.Geometry.Vector}</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-docstring">        @param occupancy: the occupancy</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-docstring">        @type occupancy: C{float}</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-docstring">        @param temperature_factor: the temperature factor (B-factor)</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-docstring">        @type temperature_factor: C{float}</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-docstring">        @param element: the chemical element</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">        @type element: C{str}</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">het_flag</tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">            <tt id="link-214" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-214', 'type', 'link-43');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'HETATM'</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">            <tt id="link-215" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-215', 'type', 'link-43');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'ATOM'</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">        <tt id="link-216" class="py-name" targets="Variable bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name=bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe-class.html#name,Variable bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name=bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1-class.html#name,Variable bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name=bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1-class.html#name,Variable bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name=bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1-class.html#name,Variable bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name=bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1-class.html#name,Variable bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name=bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1-class.html#name,Variable bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name=bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1-class.html#name,Method gui.user_functions.spin.Spin.name()=gui.user_functions.spin.Spin-class.html#name,Method prompt.molecule.Molecule.name()=prompt.molecule.Molecule-class.html#name,Method prompt.residue.Residue.name()=prompt.residue.Residue-class.html#name,Method prompt.spin.Spin.name()=prompt.spin.Spin-class.html#name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-216', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt id="link-217" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-217', 'name', 'link-216');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-218" class="py-name" targets="Method gui.user_functions.spin.Spin.element()=gui.user_functions.spin.Spin-class.html#element,Method prompt.spin.Spin.element()=prompt.spin.Spin-class.html#element"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-218', 'element', 'link-218');">element</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-219" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-219', 'element', 'link-218');">element</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">and</tt> <tt id="link-220" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-220', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">and</tt> <tt id="link-221" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-221', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-222" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-222', 'element', 'link-218');">element</a></tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">            <tt id="link-223" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-223', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt id="link-224" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-224', 'name', 'link-216');">name</a></tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-225', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'name'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-226" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-226', 'name', 'link-216');">name</a></tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-227', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'position'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">position</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-228', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-229', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">100000</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-230', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'occupancy'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">occupancy</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-231', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'temperature_factor'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">temperature_factor</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-232', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-233" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-233', 'element', 'link-218');">element</a></tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.writeLine()=extern.scientific_python.IO.PDB.PDBFile-class.html#writeLine"><a title="extern.scientific_python.IO.PDB.PDBFile.writeLine" class="py-name" href="#" onclick="return doclink('link-234', 'writeLine', 'link-234');">writeLine</a></tt><tt class="py-op">(</tt><tt id="link-235" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-235', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-236', 'data', 'link-39');">data</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.nextResidue"></a><div id="PDBFile.nextResidue-def"><a name="L433"></a><tt class="py-lineno"> 433</tt> <a class="py-toggle" href="#" id="PDBFile.nextResidue-toggle" onclick="return toggle('PDBFile.nextResidue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#nextResidue">nextResidue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">terminus</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.nextResidue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.nextResidue-expanded"><a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line"><tt class="py-docstring">        Signal the beginning of a new residue, starting with the</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line"><tt class="py-docstring">        next call to L{writeAtom}.</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">        @param name: the residue name</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring">        @type name: C{str}</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: the residue number. If C{None}, the residues</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">                       will be numbered sequentially, starting from 1.</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: C{int} or C{NoneType}</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-docstring">        @param terminus: C{None}, "C", or "N". This information</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"><tt class="py-docstring">                         is passed to export filters that can use this</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"><tt class="py-docstring">                         information in order to use different atom or</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"><tt class="py-docstring">                         residue names in terminal residues.</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">        <tt id="link-237" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-237', 'name', 'link-216');">name</a></tt>  <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">upper</tt><tt class="py-op">(</tt><tt id="link-238" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-238', 'name', 'link-216');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">            <tt id="link-239" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-239', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt id="link-240" class="py-name" targets="Method gui.user_functions.spin.Spin.number()=gui.user_functions.spin.Spin-class.html#number,Variable info.Clore90.number=info.Clore90-class.html#number,Variable info.Delaglio95.number=info.Delaglio95-class.html#number,Variable info.LipariSzabo82a.number=info.LipariSzabo82a-class.html#number,Variable info.LipariSzabo82b.number=info.LipariSzabo82b-class.html#number,Variable info.Ref.number=info.Ref-class.html#number,Variable info.dAuvergneGooley03.number=info.dAuvergneGooley03-class.html#number,Variable info.dAuvergneGooley06.number=info.dAuvergneGooley06-class.html#number,Variable info.dAuvergneGooley07.number=info.dAuvergneGooley07-class.html#number,Variable info.dAuvergneGooley08a.number=info.dAuvergneGooley08a-class.html#number,Variable info.dAuvergneGooley08b.number=info.dAuvergneGooley08b-class.html#number,Method prompt.residue.Residue.number()=prompt.residue.Residue-class.html#number,Method prompt.spin.Spin.number()=prompt.spin.Spin-class.html#number"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-240', 'number', 'link-240');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt><tt class="py-op">.</tt><tt class="py-name">processResidue</tt><tt class="py-op">(</tt><tt id="link-241" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-241', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt id="link-242" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-242', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">                                                             <tt class="py-name">terminus</tt><tt class="py-op">)</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">het_flag</tt> <tt class="py-op">=</tt>  <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt id="link-243" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-243', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt id="link-244" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-244', 'amino_acids', 'link-28');">amino_acids</a></tt> <tt class="py-keyword">or</tt> <tt id="link-245" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-245', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt id="link-246" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-246', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt><tt class="py-op">)</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-247', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-248" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-248', 'name', 'link-216');">name</a></tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-249', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-250', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">10000</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-251', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-252" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-252', 'number', 'link-240');">number</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-253" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-253', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-254" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-254', 'number', 'link-240');">number</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-255', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-256" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-256', 'number', 'link-240');">number</a></tt> <tt class="py-op">%</tt> <tt class="py-number">10000</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-257', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt id="link-258" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-258', 'number', 'link-240');">number</a></tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">1000</tt><tt class="py-op">)</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-259', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-260" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-260', 'number', 'link-240');">number</a></tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-261', 'number', 'link-240');">number</a></tt> <tt class="py-op">%</tt> <tt class="py-number">10000</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-262', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-263" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-263', 'number', 'link-240');">number</a></tt><tt class="py-op">.</tt><tt class="py-name">insertion_code</tt> </tt>
</div><a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.nextChain"></a><div id="PDBFile.nextChain-def"><a name="L466"></a><tt class="py-lineno"> 466</tt> <a class="py-toggle" href="#" id="PDBFile.nextChain-toggle" onclick="return toggle('PDBFile.nextChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#nextChain">nextChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">segment_id</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.nextChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.nextChain-expanded"><a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line"><tt class="py-docstring">        Signal the beginning of a new chain.</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"><tt class="py-docstring">        @param chain_id: a chain identifier. If C{None}, consecutive letters</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"><tt class="py-docstring">                         from the alphabet are used.</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain_id: C{str} or C{NoneType}</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line"><tt class="py-docstring">        @param segment_id: a chain identifier</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"><tt class="py-docstring">        @type segment_id: C{str}</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">chain_id</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_number</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_number</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name" targets="Variable extern.scientific_python.IO.PDB.PDBFile._chain_ids=extern.scientific_python.IO.PDB.PDBFile-class.html#_chain_ids"><a title="extern.scientific_python.IO.PDB.PDBFile._chain_ids" class="py-name" href="#" onclick="return doclink('link-264', '_chain_ids', 'link-264');">_chain_ids</a></tt><tt class="py-op">)</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">            <tt class="py-name">chain_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile._chain_ids" class="py-name" href="#" onclick="return doclink('link-265', '_chain_ids', 'link-264');">_chain_ids</a></tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_number</tt><tt class="py-op">]</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">            <tt class="py-name">chain_id</tt><tt class="py-op">,</tt> <tt class="py-name">segment_id</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">                      <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt><tt class="py-op">.</tt><tt class="py-name">processChain</tt><tt class="py-op">(</tt><tt class="py-name">chain_id</tt><tt class="py-op">,</tt> <tt class="py-name">segment_id</tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-266', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">chain_id</tt><tt class="py-op">+</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-267', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">segment_id</tt><tt class="py-op">+</tt><tt class="py-string">'    '</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-268', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">    <tt id="link-269" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile._chain_ids" class="py-name" href="#" onclick="return doclink('link-269', '_chain_ids', 'link-264');">_chain_ids</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'ABCDEFGHIJKLMNOPQRSTUVWXYZ'</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.terminateChain"></a><div id="PDBFile.terminateChain-def"><a name="L488"></a><tt class="py-lineno"> 488</tt> <a class="py-toggle" href="#" id="PDBFile.terminateChain-toggle" onclick="return toggle('PDBFile.terminateChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#terminateChain">terminateChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.terminateChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.terminateChain-expanded"><a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"><tt class="py-docstring">        Signal the end of a chain.</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">export_filter</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.terminateChain()=extern.scientific_python.IO.PDB.PDBFile-class.html#terminateChain"><a title="extern.scientific_python.IO.PDB.PDBFile.terminateChain" class="py-name" href="#" onclick="return doclink('link-270', 'terminateChain', 'link-270');">terminateChain</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-271', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-272', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">100000</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.writeLine" class="py-name" href="#" onclick="return doclink('link-273', 'writeLine', 'link-234');">writeLine</a></tt><tt class="py-op">(</tt><tt class="py-string">'TER'</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-274', 'data', 'link-39');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-275', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-276', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
</div><a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">         </tt>
<a name="PDBFile.close"></a><div id="PDBFile.close-def"><a name="L499"></a><tt class="py-lineno"> 499</tt> <a class="py-toggle" href="#" id="PDBFile.close-toggle" onclick="return toggle('PDBFile.close');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#close">close</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.close-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.close-expanded"><a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"><tt class="py-docstring">        Close the file. This method B{must} be called for write mode</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring">        because otherwise the file will be incomplete.</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">open</tt><tt class="py-op">:</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">output</tt><tt class="py-op">:</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-277', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="bmrblib.nmr_star_dict.NMR_STAR.write
bmrblib.pystarlib.File.File.write
extern.scientific_python.IO.TextFile.TextFile.write
generic_fns.bmrb.write
generic_fns.grace.write
generic_fns.pcs.write
generic_fns.rdc.write
generic_fns.relax_data.write
generic_fns.results.write
generic_fns.sequence.write
generic_fns.value.write
gui.controller.LogCtrl.write
gui.controller.Redirect_text.write
gui.user_functions.grace.Grace.write
gui.user_functions.results.Results.write
gui.user_functions.sequence.Sequence.write
multi.processor_io.IO_filter.write
multi.processor_io.PrependStringIO.write
prompt.bmrb.BMRB.write
prompt.grace.Grace.write
prompt.pcs.PCS.write
prompt.rdc.RDC.write
prompt.relax_data.Relax_data.write
prompt.results.Results.write
prompt.sequence.Sequence.write
prompt.value.Value.write
relax_io.DummyFileObject.write
relax_io.SplitIO.write" class="py-name" href="#" onclick="return doclink('link-278', 'write', 'link-202');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">'END\n'</tt><tt class="py-op">)</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-279', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-280" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.close()=extern.scientific_python.IO.PDB.PDBFile-class.html#close,Method extern.scientific_python.IO.TextFile.TextFile.close()=extern.scientific_python.IO.TextFile.TextFile-class.html#close,Method relax_io.DummyFileObject.close()=relax_io.DummyFileObject-class.html#close"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-280', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">open</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line"> </tt>
<a name="PDBFile.__del__"></a><div id="PDBFile.__del__-def"><a name="L510"></a><tt class="py-lineno"> 510</tt> <a class="py-toggle" href="#" id="PDBFile.__del__-toggle" onclick="return toggle('PDBFile.__del__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PDBFile-class.html#__del__">__del__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PDBFile.__del__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PDBFile.__del__-expanded"><a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-281', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line"> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"><tt class="py-comment"># High-level object representation of PDB file contents.</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-comment"># Representation of objects.</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="Atom"></a><div id="Atom-def"><a name="L521"></a><tt class="py-lineno"> 521</tt> <a class="py-toggle" href="#" id="Atom-toggle" onclick="return toggle('Atom');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html">Atom</a><tt class="py-op">:</tt> </tt>
</div><div id="Atom-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Atom-expanded"><a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line"><tt class="py-docstring">    Atom in a PDB structure</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">     </tt>
<a name="Atom.__init__"></a><div id="Atom.__init__-def"><a name="L527"></a><tt class="py-lineno"> 527</tt> <a class="py-toggle" href="#" id="Atom.__init__-toggle" onclick="return toggle('Atom.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">position</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">properties</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.__init__-expanded"><a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line"><tt class="py-docstring">        @param name: the atom name</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line"><tt class="py-docstring">        @type name: C{str}</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"><tt class="py-docstring">        @param position: the atom position</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line"><tt class="py-docstring">        @type position: L{extern.scientific_python.Geometry.Vector}</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line"><tt class="py-docstring">        @param properties: any other atom properties as keyword parameters.</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line"><tt class="py-docstring">                           These properties are stored in the atom object</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"><tt class="py-docstring">                           and can be accessed by indexing, as for</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line"><tt class="py-docstring">                           dictionaries.</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">position</tt> <tt class="py-op">=</tt> <tt class="py-name">position</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt> <tt class="py-op">=</tt> <tt class="py-name">properties</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'element'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">''</tt><tt class="py-op">:</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-282" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-282', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">or</tt> <tt id="link-283" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-283', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">digits</tt><tt class="py-op">:</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-284" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-284', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-285" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-285', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">digits</tt><tt class="py-op">:</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-286" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-286', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-287" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-287', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-288', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-289', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-290', 'name', 'link-216');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"> </tt>
<a name="Atom.__getitem__"></a><div id="Atom.__getitem__-def"><a name="L550"></a><tt class="py-lineno"> 550</tt> <a class="py-toggle" href="#" id="Atom.__getitem__-toggle" onclick="return toggle('Atom.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.__getitem__-expanded"><a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"><tt class="py-docstring">        @param item: the name of a property, including "name" or "position"</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"><tt class="py-docstring">        @type item: C{str}</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the property value</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">item</tt> <tt class="py-op">==</tt> <tt class="py-string">'name'</tt><tt class="py-op">:</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-291', 'name', 'link-216');">name</a></tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">item</tt> <tt class="py-op">==</tt> <tt class="py-string">'position'</tt><tt class="py-op">:</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">position</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Undefined atom property: "</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">item</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"> </tt>
<a name="Atom.__setitem__"></a><div id="Atom.__setitem__-def"><a name="L566"></a><tt class="py-lineno"> 566</tt> <a class="py-toggle" href="#" id="Atom.__setitem__-toggle" onclick="return toggle('Atom.__setitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#__setitem__">__setitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.__setitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.__setitem__-expanded"><a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"><tt class="py-docstring">        @param item: the name of an existing or to be defined property</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"><tt class="py-docstring">        @type item: C{str}</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"><tt class="py-docstring">        @param value: the new value for the property</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-292" class="py-name" targets="Module generic_fns.value=generic_fns.value-module.html,Module gui.user_functions.value=gui.user_functions.value-module.html,Module prompt.value=prompt.value-module.html"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-292', 'value', 'link-292');">value</a></tt> </tt>
</div><a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"> </tt>
<a name="Atom.__str__"></a><div id="Atom.__str__-def"><a name="L574"></a><tt class="py-lineno"> 574</tt> <a class="py-toggle" href="#" id="Atom.__str__-toggle" onclick="return toggle('Atom.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.__str__-expanded"><a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">+</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-293', 'name', 'link-216');">name</a></tt> <tt class="py-op">+</tt> \ </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">               <tt class="py-string">' at '</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">position</tt><tt class="py-op">)</tt> </tt>
</div><a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">    <tt id="link-294" class="py-name" targets="Method bmrblib.pystarlib.Utils.Lister.__repr__()=bmrblib.pystarlib.Utils.Lister-class.html#__repr__,Method data.Relax_data_store.__repr__()=data.Relax_data_store-class.html#__repr__,Method data.align_tensor.AlignTensorList.__repr__()=data.align_tensor.AlignTensorList-class.html#__repr__,Method data.data_classes.Element.__repr__()=data.data_classes.Element-class.html#__repr__,Method data.mol_res_spin.MoleculeContainer.__repr__()=data.mol_res_spin.MoleculeContainer-class.html#__repr__,Method data.mol_res_spin.MoleculeList.__repr__()=data.mol_res_spin.MoleculeList-class.html#__repr__,Method data.mol_res_spin.ResidueContainer.__repr__()=data.mol_res_spin.ResidueContainer-class.html#__repr__,Method data.mol_res_spin.ResidueList.__repr__()=data.mol_res_spin.ResidueList-class.html#__repr__,Method data.mol_res_spin.SpinContainer.__repr__()=data.mol_res_spin.SpinContainer-class.html#__repr__,Method data.mol_res_spin.SpinList.__repr__()=data.mol_res_spin.SpinList-class.html#__repr__,Method data.pipe_container.PipeContainer.__repr__()=data.pipe_container.PipeContainer-class.html#__repr__,Method extern.scientific_python.IO.PDB.Structure.__repr__()=extern.scientific_python.IO.PDB.Structure-class.html#__repr__,Method generic_fns.structure.api_base.ModelContainer.__repr__()=generic_fns.structure.api_base.ModelContainer-class.html#__repr__,Method generic_fns.structure.api_base.ModelList.__repr__()=generic_fns.structure.api_base.ModelList-class.html#__repr__,Method generic_fns.structure.api_base.MolList.__repr__()=generic_fns.structure.api_base.MolList-class.html#__repr__,Method gui.relax_prompt._Exit.__repr__()=gui.relax_prompt._Exit-class.html#__repr__,Method prompt.command.Lh.__repr__()=prompt.command.Lh-class.html#__repr__,Method prompt.command.Ll.__repr__()=prompt.command.Ll-class.html#__repr__,Method prompt.command.Ls.__repr__()=prompt.command.Ls-class.html#__repr__,Method prompt.gpl.GPL.__repr__()=prompt.gpl.GPL-class.html#__repr__,Method prompt.help._Helper.__repr__()=prompt.help._Helper-class.html#__repr__,Method prompt.help._Helper_python.__repr__()=prompt.help._Helper_python-class.html#__repr__,Method prompt.interpreter._Exit.__repr__()=prompt.interpreter._Exit-class.html#__repr__"><a title="bmrblib.pystarlib.Utils.Lister.__repr__
data.Relax_data_store.__repr__
data.align_tensor.AlignTensorList.__repr__
data.data_classes.Element.__repr__
data.mol_res_spin.MoleculeContainer.__repr__
data.mol_res_spin.MoleculeList.__repr__
data.mol_res_spin.ResidueContainer.__repr__
data.mol_res_spin.ResidueList.__repr__
data.mol_res_spin.SpinContainer.__repr__
data.mol_res_spin.SpinList.__repr__
data.pipe_container.PipeContainer.__repr__
extern.scientific_python.IO.PDB.Structure.__repr__
generic_fns.structure.api_base.ModelContainer.__repr__
generic_fns.structure.api_base.ModelList.__repr__
generic_fns.structure.api_base.MolList.__repr__
gui.relax_prompt._Exit.__repr__
prompt.command.Lh.__repr__
prompt.command.Ll.__repr__
prompt.command.Ls.__repr__
prompt.gpl.GPL.__repr__
prompt.help._Helper.__repr__
prompt.help._Helper_python.__repr__
prompt.interpreter._Exit.__repr__" class="py-name" href="#" onclick="return doclink('link-294', '__repr__', 'link-294');">__repr__</a></tt> <tt class="py-op">=</tt> <tt id="link-295" class="py-name" targets="Method extern.scientific_python.IO.FortranFormat.FortranLine.__str__()=extern.scientific_python.IO.FortranFormat.FortranLine-class.html#__str__,Method extern.scientific_python.IO.PDB.Atom.__str__()=extern.scientific_python.IO.PDB.Atom-class.html#__str__,Method extern.scientific_python.IO.PDB.Group.__str__()=extern.scientific_python.IO.PDB.Group-class.html#__str__,Method extern.scientific_python.IO.PDB.ResidueNumber.__str__()=extern.scientific_python.IO.PDB.ResidueNumber-class.html#__str__,Method generic_fns.spectrum.Bruker_import.__str__()=generic_fns.spectrum.Bruker_import-class.html#__str__,Method minfx.errors.BaseError.__str__()=minfx.errors.BaseError-class.html#__str__,Method multi.processor.Capturing_exception.__str__()=multi.processor.Capturing_exception-class.html#__str__,Method relax_errors.BaseError.__str__()=relax_errors.BaseError-class.html#__str__,Method relax_errors.RelaxFault.__str__()=relax_errors.RelaxFault-class.html#__str__,Method relax_warnings.BaseWarning.__str__()=relax_warnings.BaseWarning-class.html#__str__"><a title="extern.scientific_python.IO.FortranFormat.FortranLine.__str__
extern.scientific_python.IO.PDB.Atom.__str__
extern.scientific_python.IO.PDB.Group.__str__
extern.scientific_python.IO.PDB.ResidueNumber.__str__
generic_fns.spectrum.Bruker_import.__str__
minfx.errors.BaseError.__str__
multi.processor.Capturing_exception.__str__
relax_errors.BaseError.__str__
relax_errors.RelaxFault.__str__
relax_warnings.BaseWarning.__str__" class="py-name" href="#" onclick="return doclink('link-295', '__str__', 'link-295');">__str__</a></tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"> </tt>
<a name="Atom.type"></a><div id="Atom.type-def"><a name="L579"></a><tt class="py-lineno"> 579</tt> <a class="py-toggle" href="#" id="Atom.type-toggle" onclick="return toggle('Atom.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.type-expanded"><a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the six-letter record type, ATOM or HETATM</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{str}</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'ATOM  '</tt> </tt>
</div><a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line"> </tt>
<a name="Atom.writeToFile"></a><div id="Atom.writeToFile-def"><a name="L586"></a><tt class="py-lineno"> 586</tt> <a class="py-toggle" href="#" id="Atom.writeToFile-toggle" onclick="return toggle('Atom.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Atom-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Atom.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Atom.writeToFile-expanded"><a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line"><tt class="py-docstring">        Write an atom record to a file</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">        @param file: a PDB file object or a filename</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring">        @type file: L{PDBFile} or C{str}</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt id="link-296" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-296', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-297" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-297', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-298" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-298', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-299" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-299', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">            <tt id="link-300" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-300', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-301" class="py-name" targets="Class extern.scientific_python.IO.PDB.PDBFile=extern.scientific_python.IO.PDB.PDBFile-class.html"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-301', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-302" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-302', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">            <tt id="link-303" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-303', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">        <tt id="link-304" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-304', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-305" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.writeAtom()=extern.scientific_python.IO.PDB.PDBFile-class.html#writeAtom"><a title="extern.scientific_python.IO.PDB.PDBFile.writeAtom" class="py-name" href="#" onclick="return doclink('link-305', 'writeAtom', 'link-305');">writeAtom</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-306', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">position</tt><tt class="py-op">,</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">                       <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'occupancy'</tt><tt class="py-op">,</tt> <tt class="py-number">0.</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line">                       <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'temperature_factor'</tt><tt class="py-op">,</tt> <tt class="py-number">0.</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">                       <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">.</tt><tt class="py-name">get</tt><tt class="py-op">(</tt><tt class="py-string">'element'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-307" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-307', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">            <tt id="link-308" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-308', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-309', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"> </tt>
<a name="HetAtom"></a><div id="HetAtom-def"><a name="L605"></a><tt class="py-lineno"> 605</tt> <a class="py-toggle" href="#" id="HetAtom-toggle" onclick="return toggle('HetAtom');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.HetAtom-class.html">HetAtom</a><tt class="py-op">(</tt><tt class="py-base-class">Atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HetAtom-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="HetAtom-expanded"><a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"><tt class="py-docstring">    HetAtom in a PDB structure</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"><tt class="py-docstring">    A subclass of Atom, which differs only in the return value</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"><tt class="py-docstring">    of the method type().</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"> </tt>
<a name="HetAtom.type"></a><div id="HetAtom.type-def"><a name="L614"></a><tt class="py-lineno"> 614</tt> <a class="py-toggle" href="#" id="HetAtom.type-toggle" onclick="return toggle('HetAtom.type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.HetAtom-class.html#type">type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="HetAtom.type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="HetAtom.type-expanded"><a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'HETATM'</tt> </tt>
</div></div><a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">     </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line"> </tt>
<a name="Group"></a><div id="Group-def"><a name="L618"></a><tt class="py-lineno"> 618</tt> <a class="py-toggle" href="#" id="Group-toggle" onclick="return toggle('Group');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html">Group</a><tt class="py-op">:</tt> </tt>
</div><div id="Group-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Group-expanded"><a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line"><tt class="py-docstring">    Atom group (residue or molecule) in a PDB file</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line"><tt class="py-docstring">    This is an abstract base class. Instances can be created using</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line"><tt class="py-docstring">    one of the subclasses (L{Molecule}, L{AminoAcidResidue},</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line"><tt class="py-docstring">    L{NucleotideResidue}).</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"><tt class="py-docstring">    Group objects permit iteration over atoms with for-loops,</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line"><tt class="py-docstring">    as well as extraction of atoms by indexing with the</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line"><tt class="py-docstring">    atom name.</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"> </tt>
<a name="Group.__init__"></a><div id="Group.__init__-def"><a name="L632"></a><tt class="py-lineno"> 632</tt> <a class="py-toggle" href="#" id="Group.__init__-toggle" onclick="return toggle('Group.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">atoms</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.__init__-expanded"><a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line"><tt class="py-docstring">        @param name: the name of the group</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line"><tt class="py-docstring">        @type name: C{str}</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line"><tt class="py-docstring">        @param atoms: a list of atoms (or C{None} for no atoms)</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line"><tt class="py-docstring">        @type atoms: C{list} or C{NoneType}</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: the PDB residue number (or C{None})</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: C{int} or C{NoneType}</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-310', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-311" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-311', 'name', 'link-216');">name</a></tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-312', 'number', 'link-240');">number</a></tt> <tt class="py-op">=</tt> <tt id="link-313" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-313', 'number', 'link-240');">number</a></tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">atoms</tt><tt class="py-op">:</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt> <tt class="py-op">=</tt> <tt class="py-name">atoms</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atoms</tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">[</tt><tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-314', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">a</tt> </tt>
</div><a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line"> </tt>
<a name="Group.__len__"></a><div id="Group.__len__-def"><a name="L650"></a><tt class="py-lineno"> 650</tt> <a class="py-toggle" href="#" id="Group.__len__-toggle" onclick="return toggle('Group.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.__len__-expanded"><a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">)</tt> </tt>
</div><a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line"> </tt>
<a name="Group.__getitem__"></a><div id="Group.__getitem__-def"><a name="L653"></a><tt class="py-lineno"> 653</tt> <a class="py-toggle" href="#" id="Group.__getitem__-toggle" onclick="return toggle('Group.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.__getitem__-expanded"><a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line"><tt class="py-docstring">        @param item: an integer index or an atom name</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"><tt class="py-docstring">        @type item: C{int} or C{str}</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">item</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> </tt>
</div><a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"> </tt>
<a name="Group.__str__"></a><div id="Group.__str__-def"><a name="L663"></a><tt class="py-lineno"> 663</tt> <a class="py-toggle" href="#" id="Group.__str__-toggle" onclick="return toggle('Group.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.__str__-expanded"><a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">+</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-315', 'name', 'link-216');">name</a></tt> <tt class="py-op">+</tt> <tt class="py-string">':\n'</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">'  '</tt> <tt class="py-op">+</tt> <tt class="py-op">`</tt><tt class="py-name">atom</tt><tt class="py-op">`</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">s</tt> </tt>
</div><a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">    <tt id="link-316" class="py-name"><a title="bmrblib.pystarlib.Utils.Lister.__repr__
data.Relax_data_store.__repr__
data.align_tensor.AlignTensorList.__repr__
data.data_classes.Element.__repr__
data.mol_res_spin.MoleculeContainer.__repr__
data.mol_res_spin.MoleculeList.__repr__
data.mol_res_spin.ResidueContainer.__repr__
data.mol_res_spin.ResidueList.__repr__
data.mol_res_spin.SpinContainer.__repr__
data.mol_res_spin.SpinList.__repr__
data.pipe_container.PipeContainer.__repr__
extern.scientific_python.IO.PDB.Structure.__repr__
generic_fns.structure.api_base.ModelContainer.__repr__
generic_fns.structure.api_base.ModelList.__repr__
generic_fns.structure.api_base.MolList.__repr__
gui.relax_prompt._Exit.__repr__
prompt.command.Lh.__repr__
prompt.command.Ll.__repr__
prompt.command.Ls.__repr__
prompt.gpl.GPL.__repr__
prompt.help._Helper.__repr__
prompt.help._Helper_python.__repr__
prompt.interpreter._Exit.__repr__" class="py-name" href="#" onclick="return doclink('link-316', '__repr__', 'link-294');">__repr__</a></tt> <tt class="py-op">=</tt> <tt id="link-317" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine.__str__
extern.scientific_python.IO.PDB.Atom.__str__
extern.scientific_python.IO.PDB.Group.__str__
extern.scientific_python.IO.PDB.ResidueNumber.__str__
generic_fns.spectrum.Bruker_import.__str__
minfx.errors.BaseError.__str__
multi.processor.Capturing_exception.__str__
relax_errors.BaseError.__str__
relax_errors.RelaxFault.__str__
relax_warnings.BaseWarning.__str__" class="py-name" href="#" onclick="return doclink('link-317', '__str__', 'link-295');">__str__</a></tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"> </tt>
<a name="Group.isCompatible"></a><div id="Group.isCompatible-def"><a name="L670"></a><tt class="py-lineno"> 670</tt> <a class="py-toggle" href="#" id="Group.isCompatible-toggle" onclick="return toggle('Group.isCompatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#isCompatible">isCompatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.isCompatible-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.isCompatible-expanded"><a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-318', 'name', 'link-216');">name</a></tt> \ </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">               <tt class="py-keyword">and</tt> <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-319', 'number', 'link-240');">number</a></tt> </tt>
</div><a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line"> </tt>
<a name="Group.addAtom"></a><div id="Group.addAtom-def"><a name="L674"></a><tt class="py-lineno"> 674</tt> <a class="py-toggle" href="#" id="Group.addAtom-toggle" onclick="return toggle('Group.addAtom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#addAtom">addAtom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.addAtom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.addAtom-expanded"><a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line"><tt class="py-docstring">        Add an atom to the group</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line"><tt class="py-docstring">        </tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line"><tt class="py-docstring">        @param atom: the atom</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line"><tt class="py-docstring">        @type atom: L{Atom}</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-320', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-321', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line"> </tt>
<a name="Group.deleteAtom"></a><div id="Group.deleteAtom-def"><a name="L685"></a><tt class="py-lineno"> 685</tt> <a class="py-toggle" href="#" id="Group.deleteAtom-toggle" onclick="return toggle('Group.deleteAtom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#deleteAtom">deleteAtom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.deleteAtom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.deleteAtom-expanded"><a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line"><tt class="py-docstring">        Remove an atom from the group</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line"><tt class="py-docstring">        @param atom: the atom to be removed</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line"><tt class="py-docstring">        @type atom: L{Atom}</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"><tt class="py-docstring">        @raises KeyError: if the atom is not part of the group</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">[</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-322', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"> </tt>
<a name="Group.deleteHydrogens"></a><div id="Group.deleteHydrogens-def"><a name="L697"></a><tt class="py-lineno"> 697</tt> <a class="py-toggle" href="#" id="Group.deleteHydrogens-toggle" onclick="return toggle('Group.deleteHydrogens');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#deleteHydrogens">deleteHydrogens</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.deleteHydrogens-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.deleteHydrogens-expanded"><a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line"><tt class="py-docstring">        Remove all hydrogen atoms of the group</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">        <tt id="link-323" class="py-name" targets="Function generic_fns.align_tensor.delete()=generic_fns.align_tensor-module.html#delete,Function generic_fns.diffusion_tensor.delete()=generic_fns.diffusion_tensor-module.html#delete,Function generic_fns.pipes.delete()=generic_fns.pipes-module.html#delete,Function generic_fns.relax_data.delete()=generic_fns.relax_data-module.html#delete,Function generic_fns.spectrum.delete()=generic_fns.spectrum-module.html#delete,Method generic_fns.structure.api_base.Base_struct_API.delete()=generic_fns.structure.api_base.Base_struct_API-class.html#delete,Method generic_fns.structure.internal.Internal.delete()=generic_fns.structure.internal.Internal-class.html#delete,Function generic_fns.structure.main.delete()=generic_fns.structure.main-module.html#delete,Method gui.analyses.auto_model_free.Auto_model_free.delete()=gui.analyses.auto_model_free.Auto_model_free-class.html#delete,Method gui.analyses.auto_noe.Auto_noe.delete()=gui.analyses.auto_noe.Auto_noe-class.html#delete,Method gui.analyses.auto_rx_base.Auto_rx.delete()=gui.analyses.auto_rx_base.Auto_rx-class.html#delete,Method gui.components.relax_data.Relax_data_list.delete()=gui.components.relax_data.Relax_data_list-class.html#delete,Method gui.components.spectrum.Spectra_list.delete()=gui.components.spectrum.Spectra_list-class.html#delete,Method gui.user_functions.molecule.Molecule.delete()=gui.user_functions.molecule.Molecule-class.html#delete,Method gui.user_functions.pipe.Pipe.delete()=gui.user_functions.pipe.Pipe-class.html#delete,Method gui.user_functions.relax_data.Relax_data.delete()=gui.user_functions.relax_data.Relax_data-class.html#delete,Method gui.user_functions.residue.Residue.delete()=gui.user_functions.residue.Residue-class.html#delete,Method gui.user_functions.spectrum.Spectrum.delete()=gui.user_functions.spectrum.Spectrum-class.html#delete,Method gui.user_functions.spin.Spin.delete()=gui.user_functions.spin.Spin-class.html#delete,Method gui.user_functions.structure.Structure.delete()=gui.user_functions.structure.Structure-class.html#delete,Method prompt.align_tensor.Align_tensor.delete()=prompt.align_tensor.Align_tensor-class.html#delete,Method prompt.diffusion_tensor.Diffusion_tensor.delete()=prompt.diffusion_tensor.Diffusion_tensor-class.html#delete,Method prompt.model_free.Model_free.delete()=prompt.model_free.Model_free-class.html#delete,Method prompt.molecule.Molecule.delete()=prompt.molecule.Molecule-class.html#delete,Method prompt.pcs.PCS.delete()=prompt.pcs.PCS-class.html#delete,Method prompt.pipe.Pipe.delete()=prompt.pipe.Pipe-class.html#delete,Method prompt.rdc.RDC.delete()=prompt.rdc.RDC-class.html#delete,Method prompt.relax_data.Relax_data.delete()=prompt.relax_data.Relax_data-class.html#delete,Method prompt.residue.Residue.delete()=prompt.residue.Residue-class.html#delete,Method prompt.spectrum.Spectrum.delete()=prompt.spectrum.Spectrum-class.html#delete,Method prompt.spin.Spin.delete()=prompt.spin.Spin-class.html#delete,Method prompt.structure.Structure.delete()=prompt.structure.Structure-class.html#delete,Function relax_io.delete()=relax_io-module.html#delete"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-323', 'delete', 'link-323');">delete</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-324', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'H'</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-325', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">digits</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">                                    <tt class="py-keyword">and</tt> <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-326', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'H'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">                <tt id="link-327" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-327', 'delete', 'link-323');">delete</a></tt><tt class="py-op">.</tt><tt id="link-328" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-328', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">a</tt><tt class="py-op">)</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt id="link-329" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-329', 'delete', 'link-323');">delete</a></tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name" targets="Method extern.scientific_python.IO.PDB.Group.deleteAtom()=extern.scientific_python.IO.PDB.Group-class.html#deleteAtom"><a title="extern.scientific_python.IO.PDB.Group.deleteAtom" class="py-name" href="#" onclick="return doclink('link-330', 'deleteAtom', 'link-330');">deleteAtom</a></tt><tt class="py-op">(</tt><tt class="py-name">a</tt><tt class="py-op">)</tt> </tt>
</div><a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line"> </tt>
<a name="Group.changeName"></a><div id="Group.changeName-def"><a name="L709"></a><tt class="py-lineno"> 709</tt> <a class="py-toggle" href="#" id="Group.changeName-toggle" onclick="return toggle('Group.changeName');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#changeName">changeName</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.changeName-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.changeName-expanded"><a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line"><tt class="py-docstring">        Set the PDB residue name</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"><tt class="py-docstring">        @param name: the new name</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line"><tt class="py-docstring">        @type name: C{str}</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-331', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-332" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-332', 'name', 'link-216');">name</a></tt> </tt>
</div><a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line"> </tt>
<a name="Group.writeToFile"></a><div id="Group.writeToFile-def"><a name="L718"></a><tt class="py-lineno"> 718</tt> <a class="py-toggle" href="#" id="Group.writeToFile-toggle" onclick="return toggle('Group.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Group-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Group.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Group.writeToFile-expanded"><a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line"><tt class="py-docstring">        Write the group to a file</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line"><tt class="py-docstring">        @param file: a PDBFile object or a file name</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line"><tt class="py-docstring">        @type file: L{PDBFile} or C{str}</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">        <tt id="link-333" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-333', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-334" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-334', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-335" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-335', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-336" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-336', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">            <tt id="link-337" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-337', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-338" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-338', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-339', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">            <tt id="link-340" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-340', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">        <tt id="link-341" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-341', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-342" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.nextResidue()=extern.scientific_python.IO.PDB.PDBFile-class.html#nextResidue"><a title="extern.scientific_python.IO.PDB.PDBFile.nextResidue" class="py-name" href="#" onclick="return doclink('link-342', 'nextResidue', 'link-342');">nextResidue</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-343', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-344', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">            <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name" targets="Method extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile()=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#writeToFile,Method extern.scientific_python.IO.PDB.Atom.writeToFile()=extern.scientific_python.IO.PDB.Atom-class.html#writeToFile,Method extern.scientific_python.IO.PDB.Chain.writeToFile()=extern.scientific_python.IO.PDB.Chain-class.html#writeToFile,Method extern.scientific_python.IO.PDB.Group.writeToFile()=extern.scientific_python.IO.PDB.Group-class.html#writeToFile,Method extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#writeToFile,Method extern.scientific_python.IO.PDB.Structure.writeToFile()=extern.scientific_python.IO.PDB.Structure-class.html#writeToFile"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile
extern.scientific_python.IO.PDB.Atom.writeToFile
extern.scientific_python.IO.PDB.Chain.writeToFile
extern.scientific_python.IO.PDB.Group.writeToFile
extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile
extern.scientific_python.IO.PDB.Structure.writeToFile" class="py-name" href="#" onclick="return doclink('link-345', 'writeToFile', 'link-345');">writeToFile</a></tt><tt class="py-op">(</tt><tt id="link-346" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-346', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-347" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-347', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">            <tt id="link-348" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-348', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-349', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line"> </tt>
<a name="Molecule"></a><div id="Molecule-def"><a name="L735"></a><tt class="py-lineno"> 735</tt> <a class="py-toggle" href="#" id="Molecule-toggle" onclick="return toggle('Molecule');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Molecule-class.html">Molecule</a><tt class="py-op">(</tt><tt class="py-base-class">Group</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Molecule-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Molecule-expanded"><a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"><tt class="py-docstring">    Molecule in a PDB file</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring">    B{Note:} In PDB files, non-chain molecules are treated as residues,</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">    there is no separate molecule definition. This module defines</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring">    every residue as a molecule that is not an amino acid residue or a</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">    nucleotide residue.</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"> </tt>
<a name="Residue"></a><div id="Residue-def"><a name="L748"></a><tt class="py-lineno"> 748</tt> <a class="py-toggle" href="#" id="Residue-toggle" onclick="return toggle('Residue');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Residue-class.html">Residue</a><tt class="py-op">(</tt><tt class="py-base-class">Group</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Residue-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Residue-expanded"><a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line"> </tt>
<a name="AminoAcidResidue"></a><div id="AminoAcidResidue-def"><a name="L752"></a><tt class="py-lineno"> 752</tt> <a class="py-toggle" href="#" id="AminoAcidResidue-toggle" onclick="return toggle('AminoAcidResidue');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.AminoAcidResidue-class.html">AminoAcidResidue</a><tt class="py-op">(</tt><tt class="py-base-class">Residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AminoAcidResidue-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="AminoAcidResidue-expanded"><a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"><tt class="py-docstring">    Amino acid residue in a PDB file</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line"> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">    <tt id="link-350" class="py-name" targets="Variable extern.scientific_python.IO.PDB.AminoAcidResidue.is_amino_acid=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#is_amino_acid"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.is_amino_acid" class="py-name" href="#" onclick="return doclink('link-350', 'is_amino_acid', 'link-350');">is_amino_acid</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line"> </tt>
<a name="AminoAcidResidue.isCTerminus"></a><div id="AminoAcidResidue.isCTerminus-def"><a name="L760"></a><tt class="py-lineno"> 760</tt> <a class="py-toggle" href="#" id="AminoAcidResidue.isCTerminus-toggle" onclick="return toggle('AminoAcidResidue.isCTerminus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#isCTerminus">isCTerminus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AminoAcidResidue.isCTerminus-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AminoAcidResidue.isCTerminus-expanded"><a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue is in C-terminal configuration,</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line"><tt class="py-docstring">        i.e. if it has a second oxygen bound to the carbon atom of</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line"><tt class="py-docstring">        the peptide group. C{False} otherwise.</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-351', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'NME'</tt> \ </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">               <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'OXT'</tt><tt class="py-op">)</tt> \ </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">               <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'OT2'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"> </tt>
<a name="AminoAcidResidue.isNTerminus"></a><div id="AminoAcidResidue.isNTerminus-def"><a name="L771"></a><tt class="py-lineno"> 771</tt> <a class="py-toggle" href="#" id="AminoAcidResidue.isNTerminus-toggle" onclick="return toggle('AminoAcidResidue.isNTerminus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#isNTerminus">isNTerminus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AminoAcidResidue.isNTerminus-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AminoAcidResidue.isNTerminus-expanded"><a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue is in N-terminal configuration,</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"><tt class="py-docstring">        i.e. if it contains more than one hydrogen bound to be</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"><tt class="py-docstring">        nitrogen atom of the peptide group. C{False} otherwise.</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'1HT'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'2HT'</tt><tt class="py-op">)</tt> \ </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">               <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'3HT'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line"> </tt>
<a name="AminoAcidResidue.addAtom"></a><div id="AminoAcidResidue.addAtom-def"><a name="L781"></a><tt class="py-lineno"> 781</tt> <a class="py-toggle" href="#" id="AminoAcidResidue.addAtom-toggle" onclick="return toggle('AminoAcidResidue.addAtom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#addAtom">addAtom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AminoAcidResidue.addAtom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AminoAcidResidue.addAtom-expanded"><a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt id="link-352" class="py-name" targets="Class extern.scientific_python.IO.PDB.Residue=extern.scientific_python.IO.PDB.Residue-class.html,Class gui.spin_viewer.containers.Residue=gui.spin_viewer.containers.Residue-class.html,Class gui.user_functions.residue.Residue=gui.user_functions.residue.Residue-class.html,Class prompt.residue.Residue=prompt.residue.Residue-class.html"><a title="extern.scientific_python.IO.PDB.Residue
gui.spin_viewer.containers.Residue
gui.user_functions.residue.Residue
prompt.residue.Residue" class="py-name" href="#" onclick="return doclink('link-352', 'Residue', 'link-352');">Residue</a></tt><tt class="py-op">.</tt><tt id="link-353" class="py-name" targets="Method extern.scientific_python.IO.PDB.AminoAcidResidue.addAtom()=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#addAtom,Method extern.scientific_python.IO.PDB.Group.addAtom()=extern.scientific_python.IO.PDB.Group-class.html#addAtom,Method extern.scientific_python.IO.PDB.NucleotideResidue.addAtom()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#addAtom"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.addAtom
extern.scientific_python.IO.PDB.Group.addAtom
extern.scientific_python.IO.PDB.NucleotideResidue.addAtom" class="py-name" href="#" onclick="return doclink('link-353', 'addAtom', 'link-353');">addAtom</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-354', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CA'</tt><tt class="py-op">:</tt> <tt class="py-comment"># Make sure it's not a calcium</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">            <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt class="py-name">properties</tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'C'</tt> </tt>
</div><a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"> </tt>
<a name="AminoAcidResidue.writeToFile"></a><div id="AminoAcidResidue.writeToFile-def"><a name="L786"></a><tt class="py-lineno"> 786</tt> <a class="py-toggle" href="#" id="AminoAcidResidue.writeToFile-toggle" onclick="return toggle('AminoAcidResidue.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="AminoAcidResidue.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="AminoAcidResidue.writeToFile-expanded"><a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt id="link-355" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-355', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-356" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-356', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-357" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-357', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-358" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-358', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">            <tt id="link-359" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-359', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-360" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-360', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-361" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-361', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">            <tt id="link-362" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-362', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">        <tt class="py-name">terminus</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name" targets="Method extern.scientific_python.IO.PDB.AminoAcidResidue.isCTerminus()=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#isCTerminus"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.isCTerminus" class="py-name" href="#" onclick="return doclink('link-363', 'isCTerminus', 'link-363');">isCTerminus</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-name">terminus</tt> <tt class="py-op">=</tt> <tt class="py-string">'C'</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name" targets="Method extern.scientific_python.IO.PDB.AminoAcidResidue.isNTerminus()=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html#isNTerminus"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.isNTerminus" class="py-name" href="#" onclick="return doclink('link-364', 'isNTerminus', 'link-364');">isNTerminus</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-name">terminus</tt> <tt class="py-op">=</tt> <tt class="py-string">'N'</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">        <tt id="link-365" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-365', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.nextResidue" class="py-name" href="#" onclick="return doclink('link-366', 'nextResidue', 'link-342');">nextResidue</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-367" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-367', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-368', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">terminus</tt><tt class="py-op">)</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">            <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile
extern.scientific_python.IO.PDB.Atom.writeToFile
extern.scientific_python.IO.PDB.Chain.writeToFile
extern.scientific_python.IO.PDB.Group.writeToFile
extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile
extern.scientific_python.IO.PDB.Structure.writeToFile" class="py-name" href="#" onclick="return doclink('link-369', 'writeToFile', 'link-345');">writeToFile</a></tt><tt class="py-op">(</tt><tt id="link-370" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-370', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-371" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-371', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">            <tt id="link-372" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-372', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-373', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line"> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue"></a><div id="NucleotideResidue-def"><a name="L801"></a><tt class="py-lineno"> 801</tt> <a class="py-toggle" href="#" id="NucleotideResidue-toggle" onclick="return toggle('NucleotideResidue');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html">NucleotideResidue</a><tt class="py-op">(</tt><tt class="py-base-class">Residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="NucleotideResidue-expanded"><a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line"> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring">    Nucleotide residue in a PDB file</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">    <tt id="link-374" class="py-name" targets="Variable extern.scientific_python.IO.PDB.NucleotideResidue.is_nucleotide=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#is_nucleotide"><a title="extern.scientific_python.IO.PDB.NucleotideResidue.is_nucleotide" class="py-name" href="#" onclick="return doclink('link-374', 'is_nucleotide', 'link-374');">is_nucleotide</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.__init__"></a><div id="NucleotideResidue.__init__-def"><a name="L809"></a><tt class="py-lineno"> 809</tt> <a class="py-toggle" href="#" id="NucleotideResidue.__init__-toggle" onclick="return toggle('NucleotideResidue.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">atoms</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.__init__-expanded"><a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pdbname</tt> <tt class="py-op">=</tt> <tt id="link-375" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-375', 'name', 'link-216');">name</a></tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">        <tt id="link-376" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-376', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">string</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-377', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt id="link-378" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-378', 'name', 'link-216');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-379" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-379', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'D'</tt> <tt class="py-keyword">and</tt> <tt id="link-380" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-380', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'R'</tt><tt class="py-op">:</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">            <tt id="link-381" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-381', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'D'</tt> <tt class="py-op">+</tt> <tt id="link-382" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-382', 'name', 'link-216');">name</a></tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">        <tt id="link-383" class="py-name"><a title="extern.scientific_python.IO.PDB.Residue
gui.spin_viewer.containers.Residue
gui.user_functions.residue.Residue
prompt.residue.Residue" class="py-name" href="#" onclick="return doclink('link-383', 'Residue', 'link-352');">Residue</a></tt><tt class="py-op">.</tt><tt id="link-384" class="py-name" targets="Method auto_analyses.dauvergne_protocol.dAuvergne_protocol.__init__()=auto_analyses.dauvergne_protocol.dAuvergne_protocol-class.html#__init__,Method auto_analyses.noe.NOE_calc.__init__()=auto_analyses.noe.NOE_calc-class.html#__init__,Method auto_analyses.relax_fit.Relax_fit.__init__()=auto_analyses.relax_fit.Relax_fit-class.html#__init__,Method auto_analyses.stereochem_analysis.Stereochem_analysis.__init__()=auto_analyses.stereochem_analysis.Stereochem_analysis-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropy.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropy-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropyExperiment.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropyExperiment-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy.ChemShiftAnisotropy.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy.ChemShiftAnisotropy-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropyExperiment_v3_1.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropyExperiment_v3_1-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropySoftware_v3_1.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropySoftware_v3_1-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropy_v3_1.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropy_v3_1-class.html#__init__,Method bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.ChemShiftAnisotropy_v3_1.__init__()=bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.ChemShiftAnisotropy_v3_1-class.html#__init__,Method bmrblib.assembly_supercategory.entity.Entity.__init__()=bmrblib.assembly_supercategory.entity.Entity-class.html#__init__,Method bmrblib.assembly_supercategory.entity.EntityCompIndex.__init__()=bmrblib.assembly_supercategory.entity.EntityCompIndex-class.html#__init__,Method bmrblib.assembly_supercategory.entity_v2_1.EntityCompIndex_v2_1.__init__()=bmrblib.assembly_supercategory.entity_v2_1.EntityCompIndex_v2_1-class.html#__init__,Method bmrblib.assembly_supercategory.entity_v2_1.Entity_v2_1.__init__()=bmrblib.assembly_supercategory.entity_v2_1.Entity_v2_1-class.html#__init__,Method bmrblib.assembly_supercategory.entity_v3_1.EntityCompIndex_v3_1.__init__()=bmrblib.assembly_supercategory.entity_v3_1.EntityCompIndex_v3_1-class.html#__init__,Method bmrblib.assembly_supercategory.entity_v3_1.Entity_v3_1.__init__()=bmrblib.assembly_supercategory.entity_v3_1.Entity_v3_1-class.html#__init__,Method bmrblib.base_classes.BaseSaveframe.__init__()=bmrblib.base_classes.BaseSaveframe-class.html#__init__,Method bmrblib.base_classes.MissingSaveframe.__init__()=bmrblib.base_classes.MissingSaveframe-class.html#__init__,Method bmrblib.base_classes.TagCategory.__init__()=bmrblib.base_classes.TagCategory-class.html#__init__,Method bmrblib.base_classes.TagCategoryFree.__init__()=bmrblib.base_classes.TagCategoryFree-class.html#__init__,Method bmrblib.base_classes.TagObject.__init__()=bmrblib.base_classes.TagObject-class.html#__init__,Method bmrblib.base_classes.TagTranslationTable.__init__()=bmrblib.base_classes.TagTranslationTable-class.html#__init__,Method bmrblib.citations.citations.Citations.__init__()=bmrblib.citations.citations.Citations-class.html#__init__,Method bmrblib.citations.citations.CitationsAuthor.__init__()=bmrblib.citations.citations.CitationsAuthor-class.html#__init__,Method bmrblib.citations.citations_v3_1.CitationsAuthor_v3_1.__init__()=bmrblib.citations.citations_v3_1.CitationsAuthor_v3_1-class.html#__init__,Method bmrblib.citations.citations_v3_1.Citations_v3_1.__init__()=bmrblib.citations.citations_v3_1.Citations_v3_1-class.html#__init__,Method bmrblib.experimental_details.experiment.Experiment.__init__()=bmrblib.experimental_details.experiment.Experiment-class.html#__init__,Method bmrblib.experimental_details.experiment.ExperimentList.__init__()=bmrblib.experimental_details.experiment.ExperimentList-class.html#__init__,Method bmrblib.experimental_details.experiment_v3_1.ExperimentList_v3_1.__init__()=bmrblib.experimental_details.experiment_v3_1.ExperimentList_v3_1-class.html#__init__,Method bmrblib.experimental_details.experiment_v3_1.Experiment_v3_1.__init__()=bmrblib.experimental_details.experiment_v3_1.Experiment_v3_1-class.html#__init__,Method bmrblib.experimental_details.method.Method.__init__()=bmrblib.experimental_details.method.Method-class.html#__init__,Method bmrblib.experimental_details.method.MethodCitation.__init__()=bmrblib.experimental_details.method.MethodCitation-class.html#__init__,Method bmrblib.experimental_details.method.MethodFile.__init__()=bmrblib.experimental_details.method.MethodFile-class.html#__init__,Method bmrblib.experimental_details.method.MethodParam.__init__()=bmrblib.experimental_details.method.MethodParam-class.html#__init__,Method bmrblib.experimental_details.method_v3_1.MethodCitation_v3_1.__init__()=bmrblib.experimental_details.method_v3_1.MethodCitation_v3_1-class.html#__init__,Method bmrblib.experimental_details.method_v3_1.MethodFile_v3_1.__init__()=bmrblib.experimental_details.method_v3_1.MethodFile_v3_1-class.html#__init__,Method bmrblib.experimental_details.method_v3_1.MethodParam_v3_1.__init__()=bmrblib.experimental_details.method_v3_1.MethodParam_v3_1-class.html#__init__,Method bmrblib.experimental_details.method_v3_1.Method_v3_1.__init__()=bmrblib.experimental_details.method_v3_1.Method_v3_1-class.html#__init__,Method bmrblib.experimental_details.nmr_spectrometer.NMRSpectrometer.__init__()=bmrblib.experimental_details.nmr_spectrometer.NMRSpectrometer-class.html#__init__,Method bmrblib.experimental_details.nmr_spectrometer_v3_1.NMRSpectrometer_v3_1.__init__()=bmrblib.experimental_details.nmr_spectrometer_v3_1.NMRSpectrometer_v3_1-class.html#__init__,Method bmrblib.experimental_details.sample_conditions.SampleConditionCitation.__init__()=bmrblib.experimental_details.sample_conditions.SampleConditionCitation-class.html#__init__,Method bmrblib.experimental_details.sample_conditions.SampleConditionList.__init__()=bmrblib.experimental_details.sample_conditions.SampleConditionList-class.html#__init__,Method bmrblib.experimental_details.sample_conditions.SampleConditionVariable.__init__()=bmrblib.experimental_details.sample_conditions.SampleConditionVariable-class.html#__init__,Method bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionList_v2_1.__init__()=bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionList_v2_1-class.html#__init__,Method bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionVariable_v2_1.__init__()=bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionVariable_v2_1-class.html#__init__,Method bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionCitation_v3_1.__init__()=bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionCitation_v3_1-class.html#__init__,Method bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionList_v3_1.__init__()=bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionList_v3_1-class.html#__init__,Method bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionVariable_v3_1.__init__()=bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionVariable_v3_1-class.html#__init__,Method bmrblib.experimental_details.software.Software.__init__()=bmrblib.experimental_details.software.Software-class.html#__init__,Method bmrblib.experimental_details.software.SoftwareCitation.__init__()=bmrblib.experimental_details.software.SoftwareCitation-class.html#__init__,Method bmrblib.experimental_details.software.Task.__init__()=bmrblib.experimental_details.software.Task-class.html#__init__,Method bmrblib.experimental_details.software.Vendor.__init__()=bmrblib.experimental_details.software.Vendor-class.html#__init__,Method bmrblib.experimental_details.software_v3_1.SoftwareCitation_v3_1.__init__()=bmrblib.experimental_details.software_v3_1.SoftwareCitation_v3_1-class.html#__init__,Method bmrblib.experimental_details.software_v3_1.Software_v3_1.__init__()=bmrblib.experimental_details.software_v3_1.Software_v3_1-class.html#__init__,Method bmrblib.experimental_details.software_v3_1.Task_v3_1.__init__()=bmrblib.experimental_details.software_v3_1.Task_v3_1-class.html#__init__,Method bmrblib.experimental_details.software_v3_1.Vendor_v3_1.__init__()=bmrblib.experimental_details.software_v3_1.Vendor_v3_1-class.html#__init__,Method bmrblib.kinetics.auto_relaxation.AutoRelaxation.__init__()=bmrblib.kinetics.auto_relaxation.AutoRelaxation-class.html#__init__,Method bmrblib.kinetics.auto_relaxation.AutoRelaxationExperiment.__init__()=bmrblib.kinetics.auto_relaxation.AutoRelaxationExperiment-class.html#__init__,Method bmrblib.kinetics.auto_relaxation.AutoRelaxationList.__init__()=bmrblib.kinetics.auto_relaxation.AutoRelaxationList-class.html#__init__,Method bmrblib.kinetics.auto_relaxation.AutoRelaxationSoftware.__init__()=bmrblib.kinetics.auto_relaxation.AutoRelaxationSoftware-class.html#__init__,Method bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationExperiment_v3_1.__init__()=bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationExperiment_v3_1-class.html#__init__,Method bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationList_v3_1.__init__()=bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationList_v3_1-class.html#__init__,Method bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSoftware_v3_1.__init__()=bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSoftware_v3_1-class.html#__init__,Method bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxation_v3_1.__init__()=bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxation_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOE.__init__()=bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOE-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEExperiment.__init__()=bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEExperiment-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEList.__init__()=bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEList-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOESoftware.__init__()=bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOESoftware-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEExperiment_v2_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEExperiment_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEList_v2_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEList_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESoftware_v2_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESoftware_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOE_v2_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOE_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEExperiment_v3_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEExperiment_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEList_v3_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEList_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOESoftware_v3_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOESoftware_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOE_v3_1.__init__()=bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOE_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Experiment.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Experiment-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1List.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1List-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Software.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Software-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation.T1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation.T1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Experiment_v2_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Experiment_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1List_v2_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1List_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Software_v2_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Software_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.T1_v2_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.T1_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Experiment_v3_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Experiment_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1List_v3_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1List_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Software_v3_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Software_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.T1_v3_1.__init__()=bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.T1_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Experiment.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Experiment-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2List.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2List-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Software.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Software-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation.T2.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation.T2-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Experiment_v2_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Experiment_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2List_v2_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2List_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Software_v2_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Software_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.T2_v2_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.T2_v2_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Experiment_v3_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Experiment_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2List_v3_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2List_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Software_v3_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Software_v3_1-class.html#__init__,Method bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.T2_v3_1.__init__()=bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.T2_v3_1-class.html#__init__,Method bmrblib.kinetics.relaxation.Relaxation_v2_1.__init__()=bmrblib.kinetics.relaxation.Relaxation_v2_1-class.html#__init__,Method bmrblib.kinetics.relaxation.Relaxation_v3_0.__init__()=bmrblib.kinetics.relaxation.Relaxation_v3_0-class.html#__init__,Method bmrblib.kinetics.relaxation.Relaxation_v3_1.__init__()=bmrblib.kinetics.relaxation.Relaxation_v3_1-class.html#__init__,Method bmrblib.nmr_star_dict.NMR_STAR.__init__()=bmrblib.nmr_star_dict.NMR_STAR-class.html#__init__,Method bmrblib.pystarlib.File.File.__init__()=bmrblib.pystarlib.File.File-class.html#__init__,Method bmrblib.pystarlib.SaveFrame.SaveFrame.__init__()=bmrblib.pystarlib.SaveFrame.SaveFrame-class.html#__init__,Method bmrblib.pystarlib.TagTable.TagTable.__init__()=bmrblib.pystarlib.TagTable.TagTable-class.html#__init__,Method bmrblib.structure.tensor.Tensor.__init__()=bmrblib.structure.tensor.Tensor-class.html#__init__,Method bmrblib.structure.tensor.TensorList.__init__()=bmrblib.structure.tensor.TensorList-class.html#__init__,Method bmrblib.thermodynamics.model_free.ModelFree.__init__()=bmrblib.thermodynamics.model_free.ModelFree-class.html#__init__,Method bmrblib.thermodynamics.model_free.ModelFreeExperiment.__init__()=bmrblib.thermodynamics.model_free.ModelFreeExperiment-class.html#__init__,Method bmrblib.thermodynamics.model_free.ModelFreeList.__init__()=bmrblib.thermodynamics.model_free.ModelFreeList-class.html#__init__,Method bmrblib.thermodynamics.model_free.ModelFreeSoftware.__init__()=bmrblib.thermodynamics.model_free.ModelFreeSoftware-class.html#__init__,Method bmrblib.thermodynamics.model_free_v3_1.ModelFreeExperiment_v3_1.__init__()=bmrblib.thermodynamics.model_free_v3_1.ModelFreeExperiment_v3_1-class.html#__init__,Method bmrblib.thermodynamics.model_free_v3_1.ModelFreeList_v3_1.__init__()=bmrblib.thermodynamics.model_free_v3_1.ModelFreeList_v3_1-class.html#__init__,Method bmrblib.thermodynamics.model_free_v3_1.ModelFreeSoftware_v3_1.__init__()=bmrblib.thermodynamics.model_free_v3_1.ModelFreeSoftware_v3_1-class.html#__init__,Method bmrblib.thermodynamics.model_free_v3_1.ModelFree_v3_1.__init__()=bmrblib.thermodynamics.model_free_v3_1.ModelFree_v3_1-class.html#__init__,Method data.align_tensor.AlignTensorData.__init__()=data.align_tensor.AlignTensorData-class.html#__init__,Method data.align_tensor.AlignTensorSimList.__init__()=data.align_tensor.AlignTensorSimList-class.html#__init__,Method data.data_classes.Element.__init__()=data.data_classes.Element-class.html#__init__,Method data.data_classes.RelaxListType.__init__()=data.data_classes.RelaxListType-class.html#__init__,Method data.diff_tensor.DiffTensorData.__init__()=data.diff_tensor.DiffTensorData-class.html#__init__,Method data.diff_tensor.DiffTensorSimList.__init__()=data.diff_tensor.DiffTensorSimList-class.html#__init__,Method data.exp_info.ExpInfo.__init__()=data.exp_info.ExpInfo-class.html#__init__,Method data.gui.Analyses.__init__()=data.gui.Analyses-class.html#__init__,Method data.gui.Free_file_format.__init__()=data.gui.Free_file_format-class.html#__init__,Method data.gui.Gui.__init__()=data.gui.Gui-class.html#__init__,Method data.mol_res_spin.MoleculeContainer.__init__()=data.mol_res_spin.MoleculeContainer-class.html#__init__,Method data.mol_res_spin.MoleculeList.__init__()=data.mol_res_spin.MoleculeList-class.html#__init__,Method data.mol_res_spin.ResidueContainer.__init__()=data.mol_res_spin.ResidueContainer-class.html#__init__,Method data.mol_res_spin.ResidueList.__init__()=data.mol_res_spin.ResidueList-class.html#__init__,Method data.mol_res_spin.SpinContainer.__init__()=data.mol_res_spin.SpinContainer-class.html#__init__,Method data.mol_res_spin.SpinList.__init__()=data.mol_res_spin.SpinList-class.html#__init__,Method data.pipe_container.PipeContainer.__init__()=data.pipe_container.PipeContainer-class.html#__init__,Method docs.latex.fetch_docstrings.Fetch_docstrings.__init__()=docs.latex.fetch_docstrings.Fetch_docstrings-class.html#__init__,Method extern.scientific_python.IO.FortranFormat.FortranFormat.__init__()=extern.scientific_python.IO.FortranFormat.FortranFormat-class.html#__init__,Method extern.scientific_python.IO.FortranFormat.FortranLine.__init__()=extern.scientific_python.IO.FortranFormat.FortranLine-class.html#__init__,Method extern.scientific_python.IO.PDB.Atom.__init__()=extern.scientific_python.IO.PDB.Atom-class.html#__init__,Method extern.scientific_python.IO.PDB.Chain.__init__()=extern.scientific_python.IO.PDB.Chain-class.html#__init__,Method extern.scientific_python.IO.PDB.Group.__init__()=extern.scientific_python.IO.PDB.Group-class.html#__init__,Method extern.scientific_python.IO.PDB.NucleotideResidue.__init__()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#__init__,Method extern.scientific_python.IO.PDB.PDBFile.__init__()=extern.scientific_python.IO.PDB.PDBFile-class.html#__init__,Method extern.scientific_python.IO.PDB.ResidueNumber.__init__()=extern.scientific_python.IO.PDB.ResidueNumber-class.html#__init__,Method extern.scientific_python.IO.PDB.Structure.__init__()=extern.scientific_python.IO.PDB.Structure-class.html#__init__,Method extern.scientific_python.IO.PDBSpaceGroups.SpaceGroup.__init__()=extern.scientific_python.IO.PDBSpaceGroups.SpaceGroup-class.html#__init__,Method extern.scientific_python.IO.TextFile.TextFile.__init__()=extern.scientific_python.IO.TextFile.TextFile-class.html#__init__,Method generic_fns.exp_info.Software_store.__init__()=generic_fns.exp_info.Software_store-class.html#__init__,Method generic_fns.mol_res_spin.Selection.__init__()=generic_fns.mol_res_spin.Selection-class.html#__init__,Method generic_fns.molmol.Molmol.__init__()=generic_fns.molmol.Molmol-class.html#__init__,Method generic_fns.pymol_control.Pymol.__init__()=generic_fns.pymol_control.Pymol-class.html#__init__,Method generic_fns.spectrum.Bruker_import.__init__()=generic_fns.spectrum.Bruker_import-class.html#__init__,Method generic_fns.structure.api_base.Base_struct_API.__init__()=generic_fns.structure.api_base.Base_struct_API-class.html#__init__,Method generic_fns.structure.api_base.Displacements.__init__()=generic_fns.structure.api_base.Displacements-class.html#__init__,Method generic_fns.structure.api_base.ModelContainer.__init__()=generic_fns.structure.api_base.ModelContainer-class.html#__init__,Method generic_fns.structure.cones.Base.__init__()=generic_fns.structure.cones.Base-class.html#__init__,Method generic_fns.structure.cones.Cosine.__init__()=generic_fns.structure.cones.Cosine-class.html#__init__,Method generic_fns.structure.cones.Iso_cone.__init__()=generic_fns.structure.cones.Iso_cone-class.html#__init__,Method generic_fns.structure.cones.Zig_zag.__init__()=generic_fns.structure.cones.Zig_zag-class.html#__init__,Method generic_fns.structure.internal.MolContainer.__init__()=generic_fns.structure.internal.MolContainer-class.html#__init__,Method generic_fns.structure.scientific.Scientific_data.__init__()=generic_fns.structure.scientific.Scientific_data-class.html#__init__,Method gui.App.__init__()=gui.App-class.html#__init__,Method gui.about.About_base.__init__()=gui.about.About_base-class.html#__init__,Method gui.about.About_relax.__init__()=gui.about.About_relax-class.html#__init__,Method gui.analyses.Analysis_controller.__init__()=gui.analyses.Analysis_controller-class.html#__init__,Method gui.analyses.auto_model_free.About_window.__init__()=gui.analyses.auto_model_free.About_window-class.html#__init__,Method gui.analyses.auto_model_free.Auto_model_free.__init__()=gui.analyses.auto_model_free.Auto_model_free-class.html#__init__,Method gui.analyses.auto_model_free.Local_tm_list.__init__()=gui.analyses.auto_model_free.Local_tm_list-class.html#__init__,Method gui.analyses.auto_model_free.ModelSelListCtrl.__init__()=gui.analyses.auto_model_free.ModelSelListCtrl-class.html#__init__,Method gui.analyses.auto_model_free.Model_sel_window.__init__()=gui.analyses.auto_model_free.Model_sel_window-class.html#__init__,Method gui.analyses.auto_model_free.Protocol_mode_sel_window.__init__()=gui.analyses.auto_model_free.Protocol_mode_sel_window-class.html#__init__,Method gui.analyses.auto_noe.Auto_noe.__init__()=gui.analyses.auto_noe.Auto_noe-class.html#__init__,Method gui.analyses.auto_rx_base.Auto_rx.__init__()=gui.analyses.auto_rx_base.Auto_rx-class.html#__init__,Method gui.analyses.base.Base_analysis.__init__()=gui.analyses.base.Base_analysis-class.html#__init__,Method gui.analyses.elements.Spin_ctrl.__init__()=gui.analyses.elements.Spin_ctrl-class.html#__init__,Method gui.analyses.elements.Text_ctrl.__init__()=gui.analyses.elements.Text_ctrl-class.html#__init__,Method gui.analyses.execute.Execute.__init__()=gui.analyses.execute.Execute-class.html#__init__,Method gui.analyses.results_analysis.Final_results.__init__()=gui.analyses.results_analysis.Final_results-class.html#__init__,Method gui.components.combo_list.Combo_list.__init__()=gui.components.combo_list.Combo_list-class.html#__init__,Method gui.components.grid.Delay_num_cell_editor.__init__()=gui.components.grid.Delay_num_cell_editor-class.html#__init__,Method gui.components.grid.Grid_base.__init__()=gui.components.grid.Grid_base-class.html#__init__,Method gui.components.relax_data.Relax_data_list.__init__()=gui.components.relax_data.Relax_data_list-class.html#__init__,Method gui.components.spectrum.Spectra_list.__init__()=gui.components.spectrum.Spectra_list-class.html#__init__,Method gui.controller.Controller.__init__()=gui.controller.Controller-class.html#__init__,Method gui.controller.LogCtrl.__init__()=gui.controller.LogCtrl-class.html#__init__,Method gui.controller.Redirect_text.__init__()=gui.controller.Redirect_text-class.html#__init__,Method gui.filedialog.RelaxDirDialog.__init__()=gui.filedialog.RelaxDirDialog-class.html#__init__,Method gui.filedialog.RelaxFileDialog.__init__()=gui.filedialog.RelaxFileDialog-class.html#__init__,Method gui.icons.Relax_task_bar_icon.__init__()=gui.icons.Relax_task_bar_icon-class.html#__init__,Method gui.interpreter.Interpreter_thread.__init__()=gui.interpreter.Interpreter_thread-class.html#__init__,Method gui.menu.Menu.__init__()=gui.menu.Menu-class.html#__init__,Method gui.menu.Uf_menus.__init__()=gui.menu.Uf_menus-class.html#__init__,Method gui.message.Missing_data.__init__()=gui.message.Missing_data-class.html#__init__,Method gui.message.Question.__init__()=gui.message.Question-class.html#__init__,Method gui.paths.I16x16.__init__()=gui.paths.I16x16-class.html#__init__,Method gui.paths.I22x22.__init__()=gui.paths.I22x22-class.html#__init__,Method gui.paths.I32x32.__init__()=gui.paths.I32x32-class.html#__init__,Method gui.paths.I48x48.__init__()=gui.paths.I48x48-class.html#__init__,Method gui.pipe_editor.Pipe_editor.__init__()=gui.pipe_editor.Pipe_editor-class.html#__init__,Method gui.references.References.__init__()=gui.references.References-class.html#__init__,Method gui.relax_gui.Main.__init__()=gui.relax_gui.Main-class.html#__init__,Method gui.relax_prompt.InterpClass.__init__()=gui.relax_prompt.InterpClass-class.html#__init__,Method gui.relax_prompt.Prompt.__init__()=gui.relax_prompt.Prompt-class.html#__init__,Method gui.relax_prompt._Exit.__init__()=gui.relax_prompt._Exit-class.html#__init__,Method gui.results_viewer.Results_viewer.__init__()=gui.results_viewer.Results_viewer-class.html#__init__,Method gui.settings.Base_window.__init__()=gui.settings.Base_window-class.html#__init__,Method gui.settings.Free_file_format.__init__()=gui.settings.Free_file_format-class.html#__init__,Method gui.spin_viewer.containers.Container.__init__()=gui.spin_viewer.containers.Container-class.html#__init__,Method gui.spin_viewer.containers.Molecule.__init__()=gui.spin_viewer.containers.Molecule-class.html#__init__,Method gui.spin_viewer.containers.Residue.__init__()=gui.spin_viewer.containers.Residue-class.html#__init__,Method gui.spin_viewer.containers.Root.__init__()=gui.spin_viewer.containers.Root-class.html#__init__,Method gui.spin_viewer.containers.Spin.__init__()=gui.spin_viewer.containers.Spin-class.html#__init__,Method gui.spin_viewer.frame.Spin_view_window.__init__()=gui.spin_viewer.frame.Spin_view_window-class.html#__init__,Method gui.spin_viewer.splitter.Tree_splitter.__init__()=gui.spin_viewer.splitter.Tree_splitter-class.html#__init__,Method gui.spin_viewer.tree.Mol_res_spin_tree.__init__()=gui.spin_viewer.tree.Mol_res_spin_tree-class.html#__init__,Method gui.user_functions.User_functions.__init__()=gui.user_functions.User_functions-class.html#__init__,Method gui.user_functions.base.UF_base.__init__()=gui.user_functions.base.UF_base-class.html#__init__,Method gui.user_functions.base.UF_page.__init__()=gui.user_functions.base.UF_page-class.html#__init__,Method gui.user_functions.sys_info.Sys_info_page.__init__()=gui.user_functions.sys_info.Sys_info_page-class.html#__init__,Method gui.wizard.Wiz_page.__init__()=gui.wizard.Wiz_page-class.html#__init__,Method gui.wizard.Wiz_window.__init__()=gui.wizard.Wiz_window-class.html#__init__,Method info.Info_box.__init__()=info.Info_box-class.html#__init__,Method maths_fns.consistency_tests.Consistency.__init__()=maths_fns.consistency_tests.Consistency-class.html#__init__,Method maths_fns.consistency_tests.Data.__init__()=maths_fns.consistency_tests.Data-class.html#__init__,Method maths_fns.frame_order.Frame_order.__init__()=maths_fns.frame_order.Frame_order-class.html#__init__,Method maths_fns.jw_mapping.Data.__init__()=maths_fns.jw_mapping.Data-class.html#__init__,Method maths_fns.jw_mapping.Mapping.__init__()=maths_fns.jw_mapping.Mapping-class.html#__init__,Method maths_fns.mf.Data.__init__()=maths_fns.mf.Data-class.html#__init__,Method maths_fns.mf.Mf.__init__()=maths_fns.mf.Mf-class.html#__init__,Method maths_fns.n_state_model.N_state_opt.__init__()=maths_fns.n_state_model.N_state_opt-class.html#__init__,Method minfx.base_classes.Conjugate_gradient.__init__()=minfx.base_classes.Conjugate_gradient-class.html#__init__,Method minfx.base_classes.Hessian_mods.__init__()=minfx.base_classes.Hessian_mods-class.html#__init__,Method minfx.base_classes.Line_search.__init__()=minfx.base_classes.Line_search-class.html#__init__,Method minfx.base_classes.Min.__init__()=minfx.base_classes.Min-class.html#__init__,Method minfx.base_classes.Trust_region.__init__()=minfx.base_classes.Trust_region-class.html#__init__,Method minfx.bfgs.Bfgs.__init__()=minfx.bfgs.Bfgs-class.html#__init__,Method minfx.cauchy_point.Cauchy_point.__init__()=minfx.cauchy_point.Cauchy_point-class.html#__init__,Method minfx.constraint_linear.Constraint_linear.__init__()=minfx.constraint_linear.Constraint_linear-class.html#__init__,Method minfx.coordinate_descent.Coordinate_descent.__init__()=minfx.coordinate_descent.Coordinate_descent-class.html#__init__,Method minfx.dogleg.Dogleg.__init__()=minfx.dogleg.Dogleg-class.html#__init__,Method minfx.errors.MinfxError.__init__()=minfx.errors.MinfxError-class.html#__init__,Method minfx.exact_trust_region.Exact_trust_region.__init__()=minfx.exact_trust_region.Exact_trust_region-class.html#__init__,Method minfx.fletcher_reeves_cg.Fletcher_reeves.__init__()=minfx.fletcher_reeves_cg.Fletcher_reeves-class.html#__init__,Method minfx.hestenes_stiefel_cg.Hestenes_stiefel.__init__()=minfx.hestenes_stiefel_cg.Hestenes_stiefel-class.html#__init__,Method minfx.levenberg_marquardt.Levenberg_marquardt.__init__()=minfx.levenberg_marquardt.Levenberg_marquardt-class.html#__init__,Method minfx.method_of_multipliers.Method_of_multipliers.__init__()=minfx.method_of_multipliers.Method_of_multipliers-class.html#__init__,Method minfx.ncg.Ncg.__init__()=minfx.ncg.Ncg-class.html#__init__,Method minfx.newton.Newton.__init__()=minfx.newton.Newton-class.html#__init__,Method minfx.polak_ribiere_cg.Polak_ribiere.__init__()=minfx.polak_ribiere_cg.Polak_ribiere-class.html#__init__,Method minfx.polak_ribiere_plus_cg.Polak_ribiere_plus.__init__()=minfx.polak_ribiere_plus_cg.Polak_ribiere_plus-class.html#__init__,Method minfx.scipy_subset.anneal._state.__init__()=minfx.scipy_subset.anneal._state-class.html#__init__,Method minfx.scipy_subset.anneal.base_schedule.__init__()=minfx.scipy_subset.anneal.base_schedule-class.html#__init__,Method minfx.simplex.Simplex.__init__()=minfx.simplex.Simplex-class.html#__init__,Method minfx.steepest_descent.Steepest_descent.__init__()=minfx.steepest_descent.Steepest_descent-class.html#__init__,Method minfx.steihaug_cg.Steihaug.__init__()=minfx.steihaug_cg.Steihaug-class.html#__init__,Method multi.commands.Exit_command.__init__()=multi.commands.Exit_command-class.html#__init__,Method multi.commands.Get_name_command.__init__()=multi.commands.Get_name_command-class.html#__init__,Method multi.commands.Set_processor_property_command.__init__()=multi.commands.Set_processor_property_command-class.html#__init__,Method multi.mpi4py_processor.Mpi4py_processor.__init__()=multi.mpi4py_processor.Mpi4py_processor-class.html#__init__,Method multi.multi_processor_base.Batched_result_command.__init__()=multi.multi_processor_base.Batched_result_command-class.html#__init__,Method multi.multi_processor_base.Exit_queue_result_command.__init__()=multi.multi_processor_base.Exit_queue_result_command-class.html#__init__,Method multi.multi_processor_base.Multi_processor.__init__()=multi.multi_processor_base.Multi_processor-class.html#__init__,Method multi.multi_processor_base.Result_queue.__init__()=multi.multi_processor_base.Result_queue-class.html#__init__,Method multi.multi_processor_base.Threaded_result_queue.__init__()=multi.multi_processor_base.Threaded_result_queue-class.html#__init__,Method multi.multi_processor_base.Too_few_slaves_exception.__init__()=multi.multi_processor_base.Too_few_slaves_exception-class.html#__init__,Method multi.processor.Application_callback.__init__()=multi.processor.Application_callback-class.html#__init__,Method multi.processor.Capturing_exception.__init__()=multi.processor.Capturing_exception-class.html#__init__,Method multi.processor.Null_result_command.__init__()=multi.processor.Null_result_command-class.html#__init__,Method multi.processor.Processor.__init__()=multi.processor.Processor-class.html#__init__,Method multi.processor.Result.__init__()=multi.processor.Result-class.html#__init__,Method multi.processor.Result_command.__init__()=multi.processor.Result_command-class.html#__init__,Method multi.processor.Result_exception.__init__()=multi.processor.Result_exception-class.html#__init__,Method multi.processor.Result_string.__init__()=multi.processor.Result_string-class.html#__init__,Method multi.processor.Slave_command.__init__()=multi.processor.Slave_command-class.html#__init__,Method multi.processor_io.IO_filter.__init__()=multi.processor_io.IO_filter-class.html#__init__,Method multi.processor_io.PrependStringIO.__init__()=multi.processor_io.PrependStringIO-class.html#__init__,Method multi.uni_processor.Uni_processor.__init__()=multi.uni_processor.Uni_processor-class.html#__init__,Method opendx.base_map.Base_Map.__init__()=opendx.base_map.Base_Map-class.html#__init__,Method prompt.base_class.Basic_class.__init__()=prompt.base_class.Basic_class-class.html#__init__,Method prompt.base_class.Exec_info.__init__()=prompt.base_class.Exec_info-class.html#__init__,Method prompt.base_class.User_fn_class.__init__()=prompt.base_class.User_fn_class-class.html#__init__,Method prompt.doc_string.Strings.__init__()=prompt.doc_string.Strings-class.html#__init__,Method prompt.interpreter.Interpreter.__init__()=prompt.interpreter.Interpreter-class.html#__init__,Method prompt.tab_completion.Tab_completion.__init__()=prompt.tab_completion.Tab_completion-class.html#__init__,Method relax.Relax.__init__()=relax.Relax-class.html#__init__,Method relax.RelaxParser.__init__()=relax.RelaxParser-class.html#__init__,Method relax_errors.BaseArgError.__init__()=relax_errors.BaseArgError-class.html#__init__,Method relax_errors.RelaxArgNotInListError.__init__()=relax_errors.RelaxArgNotInListError-class.html#__init__,Method relax_errors.RelaxArgNotNoneError.__init__()=relax_errors.RelaxArgNotNoneError-class.html#__init__,Method relax_errors.RelaxDiffMolNumError.__init__()=relax_errors.RelaxDiffMolNumError-class.html#__init__,Method relax_errors.RelaxDiffResNumError.__init__()=relax_errors.RelaxDiffResNumError-class.html#__init__,Method relax_errors.RelaxDiffSeqError.__init__()=relax_errors.RelaxDiffSeqError-class.html#__init__,Method relax_errors.RelaxDiffSpinNumError.__init__()=relax_errors.RelaxDiffSpinNumError-class.html#__init__,Method relax_errors.RelaxDirError.__init__()=relax_errors.RelaxDirError-class.html#__init__,Method relax_errors.RelaxError.__init__()=relax_errors.RelaxError-class.html#__init__,Method relax_errors.RelaxFault.__init__()=relax_errors.RelaxFault-class.html#__init__,Method relax_errors.RelaxFileEmptyError.__init__()=relax_errors.RelaxFileEmptyError-class.html#__init__,Method relax_errors.RelaxFileError.__init__()=relax_errors.RelaxFileError-class.html#__init__,Method relax_errors.RelaxFileOverwriteError.__init__()=relax_errors.RelaxFileOverwriteError-class.html#__init__,Method relax_errors.RelaxFromXMLNotEmptyError.__init__()=relax_errors.RelaxFromXMLNotEmptyError-class.html#__init__,Method relax_errors.RelaxFuncSetupError.__init__()=relax_errors.RelaxFuncSetupError-class.html#__init__,Method relax_errors.RelaxImplementError.__init__()=relax_errors.RelaxImplementError-class.html#__init__,Method relax_errors.RelaxInfError.__init__()=relax_errors.RelaxInfError-class.html#__init__,Method relax_errors.RelaxInvalidColourError.__init__()=relax_errors.RelaxInvalidColourError-class.html#__init__,Method relax_errors.RelaxInvalidDataError.__init__()=relax_errors.RelaxInvalidDataError-class.html#__init__,Method relax_errors.RelaxInvalidError.__init__()=relax_errors.RelaxInvalidError-class.html#__init__,Method relax_errors.RelaxInvalidSeqError.__init__()=relax_errors.RelaxInvalidSeqError-class.html#__init__,Method relax_errors.RelaxLenError.__init__()=relax_errors.RelaxLenError-class.html#__init__,Method relax_errors.RelaxMfError.__init__()=relax_errors.RelaxMfError-class.html#__init__,Method relax_errors.RelaxMissingBinaryError.__init__()=relax_errors.RelaxMissingBinaryError-class.html#__init__,Method relax_errors.RelaxModelError.__init__()=relax_errors.RelaxModelError-class.html#__init__,Method relax_errors.RelaxMolSelectDisallowError.__init__()=relax_errors.RelaxMolSelectDisallowError-class.html#__init__,Method relax_errors.RelaxNaNError.__init__()=relax_errors.RelaxNaNError-class.html#__init__,Method relax_errors.RelaxNoInPathError.__init__()=relax_errors.RelaxNoInPathError-class.html#__init__,Method relax_errors.RelaxNoModelError.__init__()=relax_errors.RelaxNoModelError-class.html#__init__,Method relax_errors.RelaxNoModuleInstallError.__init__()=relax_errors.RelaxNoModuleInstallError-class.html#__init__,Method relax_errors.RelaxNoPCSError.__init__()=relax_errors.RelaxNoPCSError-class.html#__init__,Method relax_errors.RelaxNoPdbChainError.__init__()=relax_errors.RelaxNoPdbChainError-class.html#__init__,Method relax_errors.RelaxNoPdbError.__init__()=relax_errors.RelaxNoPdbError-class.html#__init__,Method relax_errors.RelaxNoPipeError.__init__()=relax_errors.RelaxNoPipeError-class.html#__init__,Method relax_errors.RelaxNoRDCError.__init__()=relax_errors.RelaxNoRDCError-class.html#__init__,Method relax_errors.RelaxNoResError.__init__()=relax_errors.RelaxNoResError-class.html#__init__,Method relax_errors.RelaxNoRiError.__init__()=relax_errors.RelaxNoRiError-class.html#__init__,Method relax_errors.RelaxNoSequenceError.__init__()=relax_errors.RelaxNoSequenceError-class.html#__init__,Method relax_errors.RelaxNoSimError.__init__()=relax_errors.RelaxNoSimError-class.html#__init__,Method relax_errors.RelaxNoSpectraError.__init__()=relax_errors.RelaxNoSpectraError-class.html#__init__,Method relax_errors.RelaxNoSpinError.__init__()=relax_errors.RelaxNoSpinError-class.html#__init__,Method relax_errors.RelaxNoSpinSpecError.__init__()=relax_errors.RelaxNoSpinSpecError-class.html#__init__,Method relax_errors.RelaxNoTensorError.__init__()=relax_errors.RelaxNoTensorError-class.html#__init__,Method relax_errors.RelaxNoValueError.__init__()=relax_errors.RelaxNoValueError-class.html#__init__,Method relax_errors.RelaxNoVectorsError.__init__()=relax_errors.RelaxNoVectorsError-class.html#__init__,Method relax_errors.RelaxNonExecError.__init__()=relax_errors.RelaxNonExecError-class.html#__init__,Method relax_errors.RelaxNoneError.__init__()=relax_errors.RelaxNoneError-class.html#__init__,Method relax_errors.RelaxNucleusError.__init__()=relax_errors.RelaxNucleusError-class.html#__init__,Method relax_errors.RelaxPCSError.__init__()=relax_errors.RelaxPCSError-class.html#__init__,Method relax_errors.RelaxParamSetError.__init__()=relax_errors.RelaxParamSetError-class.html#__init__,Method relax_errors.RelaxPdbError.__init__()=relax_errors.RelaxPdbError-class.html#__init__,Method relax_errors.RelaxPdbLoadError.__init__()=relax_errors.RelaxPdbLoadError-class.html#__init__,Method relax_errors.RelaxPipeError.__init__()=relax_errors.RelaxPipeError-class.html#__init__,Method relax_errors.RelaxProgError.__init__()=relax_errors.RelaxProgError-class.html#__init__,Method relax_errors.RelaxProgFailError.__init__()=relax_errors.RelaxProgFailError-class.html#__init__,Method relax_errors.RelaxProtonTypeError.__init__()=relax_errors.RelaxProtonTypeError-class.html#__init__,Method relax_errors.RelaxRDCError.__init__()=relax_errors.RelaxRDCError-class.html#__init__,Method relax_errors.RelaxRegExpError.__init__()=relax_errors.RelaxRegExpError-class.html#__init__,Method relax_errors.RelaxResSelectDisallowError.__init__()=relax_errors.RelaxResSelectDisallowError-class.html#__init__,Method relax_errors.RelaxRiError.__init__()=relax_errors.RelaxRiError-class.html#__init__,Method relax_errors.RelaxSequenceError.__init__()=relax_errors.RelaxSequenceError-class.html#__init__,Method relax_errors.RelaxSpectraError.__init__()=relax_errors.RelaxSpectraError-class.html#__init__,Method relax_errors.RelaxSpinSelectDisallowError.__init__()=relax_errors.RelaxSpinSelectDisallowError-class.html#__init__,Method relax_errors.RelaxSpinTypeError.__init__()=relax_errors.RelaxSpinTypeError-class.html#__init__,Method relax_errors.RelaxSpinsNotLoadedError.__init__()=relax_errors.RelaxSpinsNotLoadedError-class.html#__init__,Method relax_errors.RelaxStyleError.__init__()=relax_errors.RelaxStyleError-class.html#__init__,Method relax_errors.RelaxTensorError.__init__()=relax_errors.RelaxTensorError-class.html#__init__,Method relax_errors.RelaxUnknownDataTypeError.__init__()=relax_errors.RelaxUnknownDataTypeError-class.html#__init__,Method relax_errors.RelaxUnknownParamCombError.__init__()=relax_errors.RelaxUnknownParamCombError-class.html#__init__,Method relax_errors.RelaxUnknownParamError.__init__()=relax_errors.RelaxUnknownParamError-class.html#__init__,Method relax_errors.RelaxValueError.__init__()=relax_errors.RelaxValueError-class.html#__init__,Method relax_io.DummyFileObject.__init__()=relax_io.DummyFileObject-class.html#__init__,Method relax_io.SplitIO.__init__()=relax_io.SplitIO-class.html#__init__,Method relax_warnings.RelaxDeselectWarning.__init__()=relax_warnings.RelaxDeselectWarning-class.html#__init__,Method relax_warnings.RelaxFileEmptyWarning.__init__()=relax_warnings.RelaxFileEmptyWarning-class.html#__init__,Method relax_warnings.RelaxNoAtomWarning.__init__()=relax_warnings.RelaxNoAtomWarning-class.html#__init__,Method relax_warnings.RelaxNoPDBFileWarning.__init__()=relax_warnings.RelaxNoPDBFileWarning-class.html#__init__,Method relax_warnings.RelaxNoSpinWarning.__init__()=relax_warnings.RelaxNoSpinWarning-class.html#__init__,Method relax_warnings.RelaxWarning.__init__()=relax_warnings.RelaxWarning-class.html#__init__,Method relax_warnings.RelaxZeroVectorWarning.__init__()=relax_warnings.RelaxZeroVectorWarning-class.html#__init__,Method script-sconstruct.Main.__init__()=script-sconstruct.Main-class.html#__init__,Method specific_fns.consistency_tests.Consistency_tests.__init__()=specific_fns.consistency_tests.Consistency_tests-class.html#__init__,Method specific_fns.frame_order.Frame_order.__init__()=specific_fns.frame_order.Frame_order-class.html#__init__,Method specific_fns.jw_mapping.Jw_mapping.__init__()=specific_fns.jw_mapping.Jw_mapping-class.html#__init__,Method specific_fns.model_free.Model_free.__init__()=specific_fns.model_free.Model_free-class.html#__init__,Method specific_fns.model_free.multi_processor_commands.MF_grid_command.__init__()=specific_fns.model_free.multi_processor_commands.MF_grid_command-class.html#__init__,Method specific_fns.model_free.multi_processor_commands.MF_memo.__init__()=specific_fns.model_free.multi_processor_commands.MF_memo-class.html#__init__,Method specific_fns.model_free.multi_processor_commands.MF_minimise_command.__init__()=specific_fns.model_free.multi_processor_commands.MF_minimise_command-class.html#__init__,Method specific_fns.model_free.multi_processor_commands.MF_result_command.__init__()=specific_fns.model_free.multi_processor_commands.MF_result_command-class.html#__init__,Method specific_fns.n_state_model.N_state_model.__init__()=specific_fns.n_state_model.N_state_model-class.html#__init__,Method specific_fns.noe.Noe.__init__()=specific_fns.noe.Noe-class.html#__init__,Method specific_fns.relax_fit.Relax_fit.__init__()=specific_fns.relax_fit.Relax_fit-class.html#__init__,Method status.Auto_analysis.__init__()=status.Auto_analysis-class.html#__init__,Method status.Exec_lock.__init__()=status.Exec_lock-class.html#__init__,Method status.Observer.__init__()=status.Observer-class.html#__init__,Method status.Relax_lock.__init__()=status.Relax_lock-class.html#__init__,Method test_suite.gui_tests.base_classes.GuiTestCase.__init__()=test_suite.gui_tests.base_classes.GuiTestCase-class.html#__init__,Method test_suite.system_tests.base_classes.SystemTestCase.__init__()=test_suite.system_tests.base_classes.SystemTestCase-class.html#__init__,Method test_suite.system_tests.bmrb.Bmrb.__init__()=test_suite.system_tests.bmrb.Bmrb-class.html#__init__,Method test_suite.system_tests.frame_order.Frame_order.__init__()=test_suite.system_tests.frame_order.Frame_order-class.html#__init__,Method test_suite.system_tests.structure.Structure.__init__()=test_suite.system_tests.structure.Structure-class.html#__init__,Method test_suite.system_tests.unit_vectors.Unit_vectors.__init__()=test_suite.system_tests.unit_vectors.Unit_vectors-class.html#__init__,Method test_suite.test_suite_runner.Test_suite_runner.__init__()=test_suite.test_suite_runner.Test_suite_runner-class.html#__init__,Method test_suite.unit_tests._generic_fns._structure.test_api_base.Test_api_base.__init__()=test_suite.unit_tests._generic_fns._structure.test_api_base.Test_api_base-class.html#__init__,Method test_suite.unit_tests._generic_fns._structure.test_scientific.Test_scientific.__init__()=test_suite.unit_tests._generic_fns._structure.test_scientific.Test_scientific-class.html#__init__,Method test_suite.unit_tests._maths_fns.test_frame_order_matrix_ops.Test_frame_order_matrix_ops.__init__()=test_suite.unit_tests._maths_fns.test_frame_order_matrix_ops.Test_frame_order_matrix_ops-class.html#__init__,Method test_suite.unit_tests.unit_test_runner.ImportErrorTestCase.__init__()=test_suite.unit_tests.unit_test_runner.ImportErrorTestCase-class.html#__init__,Method test_suite.unit_tests.unit_test_runner.Test_finder.__init__()=test_suite.unit_tests.unit_test_runner.Test_finder-class.html#__init__,Method test_suite.unit_tests.unit_test_runner.Unit_test_runner.__init__()=test_suite.unit_tests.unit_test_runner.Unit_test_runner-class.html#__init__"><a title="auto_analyses.dauvergne_protocol.dAuvergne_protocol.__init__
auto_analyses.noe.NOE_calc.__init__
auto_analyses.relax_fit.Relax_fit.__init__
auto_analyses.stereochem_analysis.Stereochem_analysis.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropy.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy.CSAnisotropyExperiment.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy.ChemShiftAnisotropy.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropyExperiment_v3_1.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropySoftware_v3_1.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.CSAnisotropy_v3_1.__init__
bmrblib.NMR_parameters.chem_shift_anisotropy_v3_1.ChemShiftAnisotropy_v3_1.__init__
bmrblib.assembly_supercategory.entity.Entity.__init__
bmrblib.assembly_supercategory.entity.EntityCompIndex.__init__
bmrblib.assembly_supercategory.entity_v2_1.EntityCompIndex_v2_1.__init__
bmrblib.assembly_supercategory.entity_v2_1.Entity_v2_1.__init__
bmrblib.assembly_supercategory.entity_v3_1.EntityCompIndex_v3_1.__init__
bmrblib.assembly_supercategory.entity_v3_1.Entity_v3_1.__init__
bmrblib.base_classes.BaseSaveframe.__init__
bmrblib.base_classes.MissingSaveframe.__init__
bmrblib.base_classes.TagCategory.__init__
bmrblib.base_classes.TagCategoryFree.__init__
bmrblib.base_classes.TagObject.__init__
bmrblib.base_classes.TagTranslationTable.__init__
bmrblib.citations.citations.Citations.__init__
bmrblib.citations.citations.CitationsAuthor.__init__
bmrblib.citations.citations_v3_1.CitationsAuthor_v3_1.__init__
bmrblib.citations.citations_v3_1.Citations_v3_1.__init__
bmrblib.experimental_details.experiment.Experiment.__init__
bmrblib.experimental_details.experiment.ExperimentList.__init__
bmrblib.experimental_details.experiment_v3_1.ExperimentList_v3_1.__init__
bmrblib.experimental_details.experiment_v3_1.Experiment_v3_1.__init__
bmrblib.experimental_details.method.Method.__init__
bmrblib.experimental_details.method.MethodCitation.__init__
bmrblib.experimental_details.method.MethodFile.__init__
bmrblib.experimental_details.method.MethodParam.__init__
bmrblib.experimental_details.method_v3_1.MethodCitation_v3_1.__init__
bmrblib.experimental_details.method_v3_1.MethodFile_v3_1.__init__
bmrblib.experimental_details.method_v3_1.MethodParam_v3_1.__init__
bmrblib.experimental_details.method_v3_1.Method_v3_1.__init__
bmrblib.experimental_details.nmr_spectrometer.NMRSpectrometer.__init__
bmrblib.experimental_details.nmr_spectrometer_v3_1.NMRSpectrometer_v3_1.__init__
bmrblib.experimental_details.sample_conditions.SampleConditionCitation.__init__
bmrblib.experimental_details.sample_conditions.SampleConditionList.__init__
bmrblib.experimental_details.sample_conditions.SampleConditionVariable.__init__
bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionList_v2_1.__init__
bmrblib.experimental_details.sample_conditions_v2_1.SampleConditionVariable_v2_1.__init__
bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionCitation_v3_1.__init__
bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionList_v3_1.__init__
bmrblib.experimental_details.sample_conditions_v3_1.SampleConditionVariable_v3_1.__init__
bmrblib.experimental_details.software.Software.__init__
bmrblib.experimental_details.software.SoftwareCitation.__init__
bmrblib.experimental_details.software.Task.__init__
bmrblib.experimental_details.software.Vendor.__init__
bmrblib.experimental_details.software_v3_1.SoftwareCitation_v3_1.__init__
bmrblib.experimental_details.software_v3_1.Software_v3_1.__init__
bmrblib.experimental_details.software_v3_1.Task_v3_1.__init__
bmrblib.experimental_details.software_v3_1.Vendor_v3_1.__init__
bmrblib.kinetics.auto_relaxation.AutoRelaxation.__init__
bmrblib.kinetics.auto_relaxation.AutoRelaxationExperiment.__init__
bmrblib.kinetics.auto_relaxation.AutoRelaxationList.__init__
bmrblib.kinetics.auto_relaxation.AutoRelaxationSoftware.__init__
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationExperiment_v3_1.__init__
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationList_v3_1.__init__
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSoftware_v3_1.__init__
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxation_v3_1.__init__
bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOE.__init__
bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEExperiment.__init__
bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOEList.__init__
bmrblib.kinetics.heteronucl_NOEs.HeteronuclNOESoftware.__init__
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEExperiment_v2_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOEList_v2_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESoftware_v2_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOE_v2_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEExperiment_v3_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOEList_v3_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOESoftware_v3_1.__init__
bmrblib.kinetics.heteronucl_NOEs_v3_1.HeteronuclNOE_v3_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Experiment.__init__
bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1List.__init__
bmrblib.kinetics.heteronucl_T1_relaxation.HeteronuclT1Software.__init__
bmrblib.kinetics.heteronucl_T1_relaxation.T1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Experiment_v2_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1List_v2_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Software_v2_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.T1_v2_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Experiment_v3_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1List_v3_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Software_v3_1.__init__
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.T1_v3_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Experiment.__init__
bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2List.__init__
bmrblib.kinetics.heteronucl_T2_relaxation.HeteronuclT2Software.__init__
bmrblib.kinetics.heteronucl_T2_relaxation.T2.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Experiment_v2_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2List_v2_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Software_v2_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.T2_v2_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Experiment_v3_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2List_v3_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Software_v3_1.__init__
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.T2_v3_1.__init__
bmrblib.kinetics.relaxation.Relaxation_v2_1.__init__
bmrblib.kinetics.relaxation.Relaxation_v3_0.__init__
bmrblib.kinetics.relaxation.Relaxation_v3_1.__init__
bmrblib.nmr_star_dict.NMR_STAR.__init__
bmrblib.pystarlib.File.File.__init__
bmrblib.pystarlib.SaveFrame.SaveFrame.__init__
bmrblib.pystarlib.TagTable.TagTable.__init__
bmrblib.structure.tensor.Tensor.__init__
bmrblib.structure.tensor.TensorList.__init__
bmrblib.thermodynamics.model_free.ModelFree.__init__
bmrblib.thermodynamics.model_free.ModelFreeExperiment.__init__
bmrblib.thermodynamics.model_free.ModelFreeList.__init__
bmrblib.thermodynamics.model_free.ModelFreeSoftware.__init__
bmrblib.thermodynamics.model_free_v3_1.ModelFreeExperiment_v3_1.__init__
bmrblib.thermodynamics.model_free_v3_1.ModelFreeList_v3_1.__init__
bmrblib.thermodynamics.model_free_v3_1.ModelFreeSoftware_v3_1.__init__
bmrblib.thermodynamics.model_free_v3_1.ModelFree_v3_1.__init__
data.align_tensor.AlignTensorData.__init__
data.align_tensor.AlignTensorSimList.__init__
data.data_classes.Element.__init__
data.data_classes.RelaxListType.__init__
data.diff_tensor.DiffTensorData.__init__
data.diff_tensor.DiffTensorSimList.__init__
data.exp_info.ExpInfo.__init__
data.gui.Analyses.__init__
data.gui.Free_file_format.__init__
data.gui.Gui.__init__
data.mol_res_spin.MoleculeContainer.__init__
data.mol_res_spin.MoleculeList.__init__
data.mol_res_spin.ResidueContainer.__init__
data.mol_res_spin.ResidueList.__init__
data.mol_res_spin.SpinContainer.__init__
data.mol_res_spin.SpinList.__init__
data.pipe_container.PipeContainer.__init__
docs.latex.fetch_docstrings.Fetch_docstrings.__init__
extern.scientific_python.IO.FortranFormat.FortranFormat.__init__
extern.scientific_python.IO.FortranFormat.FortranLine.__init__
extern.scientific_python.IO.PDB.Atom.__init__
extern.scientific_python.IO.PDB.Chain.__init__
extern.scientific_python.IO.PDB.Group.__init__
extern.scientific_python.IO.PDB.NucleotideResidue.__init__
extern.scientific_python.IO.PDB.PDBFile.__init__
extern.scientific_python.IO.PDB.ResidueNumber.__init__
extern.scientific_python.IO.PDB.Structure.__init__
extern.scientific_python.IO.PDBSpaceGroups.SpaceGroup.__init__
extern.scientific_python.IO.TextFile.TextFile.__init__
generic_fns.exp_info.Software_store.__init__
generic_fns.mol_res_spin.Selection.__init__
generic_fns.molmol.Molmol.__init__
generic_fns.pymol_control.Pymol.__init__
generic_fns.spectrum.Bruker_import.__init__
generic_fns.structure.api_base.Base_struct_API.__init__
generic_fns.structure.api_base.Displacements.__init__
generic_fns.structure.api_base.ModelContainer.__init__
generic_fns.structure.cones.Base.__init__
generic_fns.structure.cones.Cosine.__init__
generic_fns.structure.cones.Iso_cone.__init__
generic_fns.structure.cones.Zig_zag.__init__
generic_fns.structure.internal.MolContainer.__init__
generic_fns.structure.scientific.Scientific_data.__init__
gui.App.__init__
gui.about.About_base.__init__
gui.about.About_relax.__init__
gui.analyses.Analysis_controller.__init__
gui.analyses.auto_model_free.About_window.__init__
gui.analyses.auto_model_free.Auto_model_free.__init__
gui.analyses.auto_model_free.Local_tm_list.__init__
gui.analyses.auto_model_free.ModelSelListCtrl.__init__
gui.analyses.auto_model_free.Model_sel_window.__init__
gui.analyses.auto_model_free.Protocol_mode_sel_window.__init__
gui.analyses.auto_noe.Auto_noe.__init__
gui.analyses.auto_rx_base.Auto_rx.__init__
gui.analyses.base.Base_analysis.__init__
gui.analyses.elements.Spin_ctrl.__init__
gui.analyses.elements.Text_ctrl.__init__
gui.analyses.execute.Execute.__init__
gui.analyses.results_analysis.Final_results.__init__
gui.components.combo_list.Combo_list.__init__
gui.components.grid.Delay_num_cell_editor.__init__
gui.components.grid.Grid_base.__init__
gui.components.relax_data.Relax_data_list.__init__
gui.components.spectrum.Spectra_list.__init__
gui.controller.Controller.__init__
gui.controller.LogCtrl.__init__
gui.controller.Redirect_text.__init__
gui.filedialog.RelaxDirDialog.__init__
gui.filedialog.RelaxFileDialog.__init__
gui.icons.Relax_task_bar_icon.__init__
gui.interpreter.Interpreter_thread.__init__
gui.menu.Menu.__init__
gui.menu.Uf_menus.__init__
gui.message.Missing_data.__init__
gui.message.Question.__init__
gui.paths.I16x16.__init__
gui.paths.I22x22.__init__
gui.paths.I32x32.__init__
gui.paths.I48x48.__init__
gui.pipe_editor.Pipe_editor.__init__
gui.references.References.__init__
gui.relax_gui.Main.__init__
gui.relax_prompt.InterpClass.__init__
gui.relax_prompt.Prompt.__init__
gui.relax_prompt._Exit.__init__
gui.results_viewer.Results_viewer.__init__
gui.settings.Base_window.__init__
gui.settings.Free_file_format.__init__
gui.spin_viewer.containers.Container.__init__
gui.spin_viewer.containers.Molecule.__init__
gui.spin_viewer.containers.Residue.__init__
gui.spin_viewer.containers.Root.__init__
gui.spin_viewer.containers.Spin.__init__
gui.spin_viewer.frame.Spin_view_window.__init__
gui.spin_viewer.splitter.Tree_splitter.__init__
gui.spin_viewer.tree.Mol_res_spin_tree.__init__
gui.user_functions.User_functions.__init__
gui.user_functions.base.UF_base.__init__
gui.user_functions.base.UF_page.__init__
gui.user_functions.sys_info.Sys_info_page.__init__
gui.wizard.Wiz_page.__init__
gui.wizard.Wiz_window.__init__
info.Info_box.__init__
maths_fns.consistency_tests.Consistency.__init__
maths_fns.consistency_tests.Data.__init__
maths_fns.frame_order.Frame_order.__init__
maths_fns.jw_mapping.Data.__init__
maths_fns.jw_mapping.Mapping.__init__
maths_fns.mf.Data.__init__
maths_fns.mf.Mf.__init__
maths_fns.n_state_model.N_state_opt.__init__
minfx.base_classes.Conjugate_gradient.__init__
minfx.base_classes.Hessian_mods.__init__
minfx.base_classes.Line_search.__init__
minfx.base_classes.Min.__init__
minfx.base_classes.Trust_region.__init__
minfx.bfgs.Bfgs.__init__
minfx.cauchy_point.Cauchy_point.__init__
minfx.constraint_linear.Constraint_linear.__init__
minfx.coordinate_descent.Coordinate_descent.__init__
minfx.dogleg.Dogleg.__init__
minfx.errors.MinfxError.__init__
minfx.exact_trust_region.Exact_trust_region.__init__
minfx.fletcher_reeves_cg.Fletcher_reeves.__init__
minfx.hestenes_stiefel_cg.Hestenes_stiefel.__init__
minfx.levenberg_marquardt.Levenberg_marquardt.__init__
minfx.method_of_multipliers.Method_of_multipliers.__init__
minfx.ncg.Ncg.__init__
minfx.newton.Newton.__init__
minfx.polak_ribiere_cg.Polak_ribiere.__init__
minfx.polak_ribiere_plus_cg.Polak_ribiere_plus.__init__
minfx.scipy_subset.anneal._state.__init__
minfx.scipy_subset.anneal.base_schedule.__init__
minfx.simplex.Simplex.__init__
minfx.steepest_descent.Steepest_descent.__init__
minfx.steihaug_cg.Steihaug.__init__
multi.commands.Exit_command.__init__
multi.commands.Get_name_command.__init__
multi.commands.Set_processor_property_command.__init__
multi.mpi4py_processor.Mpi4py_processor.__init__
multi.multi_processor_base.Batched_result_command.__init__
multi.multi_processor_base.Exit_queue_result_command.__init__
multi.multi_processor_base.Multi_processor.__init__
multi.multi_processor_base.Result_queue.__init__
multi.multi_processor_base.Threaded_result_queue.__init__
multi.multi_processor_base.Too_few_slaves_exception.__init__
multi.processor.Application_callback.__init__
multi.processor.Capturing_exception.__init__
multi.processor.Null_result_command.__init__
multi.processor.Processor.__init__
multi.processor.Result.__init__
multi.processor.Result_command.__init__
multi.processor.Result_exception.__init__
multi.processor.Result_string.__init__
multi.processor.Slave_command.__init__
multi.processor_io.IO_filter.__init__
multi.processor_io.PrependStringIO.__init__
multi.uni_processor.Uni_processor.__init__
opendx.base_map.Base_Map.__init__
prompt.base_class.Basic_class.__init__
prompt.base_class.Exec_info.__init__
prompt.base_class.User_fn_class.__init__
prompt.doc_string.Strings.__init__
prompt.interpreter.Interpreter.__init__
prompt.tab_completion.Tab_completion.__init__
relax.Relax.__init__
relax.RelaxParser.__init__
relax_errors.BaseArgError.__init__
relax_errors.RelaxArgNotInListError.__init__
relax_errors.RelaxArgNotNoneError.__init__
relax_errors.RelaxDiffMolNumError.__init__
relax_errors.RelaxDiffResNumError.__init__
relax_errors.RelaxDiffSeqError.__init__
relax_errors.RelaxDiffSpinNumError.__init__
relax_errors.RelaxDirError.__init__
relax_errors.RelaxError.__init__
relax_errors.RelaxFault.__init__
relax_errors.RelaxFileEmptyError.__init__
relax_errors.RelaxFileError.__init__
relax_errors.RelaxFileOverwriteError.__init__
relax_errors.RelaxFromXMLNotEmptyError.__init__
relax_errors.RelaxFuncSetupError.__init__
relax_errors.RelaxImplementError.__init__
relax_errors.RelaxInfError.__init__
relax_errors.RelaxInvalidColourError.__init__
relax_errors.RelaxInvalidDataError.__init__
relax_errors.RelaxInvalidError.__init__
relax_errors.RelaxInvalidSeqError.__init__
relax_errors.RelaxLenError.__init__
relax_errors.RelaxMfError.__init__
relax_errors.RelaxMissingBinaryError.__init__
relax_errors.RelaxModelError.__init__
relax_errors.RelaxMolSelectDisallowError.__init__
relax_errors.RelaxNaNError.__init__
relax_errors.RelaxNoInPathError.__init__
relax_errors.RelaxNoModelError.__init__
relax_errors.RelaxNoModuleInstallError.__init__
relax_errors.RelaxNoPCSError.__init__
relax_errors.RelaxNoPdbChainError.__init__
relax_errors.RelaxNoPdbError.__init__
relax_errors.RelaxNoPipeError.__init__
relax_errors.RelaxNoRDCError.__init__
relax_errors.RelaxNoResError.__init__
relax_errors.RelaxNoRiError.__init__
relax_errors.RelaxNoSequenceError.__init__
relax_errors.RelaxNoSimError.__init__
relax_errors.RelaxNoSpectraError.__init__
relax_errors.RelaxNoSpinError.__init__
relax_errors.RelaxNoSpinSpecError.__init__
relax_errors.RelaxNoTensorError.__init__
relax_errors.RelaxNoValueError.__init__
relax_errors.RelaxNoVectorsError.__init__
relax_errors.RelaxNonExecError.__init__
relax_errors.RelaxNoneError.__init__
relax_errors.RelaxNucleusError.__init__
relax_errors.RelaxPCSError.__init__
relax_errors.RelaxParamSetError.__init__
relax_errors.RelaxPdbError.__init__
relax_errors.RelaxPdbLoadError.__init__
relax_errors.RelaxPipeError.__init__
relax_errors.RelaxProgError.__init__
relax_errors.RelaxProgFailError.__init__
relax_errors.RelaxProtonTypeError.__init__
relax_errors.RelaxRDCError.__init__
relax_errors.RelaxRegExpError.__init__
relax_errors.RelaxResSelectDisallowError.__init__
relax_errors.RelaxRiError.__init__
relax_errors.RelaxSequenceError.__init__
relax_errors.RelaxSpectraError.__init__
relax_errors.RelaxSpinSelectDisallowError.__init__
relax_errors.RelaxSpinTypeError.__init__
relax_errors.RelaxSpinsNotLoadedError.__init__
relax_errors.RelaxStyleError.__init__
relax_errors.RelaxTensorError.__init__
relax_errors.RelaxUnknownDataTypeError.__init__
relax_errors.RelaxUnknownParamCombError.__init__
relax_errors.RelaxUnknownParamError.__init__
relax_errors.RelaxValueError.__init__
relax_io.DummyFileObject.__init__
relax_io.SplitIO.__init__
relax_warnings.RelaxDeselectWarning.__init__
relax_warnings.RelaxFileEmptyWarning.__init__
relax_warnings.RelaxNoAtomWarning.__init__
relax_warnings.RelaxNoPDBFileWarning.__init__
relax_warnings.RelaxNoSpinWarning.__init__
relax_warnings.RelaxWarning.__init__
relax_warnings.RelaxZeroVectorWarning.__init__
script-sconstruct.Main.__init__
specific_fns.consistency_tests.Consistency_tests.__init__
specific_fns.frame_order.Frame_order.__init__
specific_fns.jw_mapping.Jw_mapping.__init__
specific_fns.model_free.Model_free.__init__
specific_fns.model_free.multi_processor_commands.MF_grid_command.__init__
specific_fns.model_free.multi_processor_commands.MF_memo.__init__
specific_fns.model_free.multi_processor_commands.MF_minimise_command.__init__
specific_fns.model_free.multi_processor_commands.MF_result_command.__init__
specific_fns.n_state_model.N_state_model.__init__
specific_fns.noe.Noe.__init__
specific_fns.relax_fit.Relax_fit.__init__
status.Auto_analysis.__init__
status.Exec_lock.__init__
status.Observer.__init__
status.Relax_lock.__init__
test_suite.gui_tests.base_classes.GuiTestCase.__init__
test_suite.system_tests.base_classes.SystemTestCase.__init__
test_suite.system_tests.bmrb.Bmrb.__init__
test_suite.system_tests.frame_order.Frame_order.__init__
test_suite.system_tests.structure.Structure.__init__
test_suite.system_tests.unit_vectors.Unit_vectors.__init__
test_suite.test_suite_runner.Test_suite_runner.__init__
test_suite.unit_tests._generic_fns._structure.test_api_base.Test_api_base.__init__
test_suite.unit_tests._generic_fns._structure.test_scientific.Test_scientific.__init__
test_suite.unit_tests._maths_fns.test_frame_order_matrix_ops.Test_frame_order_matrix_ops.__init__
test_suite.unit_tests.unit_test_runner.ImportErrorTestCase.__init__
test_suite.unit_tests.unit_test_runner.Test_finder.__init__
test_suite.unit_tests.unit_test_runner.Unit_test_runner.__init__" class="py-name" href="#" onclick="return doclink('link-384', '__init__', 'link-384');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-385" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-385', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">atoms</tt><tt class="py-op">,</tt> <tt id="link-386" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-386', 'number', 'link-240');">number</a></tt><tt class="py-op">)</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">atoms</tt><tt class="py-op">:</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-387" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-387', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'O2*'</tt> <tt class="py-keyword">or</tt> <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-388', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"O2'"</tt><tt class="py-op">:</tt> <tt class="py-comment"># Ribose</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-389" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-389', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'R'</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-390', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
</div><a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.isCompatible"></a><div id="NucleotideResidue.isCompatible-def"><a name="L819"></a><tt class="py-lineno"> 819</tt> <a class="py-toggle" href="#" id="NucleotideResidue.isCompatible-toggle" onclick="return toggle('NucleotideResidue.isCompatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#isCompatible">isCompatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.isCompatible-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.isCompatible-expanded"><a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-391" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-391', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">or</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">                <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pdbname</tt><tt class="py-op">)</tt> \ </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">               <tt class="py-keyword">and</tt> <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-392', 'number', 'link-240');">number</a></tt> </tt>
</div><a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.addAtom"></a><div id="NucleotideResidue.addAtom-def"><a name="L824"></a><tt class="py-lineno"> 824</tt> <a class="py-toggle" href="#" id="NucleotideResidue.addAtom-toggle" onclick="return toggle('NucleotideResidue.addAtom');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#addAtom">addAtom</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.addAtom-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.addAtom-expanded"><a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">        <tt id="link-393" class="py-name"><a title="extern.scientific_python.IO.PDB.Residue
gui.spin_viewer.containers.Residue
gui.user_functions.residue.Residue
prompt.residue.Residue" class="py-name" href="#" onclick="return doclink('link-393', 'Residue', 'link-352');">Residue</a></tt><tt class="py-op">.</tt><tt id="link-394" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.addAtom
extern.scientific_python.IO.PDB.Group.addAtom
extern.scientific_python.IO.PDB.NucleotideResidue.addAtom" class="py-name" href="#" onclick="return doclink('link-394', 'addAtom', 'link-353');">addAtom</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-395', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'O2*'</tt> <tt class="py-keyword">or</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-396', 'name', 'link-216');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"O2'"</tt><tt class="py-op">:</tt> <tt class="py-comment"># Ribose</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-397" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-397', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'R'</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-398', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
</div><a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.hasRibose"></a><div id="NucleotideResidue.hasRibose-def"><a name="L829"></a><tt class="py-lineno"> 829</tt> <a class="py-toggle" href="#" id="NucleotideResidue.hasRibose-toggle" onclick="return toggle('NucleotideResidue.hasRibose');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasRibose">hasRibose</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.hasRibose-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.hasRibose-expanded"><a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue has an atom named O2*</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'O2*'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">"O2'"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.hasDesoxyribose"></a><div id="NucleotideResidue.hasDesoxyribose-def"><a name="L836"></a><tt class="py-lineno"> 836</tt> <a class="py-toggle" href="#" id="NucleotideResidue.hasDesoxyribose-toggle" onclick="return toggle('NucleotideResidue.hasDesoxyribose');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasDesoxyribose">hasDesoxyribose</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.hasDesoxyribose-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.hasDesoxyribose-expanded"><a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue has no atom named O2*</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-399" class="py-name" targets="Method extern.scientific_python.IO.PDB.NucleotideResidue.hasRibose()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasRibose"><a title="extern.scientific_python.IO.PDB.NucleotideResidue.hasRibose" class="py-name" href="#" onclick="return doclink('link-399', 'hasRibose', 'link-399');">hasRibose</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.hasPhosphate"></a><div id="NucleotideResidue.hasPhosphate-def"><a name="L843"></a><tt class="py-lineno"> 843</tt> <a class="py-toggle" href="#" id="NucleotideResidue.hasPhosphate-toggle" onclick="return toggle('NucleotideResidue.hasPhosphate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasPhosphate">hasPhosphate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.hasPhosphate-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.hasPhosphate-expanded"><a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue has a phosphate group</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'P'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.hasTerminalH"></a><div id="NucleotideResidue.hasTerminalH-def"><a name="L850"></a><tt class="py-lineno"> 850</tt> <a class="py-toggle" href="#" id="NucleotideResidue.hasTerminalH-toggle" onclick="return toggle('NucleotideResidue.hasTerminalH');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasTerminalH">hasTerminalH</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.hasTerminalH-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.hasTerminalH-expanded"><a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the residue has a 3-terminal H atom</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atoms</tt><tt class="py-op">.</tt><tt class="py-name">has_key</tt><tt class="py-op">(</tt><tt class="py-string">'H3T'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"> </tt>
<a name="NucleotideResidue.writeToFile"></a><div id="NucleotideResidue.writeToFile-def"><a name="L857"></a><tt class="py-lineno"> 857</tt> <a class="py-toggle" href="#" id="NucleotideResidue.writeToFile-toggle" onclick="return toggle('NucleotideResidue.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideResidue-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideResidue.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideResidue.writeToFile-expanded"><a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">        <tt id="link-400" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-400', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-401" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-401', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-402" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-402', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-403" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-403', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">            <tt id="link-404" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-404', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-405" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-405', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-406" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-406', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">            <tt id="link-407" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-407', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">        <tt class="py-name">terminus</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name" targets="Method extern.scientific_python.IO.PDB.NucleotideResidue.hasPhosphate()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasPhosphate"><a title="extern.scientific_python.IO.PDB.NucleotideResidue.hasPhosphate" class="py-name" href="#" onclick="return doclink('link-408', 'hasPhosphate', 'link-408');">hasPhosphate</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-name">terminus</tt> <tt class="py-op">=</tt> <tt class="py-string">'5'</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">        <tt id="link-409" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-409', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-410" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.nextResidue" class="py-name" href="#" onclick="return doclink('link-410', 'nextResidue', 'link-342');">nextResidue</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-411" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-411', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-412', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">terminus</tt><tt class="py-op">)</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">a</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">atom_list</tt><tt class="py-op">:</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">            <tt class="py-name">a</tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile
extern.scientific_python.IO.PDB.Atom.writeToFile
extern.scientific_python.IO.PDB.Chain.writeToFile
extern.scientific_python.IO.PDB.Group.writeToFile
extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile
extern.scientific_python.IO.PDB.Structure.writeToFile" class="py-name" href="#" onclick="return doclink('link-413', 'writeToFile', 'link-345');">writeToFile</a></tt><tt class="py-op">(</tt><tt id="link-414" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-414', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-415" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-415', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">            <tt id="link-416" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-416', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-417', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line"> </tt>
<a name="Chain"></a><div id="Chain-def"><a name="L870"></a><tt class="py-lineno"> 870</tt> <a class="py-toggle" href="#" id="Chain-toggle" onclick="return toggle('Chain');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html">Chain</a><tt class="py-op">:</tt> </tt>
</div><div id="Chain-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Chain-expanded"><a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">    <tt class="py-docstring">"""Chain of PDB residues</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">    This is an abstract base class. Instances can be created using</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring">    one of the subclasses (L{PeptideChain}, L{NucleotideChain}).</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring">    Chain objects respond to len() and return their residues</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">    by indexing with integers.</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"> </tt>
<a name="Chain.__init__"></a><div id="Chain.__init__-def"><a name="L881"></a><tt class="py-lineno"> 881</tt> <a class="py-toggle" href="#" id="Chain.__init__-toggle" onclick="return toggle('Chain.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residues</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">chain_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">segment_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.__init__-expanded"><a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">        @param residues: a list of residue objects, or C{None} meaning</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">                         that the chain is initially empty</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">        @type residues: C{list} or C{NoneType}</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring">        @param chain_id: a one-letter chain identifier or C{None}</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">        @type chain_id: C{str} or C{NoneType}</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">        @param segment_id: a multi-character segment identifier or C{None}</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">        @type segment_id: C{str} or C{NoneType}</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">residues</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt> <tt class="py-op">=</tt> <tt class="py-name">residues</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-op">=</tt> <tt class="py-name">chain_id</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-op">=</tt> <tt class="py-name">segment_id</tt> </tt>
</div><a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line"> </tt>
<a name="Chain.__len__"></a><div id="Chain.__len__-def"><a name="L898"></a><tt class="py-lineno"> 898</tt> <a class="py-toggle" href="#" id="Chain.__len__-toggle" onclick="return toggle('Chain.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.__len__-expanded"><a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the number of residues in the chain</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{int}</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">)</tt> </tt>
</div><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"> </tt>
<a name="Chain.sequence"></a><div id="Chain.sequence-def"><a name="L905"></a><tt class="py-lineno"> 905</tt> <a class="py-toggle" href="#" id="Chain.sequence-toggle" onclick="return toggle('Chain.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.sequence-expanded"><a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the list of residue names</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{list} of C{str}</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">r</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-418', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">]</tt> </tt>
</div><a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line"> </tt>
<a name="Chain.__getitem__"></a><div id="Chain.__getitem__-def"><a name="L912"></a><tt class="py-lineno"> 912</tt> <a class="py-toggle" href="#" id="Chain.__getitem__-toggle" onclick="return toggle('Chain.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.__getitem__-expanded"><a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"><tt class="py-docstring">        @param index: an index into the chain</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line"><tt class="py-docstring">        @type index: C{int}</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the residue corresponding to the index</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{AminoAcidResidue} or L{NucleotideResidue}</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line"><tt class="py-docstring">        @raises IndexError: if index exceeds the chain length</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">index</tt><tt class="py-op">]</tt> </tt>
</div><a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line"> </tt>
<a name="Chain.__getslice__"></a><div id="Chain.__getslice__-def"><a name="L922"></a><tt class="py-lineno"> 922</tt> <a class="py-toggle" href="#" id="Chain.__getslice__-toggle" onclick="return toggle('Chain.__getslice__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#__getslice__">__getslice__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">i1</tt><tt class="py-op">,</tt> <tt class="py-param">i2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.__getslice__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.__getslice__-expanded"><a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line"><tt class="py-docstring">        @param i1: in index into the chain</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line"><tt class="py-docstring">        @type i1: C{int}</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line"><tt class="py-docstring">        @param i2: in index into the chain</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line"><tt class="py-docstring">        @type i12 C{int}</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: the subchain from i1 to i2</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: L{PeptideChain} or L{NucleotideChain}</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">i1</tt><tt class="py-op">:</tt><tt class="py-name">i2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line"> </tt>
<a name="Chain.addResidue"></a><div id="Chain.addResidue-def"><a name="L933"></a><tt class="py-lineno"> 933</tt> <a class="py-toggle" href="#" id="Chain.addResidue-toggle" onclick="return toggle('Chain.addResidue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#addResidue">addResidue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.addResidue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.addResidue-expanded"><a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line"><tt class="py-docstring">        Add a residue at the end of the chain</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"><tt class="py-docstring">        @param residue: the residue to be added</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"><tt class="py-docstring">        @type residue: L{AminoAcidResidue} or L{NucleotideResidue}</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">.</tt><tt id="link-419" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-419', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt id="link-420" class="py-name" targets="Module gui.user_functions.residue=gui.user_functions.residue-module.html,Module prompt.residue=prompt.residue-module.html,Module test_suite.unit_tests._generic_fns.residue=test_suite.unit_tests._generic_fns.residue-module.html"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-420', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line"> </tt>
<a name="Chain.removeResidues"></a><div id="Chain.removeResidues-def"><a name="L942"></a><tt class="py-lineno"> 942</tt> <a class="py-toggle" href="#" id="Chain.removeResidues-toggle" onclick="return toggle('Chain.removeResidues');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#removeResidues">removeResidues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first</tt><tt class="py-op">,</tt> <tt class="py-param">last</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.removeResidues-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.removeResidues-expanded"><a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line"><tt class="py-docstring">        Remove residues in a given index range.</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"><tt class="py-docstring">        @param first: the index of the first residue to be removed</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"><tt class="py-docstring">        @type first: C{int}</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line"><tt class="py-docstring">        @param last: the index of the first residue to be kept, or C{None}</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line"><tt class="py-docstring">                     meaning remove everything to the end of the chain.</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"><tt class="py-docstring">        @type last: C{int} or C{NoneType}</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">last</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">first</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">first</tt><tt class="py-op">:</tt><tt class="py-name">last</tt><tt class="py-op">]</tt> </tt>
</div><a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line"> </tt>
<a name="Chain.deleteHydrogens"></a><div id="Chain.deleteHydrogens-def"><a name="L957"></a><tt class="py-lineno"> 957</tt> <a class="py-toggle" href="#" id="Chain.deleteHydrogens-toggle" onclick="return toggle('Chain.deleteHydrogens');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#deleteHydrogens">deleteHydrogens</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.deleteHydrogens-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.deleteHydrogens-expanded"><a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line"><tt class="py-docstring">        Remove all hydrogen atoms in the chain</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">:</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-name">r</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name" targets="Method extern.scientific_python.IO.PDB.Chain.deleteHydrogens()=extern.scientific_python.IO.PDB.Chain-class.html#deleteHydrogens,Method extern.scientific_python.IO.PDB.Group.deleteHydrogens()=extern.scientific_python.IO.PDB.Group-class.html#deleteHydrogens,Method extern.scientific_python.IO.PDB.Structure.deleteHydrogens()=extern.scientific_python.IO.PDB.Structure-class.html#deleteHydrogens"><a title="extern.scientific_python.IO.PDB.Chain.deleteHydrogens
extern.scientific_python.IO.PDB.Group.deleteHydrogens
extern.scientific_python.IO.PDB.Structure.deleteHydrogens" class="py-name" href="#" onclick="return doclink('link-421', 'deleteHydrogens', 'link-421');">deleteHydrogens</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"> </tt>
<a name="Chain.writeToFile"></a><div id="Chain.writeToFile-def"><a name="L964"></a><tt class="py-lineno"> 964</tt> <a class="py-toggle" href="#" id="Chain.writeToFile-toggle" onclick="return toggle('Chain.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Chain-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Chain.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Chain.writeToFile-expanded"><a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"><tt class="py-docstring">        Write the chain to a file</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring">        @param file: a PDBFile object or a file name</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring">        @type file: L{PDBFile} or C{str}</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">        <tt id="link-422" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-422', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-423" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-423', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-424" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-424', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-425" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-425', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">            <tt id="link-426" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-426', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-427" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-427', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-428" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-428', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">            <tt id="link-429" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-429', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">        <tt id="link-430" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-430', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-431" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.nextChain()=extern.scientific_python.IO.PDB.PDBFile-class.html#nextChain"><a title="extern.scientific_python.IO.PDB.PDBFile.nextChain" class="py-name" href="#" onclick="return doclink('link-431', 'nextChain', 'link-431');">nextChain</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt><tt class="py-op">)</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">:</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">            <tt class="py-name">r</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile
extern.scientific_python.IO.PDB.Atom.writeToFile
extern.scientific_python.IO.PDB.Chain.writeToFile
extern.scientific_python.IO.PDB.Group.writeToFile
extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile
extern.scientific_python.IO.PDB.Structure.writeToFile" class="py-name" href="#" onclick="return doclink('link-432', 'writeToFile', 'link-345');">writeToFile</a></tt><tt class="py-op">(</tt><tt id="link-433" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-433', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">        <tt id="link-434" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-434', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-435" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.terminateChain" class="py-name" href="#" onclick="return doclink('link-435', 'terminateChain', 'link-270');">terminateChain</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-436" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-436', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">            <tt id="link-437" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-437', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-438', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"> </tt>
<a name="PeptideChain"></a><div id="PeptideChain-def"><a name="L982"></a><tt class="py-lineno"> 982</tt> <a class="py-toggle" href="#" id="PeptideChain-toggle" onclick="return toggle('PeptideChain');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PeptideChain-class.html">PeptideChain</a><tt class="py-op">(</tt><tt class="py-base-class">Chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PeptideChain-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PeptideChain-expanded"><a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring">    Peptide chain in a PDB file</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"> </tt>
<a name="PeptideChain.isTerminated"></a><div id="PeptideChain.isTerminated-def"><a name="L988"></a><tt class="py-lineno"> 988</tt> <a class="py-toggle" href="#" id="PeptideChain.isTerminated-toggle" onclick="return toggle('PeptideChain.isTerminated');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PeptideChain-class.html#isTerminated">isTerminated</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PeptideChain.isTerminated-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PeptideChain.isTerminated-expanded"><a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the last residue is in C-terminal configuration</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.isCTerminus" class="py-name" href="#" onclick="return doclink('link-439', 'isCTerminus', 'link-363');">isCTerminus</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line"> </tt>
<a name="PeptideChain.isCompatible"></a><div id="PeptideChain.isCompatible-def"><a name="L995"></a><tt class="py-lineno"> 995</tt> <a class="py-toggle" href="#" id="PeptideChain.isCompatible-toggle" onclick="return toggle('PeptideChain.isCompatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.PeptideChain-class.html#isCompatible">isCompatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain_data</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PeptideChain.isCompatible-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PeptideChain.isCompatible-expanded"><a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">               <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">               <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt id="link-440" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-440', 'amino_acids', 'link-28');">amino_acids</a></tt> </tt>
</div></div><a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line"> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line"> </tt>
<a name="NucleotideChain"></a><div id="NucleotideChain-def"><a name="L1001"></a><tt class="py-lineno">1001</tt> <a class="py-toggle" href="#" id="NucleotideChain-toggle" onclick="return toggle('NucleotideChain');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideChain-class.html">NucleotideChain</a><tt class="py-op">(</tt><tt class="py-base-class">Chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideChain-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="NucleotideChain-expanded"><a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line"> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line"><tt class="py-docstring">    Nucleotide chain in a PDB file</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"> </tt>
<a name="NucleotideChain.isTerminated"></a><div id="NucleotideChain.isTerminated-def"><a name="L1007"></a><tt class="py-lineno">1007</tt> <a class="py-toggle" href="#" id="NucleotideChain.isTerminated-toggle" onclick="return toggle('NucleotideChain.isTerminated');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideChain-class.html#isTerminated">isTerminated</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideChain.isTerminated-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideChain.isTerminated-expanded"><a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line"><tt class="py-docstring">        @returns: C{True} if the last residue is in 3-terminal configuration</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line"><tt class="py-docstring">        @rtype: C{bool}</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line"><tt class="py-docstring">        @note: There is no way to perform this test with standard PDB files.</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring">               The algorithm used works for certain non-standard files only.</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">               <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-441', 'name', 'link-216');">name</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'3'</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name" targets="Method extern.scientific_python.IO.PDB.NucleotideResidue.hasTerminalH()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#hasTerminalH"><a title="extern.scientific_python.IO.PDB.NucleotideResidue.hasTerminalH" class="py-name" href="#" onclick="return doclink('link-442', 'hasTerminalH', 'link-442');">hasTerminalH</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"> </tt>
<a name="NucleotideChain.isCompatible"></a><div id="NucleotideChain.isCompatible-def"><a name="L1018"></a><tt class="py-lineno">1018</tt> <a class="py-toggle" href="#" id="NucleotideChain.isCompatible-toggle" onclick="return toggle('NucleotideChain.isCompatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.NucleotideChain-class.html#isCompatible">isCompatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain_data</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="NucleotideChain.isCompatible-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="NucleotideChain.isCompatible-expanded"><a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">               <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">               <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt id="link-443" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-443', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt> </tt>
</div></div><a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"> </tt>
<a name="DummyChain"></a><div id="DummyChain-def"><a name="L1023"></a><tt class="py-lineno">1023</tt> <a class="py-toggle" href="#" id="DummyChain-toggle" onclick="return toggle('DummyChain');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.DummyChain-class.html">DummyChain</a><tt class="py-op">(</tt><tt class="py-base-class">Chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DummyChain-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DummyChain-expanded"><a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"> </tt>
<a name="DummyChain.__init__"></a><div id="DummyChain.__init__-def"><a name="L1025"></a><tt class="py-lineno">1025</tt> <a class="py-toggle" href="#" id="DummyChain.__init__-toggle" onclick="return toggle('DummyChain.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.DummyChain-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">structure</tt><tt class="py-op">,</tt> <tt class="py-param">chain_id</tt><tt class="py-op">,</tt> <tt class="py-param">segment_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DummyChain.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DummyChain.__init__-expanded"><a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-444" class="py-name" targets="Package bmrblib.structure=bmrblib.structure-module.html,Package generic_fns.structure=generic_fns.structure-module.html,Module gui.user_functions.structure=gui.user_functions.structure-module.html,Module prompt.structure=prompt.structure-module.html,Module test_suite.system_tests.structure=test_suite.system_tests.structure-module.html"><a title="bmrblib.structure
generic_fns.structure
gui.user_functions.structure
prompt.structure
test_suite.system_tests.structure" class="py-name" href="#" onclick="return doclink('link-444', 'structure', 'link-444');">structure</a></tt> <tt class="py-op">=</tt> <tt id="link-445" class="py-name"><a title="bmrblib.structure
generic_fns.structure
gui.user_functions.structure
prompt.structure
test_suite.system_tests.structure" class="py-name" href="#" onclick="return doclink('link-445', 'structure', 'link-444');">structure</a></tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-op">=</tt> <tt class="py-name">chain_id</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-op">=</tt> <tt class="py-name">segment_id</tt> </tt>
</div><a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line"> </tt>
<a name="DummyChain.isTerminated"></a><div id="DummyChain.isTerminated-def"><a name="L1030"></a><tt class="py-lineno">1030</tt> <a class="py-toggle" href="#" id="DummyChain.isTerminated-toggle" onclick="return toggle('DummyChain.isTerminated');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.DummyChain-class.html#isTerminated">isTerminated</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DummyChain.isTerminated-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DummyChain.isTerminated-expanded"><a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line"> </tt>
<a name="DummyChain.addResidue"></a><div id="DummyChain.addResidue-def"><a name="L1033"></a><tt class="py-lineno">1033</tt> <a class="py-toggle" href="#" id="DummyChain.addResidue-toggle" onclick="return toggle('DummyChain.addResidue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.DummyChain-class.html#addResidue">addResidue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DummyChain.addResidue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DummyChain.addResidue-expanded"><a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-446" class="py-name"><a title="bmrblib.structure
generic_fns.structure
gui.user_functions.structure
prompt.structure
test_suite.system_tests.structure" class="py-name" href="#" onclick="return doclink('link-446', 'structure', 'link-444');">structure</a></tt><tt class="py-op">.</tt><tt id="link-447" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.addMolecule()=extern.scientific_python.IO.PDB.Structure-class.html#addMolecule"><a title="extern.scientific_python.IO.PDB.Structure.addMolecule" class="py-name" href="#" onclick="return doclink('link-447', 'addMolecule', 'link-447');">addMolecule</a></tt><tt class="py-op">(</tt><tt id="link-448" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-448', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line"> </tt>
<a name="DummyChain.isCompatible"></a><div id="DummyChain.isCompatible-def"><a name="L1036"></a><tt class="py-lineno">1036</tt> <a class="py-toggle" href="#" id="DummyChain.isCompatible-toggle" onclick="return toggle('DummyChain.isCompatible');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.DummyChain-class.html#isCompatible">isCompatible</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">chain_data</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DummyChain.isCompatible-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DummyChain.isCompatible-expanded"><a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">               <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">               <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-449" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-449', 'amino_acids', 'link-28');">amino_acids</a></tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">               <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-450" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-450', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt> </tt>
</div></div><a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"><tt class="py-comment"># Residue number class for dealing with insertion codes</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="ResidueNumber"></a><div id="ResidueNumber-def"><a name="L1045"></a><tt class="py-lineno">1045</tt> <a class="py-toggle" href="#" id="ResidueNumber-toggle" onclick="return toggle('ResidueNumber');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.ResidueNumber-class.html">ResidueNumber</a><tt class="py-op">:</tt> </tt>
</div><div id="ResidueNumber-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ResidueNumber-expanded"><a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line"> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line"><tt class="py-docstring">    PDB residue number</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line"><tt class="py-docstring">    Most PDB residue numbers are simple integers, but when insertion</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line"><tt class="py-docstring">    codes are used a number can consist of an integer plus a letter.</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line"><tt class="py-docstring">    Such compound residue numbers are represented by this class.</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line"> </tt>
<a name="ResidueNumber.__init__"></a><div id="ResidueNumber.__init__-def"><a name="L1055"></a><tt class="py-lineno">1055</tt> <a class="py-toggle" href="#" id="ResidueNumber.__init__-toggle" onclick="return toggle('ResidueNumber.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.ResidueNumber-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt><tt class="py-op">,</tt> <tt class="py-param">insertion_code</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueNumber.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueNumber.__init__-expanded"><a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: the numeric part of the residue number</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: C{int}</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"><tt class="py-docstring">        @param insertion_code: the letter part of the residue number</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"><tt class="py-docstring">        @type insertion_code: C{str}</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-451', 'number', 'link-240');">number</a></tt> <tt class="py-op">=</tt> <tt id="link-452" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-452', 'number', 'link-240');">number</a></tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">insertion_code</tt> <tt class="py-op">=</tt> <tt class="py-name">insertion_code</tt> </tt>
</div><a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"> </tt>
<a name="ResidueNumber.__cmp__"></a><div id="ResidueNumber.__cmp__-def"><a name="L1065"></a><tt class="py-lineno">1065</tt> <a class="py-toggle" href="#" id="ResidueNumber.__cmp__-toggle" onclick="return toggle('ResidueNumber.__cmp__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.ResidueNumber-class.html#__cmp__">__cmp__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueNumber.__cmp__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueNumber.__cmp__-expanded"><a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">other</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-453', 'number', 'link-240');">number</a></tt> <tt class="py-op">==</tt> <tt class="py-name">other</tt><tt class="py-op">:</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-number">1</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">cmp</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-454" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-454', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">other</tt><tt class="py-op">)</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-455', 'number', 'link-240');">number</a></tt> <tt class="py-op">==</tt> <tt class="py-name">other</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-456', 'number', 'link-240');">number</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">cmp</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">insertion_code</tt><tt class="py-op">,</tt> <tt class="py-name">other</tt><tt class="py-op">.</tt><tt class="py-name">insertion_code</tt><tt class="py-op">)</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">cmp</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-457', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">other</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-458', 'number', 'link-240');">number</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line"> </tt>
<a name="ResidueNumber.__str__"></a><div id="ResidueNumber.__str__-def"><a name="L1076"></a><tt class="py-lineno">1076</tt> <a class="py-toggle" href="#" id="ResidueNumber.__str__-toggle" onclick="return toggle('ResidueNumber.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.ResidueNumber-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ResidueNumber.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="ResidueNumber.__str__-expanded"><a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-459" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-459', 'number', 'link-240');">number</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">insertion_code</tt> </tt>
</div><a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">    <tt id="link-460" class="py-name"><a title="bmrblib.pystarlib.Utils.Lister.__repr__
data.Relax_data_store.__repr__
data.align_tensor.AlignTensorList.__repr__
data.data_classes.Element.__repr__
data.mol_res_spin.MoleculeContainer.__repr__
data.mol_res_spin.MoleculeList.__repr__
data.mol_res_spin.ResidueContainer.__repr__
data.mol_res_spin.ResidueList.__repr__
data.mol_res_spin.SpinContainer.__repr__
data.mol_res_spin.SpinList.__repr__
data.pipe_container.PipeContainer.__repr__
extern.scientific_python.IO.PDB.Structure.__repr__
generic_fns.structure.api_base.ModelContainer.__repr__
generic_fns.structure.api_base.ModelList.__repr__
generic_fns.structure.api_base.MolList.__repr__
gui.relax_prompt._Exit.__repr__
prompt.command.Lh.__repr__
prompt.command.Ll.__repr__
prompt.command.Ls.__repr__
prompt.gpl.GPL.__repr__
prompt.help._Helper.__repr__
prompt.help._Helper_python.__repr__
prompt.interpreter._Exit.__repr__" class="py-name" href="#" onclick="return doclink('link-460', '__repr__', 'link-294');">__repr__</a></tt> <tt class="py-op">=</tt> <tt id="link-461" class="py-name"><a title="extern.scientific_python.IO.FortranFormat.FortranLine.__str__
extern.scientific_python.IO.PDB.Atom.__str__
extern.scientific_python.IO.PDB.Group.__str__
extern.scientific_python.IO.PDB.ResidueNumber.__str__
generic_fns.spectrum.Bruker_import.__str__
minfx.errors.BaseError.__str__
multi.processor.Capturing_exception.__str__
relax_errors.BaseError.__str__
relax_errors.RelaxFault.__str__
relax_warnings.BaseWarning.__str__" class="py-name" href="#" onclick="return doclink('link-461', '__str__', 'link-295');">__str__</a></tt> </tt>
</div><a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line"> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line"><tt class="py-comment"># The configuration class.</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="Structure"></a><div id="Structure-def"><a name="L1083"></a><tt class="py-lineno">1083</tt> <a class="py-toggle" href="#" id="Structure-toggle" onclick="return toggle('Structure');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html">Structure</a><tt class="py-op">:</tt> </tt>
</div><div id="Structure-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Structure-expanded"><a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line"> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">    <tt class="py-docstring">"""</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"><tt class="py-docstring">    A high-level representation of the contents of a PDB file</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"><tt class="py-docstring">    The components of a structure can be accessed in several ways</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"><tt class="py-docstring">    ('s' is an instance of this class):</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.residues' is a list of all PDB residues, in the order in</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line"><tt class="py-docstring">       which they occurred in the file.</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.peptide_chains' is a list of PeptideChain objects, containing</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"><tt class="py-docstring">       all peptide chains in the file in their original order.</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.nucleotide_chains' is a list of NucleotideChain objects, containing</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line"><tt class="py-docstring">       all nucleotide chains in the file in their original order.</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.molecules' is a list of all PDB residues that are neither</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line"><tt class="py-docstring">       amino acid residues nor nucleotide residues, in their original</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line"><tt class="py-docstring">       order.</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.objects' is a list of all high-level objects (peptide chains,</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line"><tt class="py-docstring">       nucleotide chains, and molecules) in their original order.</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.to_fractional' is the transformation from real-space coordinates</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line"><tt class="py-docstring">       to fractional coordinates, as read from the SCALEn records.</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.from_fractional' is the transformation from fractional coordinates</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line"><tt class="py-docstring">       to real-space coordinates, the inverse of s.to_fractional.</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line"><tt class="py-docstring">     - 's.ncs_transformations' is a list of transformations that</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line"><tt class="py-docstring">        describe non-crystallographic symmetries, as read from the</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line"><tt class="py-docstring">        MTRIXn records.</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line"><tt class="py-docstring">     - if a CRYST1 record exists, 's.a', 's.b', 's.c', 's.alpha',</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line"><tt class="py-docstring">       's.beta', 's.gamma' are the parameters of the unit cell and</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line"><tt class="py-docstring">       's.space_group' is a string indicating the space group.</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line"><tt class="py-docstring">       If no CRYST1 record exists, all those values are None.</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line"><tt class="py-docstring">       Furthermore, 's.cs_transformations' is a list of transformations</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line"><tt class="py-docstring">       that describe crystallographic symmetries. If no CRYST1 record</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"><tt class="py-docstring">       exists, the list is empty.</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"><tt class="py-docstring">    An iteration over a Structure instance by a for-loop is equivalent</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line"><tt class="py-docstring">    to an iteration over the residue list.</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line"> </tt>
<a name="Structure.__init__"></a><div id="Structure.__init__-def"><a name="L1129"></a><tt class="py-lineno">1129</tt> <a class="py-toggle" href="#" id="Structure.__init__-toggle" onclick="return toggle('Structure.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file_or_filename</tt><tt class="py-op">,</tt> <tt class="py-param">model</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">alternate_code</tt> <tt class="py-op">=</tt> <tt class="py-string">'A'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.__init__-expanded"><a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line"><tt class="py-docstring">        @param file_or_filename: the name of the PDB file, or a file object.</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line"><tt class="py-docstring">                                 Compressed files and URLs are accepted,</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line"><tt class="py-docstring">                                 as for class L{PDBFile}.</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"><tt class="py-docstring">        @type file_or_filename: C{str} or C{file}</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line"><tt class="py-docstring">        @param model: the number of the model to read from a multiple-model</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"><tt class="py-docstring">                      file. Only one model can be treated at a time.</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line"><tt class="py-docstring">        @type model: C{int}</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"><tt class="py-docstring">        @param alternate_code: the version of the positions to be read</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line"><tt class="py-docstring">                               from a file with alternate positions.</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line"><tt class="py-docstring">        @type alternate_code: single-letter C{str}</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">file_or_filename</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-name">file_or_filename</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">filename</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt> <tt class="py-op">=</tt> <tt class="py-name">model</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alternate</tt> <tt class="py-op">=</tt> <tt class="py-name">alternate_code</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pdb_code</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">to_fractional</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">from_fractional</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ncs_transformations</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cs_transformations</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">a</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">b</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alpha</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">beta</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">gamma</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">space_group</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.parseFile()=extern.scientific_python.IO.PDB.Structure-class.html#parseFile"><a title="extern.scientific_python.IO.PDB.Structure.parseFile" class="py-name" href="#" onclick="return doclink('link-462', 'parseFile', 'link-462');">parseFile</a></tt><tt class="py-op">(</tt><tt id="link-463" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-463', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt class="py-name">file_or_filename</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-464" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.findSpaceGroupTransformations()=extern.scientific_python.IO.PDB.Structure-class.html#findSpaceGroupTransformations"><a title="extern.scientific_python.IO.PDB.Structure.findSpaceGroupTransformations" class="py-name" href="#" onclick="return doclink('link-464', 'findSpaceGroupTransformations', 'link-464');">findSpaceGroupTransformations</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line"> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">    <tt class="py-name">peptide_chain_constructor</tt> <tt class="py-op">=</tt> <tt id="link-465" class="py-name" targets="Class extern.scientific_python.IO.PDB.PeptideChain=extern.scientific_python.IO.PDB.PeptideChain-class.html"><a title="extern.scientific_python.IO.PDB.PeptideChain" class="py-name" href="#" onclick="return doclink('link-465', 'PeptideChain', 'link-465');">PeptideChain</a></tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">    <tt class="py-name">nucleotide_chain_constructor</tt> <tt class="py-op">=</tt> <tt id="link-466" class="py-name" targets="Class extern.scientific_python.IO.PDB.NucleotideChain=extern.scientific_python.IO.PDB.NucleotideChain-class.html"><a title="extern.scientific_python.IO.PDB.NucleotideChain" class="py-name" href="#" onclick="return doclink('link-466', 'NucleotideChain', 'link-466');">NucleotideChain</a></tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">    <tt class="py-name">molecule_constructor</tt> <tt class="py-op">=</tt> <tt id="link-467" class="py-name" targets="Class extern.scientific_python.IO.PDB.Molecule=extern.scientific_python.IO.PDB.Molecule-class.html,Class gui.spin_viewer.containers.Molecule=gui.spin_viewer.containers.Molecule-class.html,Class gui.user_functions.molecule.Molecule=gui.user_functions.molecule.Molecule-class.html,Class prompt.molecule.Molecule=prompt.molecule.Molecule-class.html"><a title="extern.scientific_python.IO.PDB.Molecule
gui.spin_viewer.containers.Molecule
gui.user_functions.molecule.Molecule
prompt.molecule.Molecule" class="py-name" href="#" onclick="return doclink('link-467', 'Molecule', 'link-467');">Molecule</a></tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"> </tt>
<a name="Structure.__len__"></a><div id="Structure.__len__-def"><a name="L1167"></a><tt class="py-lineno">1167</tt> <a class="py-toggle" href="#" id="Structure.__len__-toggle" onclick="return toggle('Structure.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.__len__-expanded"><a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line"> </tt>
<a name="Structure.__getitem__"></a><div id="Structure.__getitem__-def"><a name="L1170"></a><tt class="py-lineno">1170</tt> <a class="py-toggle" href="#" id="Structure.__getitem__-toggle" onclick="return toggle('Structure.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">item</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.__getitem__-expanded"><a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">item</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line"> </tt>
<a name="Structure.deleteHydrogens"></a><div id="Structure.deleteHydrogens-def"><a name="L1173"></a><tt class="py-lineno">1173</tt> <a class="py-toggle" href="#" id="Structure.deleteHydrogens-toggle" onclick="return toggle('Structure.deleteHydrogens');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#deleteHydrogens">deleteHydrogens</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.deleteHydrogens-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.deleteHydrogens-expanded"><a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line"><tt class="py-docstring">        Remove all hydrogen atoms</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">r</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">:</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">            <tt class="py-name">r</tt><tt class="py-op">.</tt><tt id="link-468" class="py-name"><a title="extern.scientific_python.IO.PDB.Chain.deleteHydrogens
extern.scientific_python.IO.PDB.Group.deleteHydrogens
extern.scientific_python.IO.PDB.Structure.deleteHydrogens" class="py-name" href="#" onclick="return doclink('link-468', 'deleteHydrogens', 'link-421');">deleteHydrogens</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line"> </tt>
<a name="Structure.splitPeptideChain"></a><div id="Structure.splitPeptideChain-def"><a name="L1180"></a><tt class="py-lineno">1180</tt> <a class="py-toggle" href="#" id="Structure.splitPeptideChain-toggle" onclick="return toggle('Structure.splitPeptideChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#splitPeptideChain">splitPeptideChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt><tt class="py-op">,</tt> <tt class="py-param">position</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.splitPeptideChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.splitPeptideChain-expanded"><a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line"><tt class="py-docstring">        Split a peptide chain into two chains</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line"><tt class="py-docstring">        The two chain fragments remain adjacent in the peptide chain</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line"><tt class="py-docstring">        list, i.e. the numbers of all following chains increase</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"><tt class="py-docstring">        by one.</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: the number of the peptide chain to be split</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: C{int}</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line"><tt class="py-docstring">        @param position: the residue index at which the chain is split.</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"><tt class="py-docstring">        @type position: C{int}</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-469" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure._splitChain()=extern.scientific_python.IO.PDB.Structure-class.html#_splitChain"><a title="extern.scientific_python.IO.PDB.Structure._splitChain" class="py-name" href="#" onclick="return doclink('link-469', '_splitChain', 'link-469');">_splitChain</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chain_constructor</tt><tt class="py-op">,</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt><tt class="py-op">,</tt> <tt id="link-470" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-470', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">position</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">         </tt>
<a name="Structure.splitNucleotideChain"></a><div id="Structure.splitNucleotideChain-def"><a name="L1196"></a><tt class="py-lineno">1196</tt> <a class="py-toggle" href="#" id="Structure.splitNucleotideChain-toggle" onclick="return toggle('Structure.splitNucleotideChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#splitNucleotideChain">splitNucleotideChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt><tt class="py-op">,</tt> <tt class="py-param">position</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.splitNucleotideChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.splitNucleotideChain-expanded"><a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"><tt class="py-docstring">        Split a nucleotide chain into two chains</tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line"><tt class="py-docstring">        The two chain fragments remain adjacent in the nucleotide chain</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line"><tt class="py-docstring">        list, i.e. the numbers of all following chains increase</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line"><tt class="py-docstring">        by one.</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line"><tt class="py-docstring">        @param number: the number of the nucleotide chain to be split</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"><tt class="py-docstring">        @type number: C{int}</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line"><tt class="py-docstring">        @param position: the residue index at which the chain is split.</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line"><tt class="py-docstring">        @type position: C{int}</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name"><a title="extern.scientific_python.IO.PDB.Structure._splitChain" class="py-name" href="#" onclick="return doclink('link-471', '_splitChain', 'link-469');">_splitChain</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chain_constructor</tt><tt class="py-op">,</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">,</tt> <tt id="link-472" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-472', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">position</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line"> </tt>
<a name="Structure._splitChain"></a><div id="Structure._splitChain-def"><a name="L1212"></a><tt class="py-lineno">1212</tt> <a class="py-toggle" href="#" id="Structure._splitChain-toggle" onclick="return toggle('Structure._splitChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#_splitChain">_splitChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">constructor</tt><tt class="py-op">,</tt> <tt class="py-param">chain_list</tt><tt class="py-op">,</tt> <tt class="py-param">number</tt><tt class="py-op">,</tt> <tt class="py-param">position</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure._splitChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure._splitChain-expanded"><a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">        <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt id="link-473" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-473', 'number', 'link-240');">number</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">        <tt class="py-name">part1</tt> <tt class="py-op">=</tt> <tt class="py-name">constructor</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">position</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">                            <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt><tt class="py-op">,</tt> <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt><tt class="py-op">)</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-name">part2</tt> <tt class="py-op">=</tt> <tt class="py-name">constructor</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">[</tt><tt class="py-name">position</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">        <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt id="link-474" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-474', 'number', 'link-240');">number</a></tt><tt class="py-op">:</tt><tt id="link-475" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-475', 'number', 'link-240');">number</a></tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">part1</tt><tt class="py-op">,</tt> <tt class="py-name">part2</tt><tt class="py-op">]</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">        <tt class="py-name">index</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt class="py-name">index</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">[</tt><tt class="py-name">index</tt><tt class="py-op">:</tt><tt class="py-name">index</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">part1</tt><tt class="py-op">,</tt> <tt class="py-name">part2</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line"> </tt>
<a name="Structure.joinPeptideChains"></a><div id="Structure.joinPeptideChains-def"><a name="L1221"></a><tt class="py-lineno">1221</tt> <a class="py-toggle" href="#" id="Structure.joinPeptideChains-toggle" onclick="return toggle('Structure.joinPeptideChains');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#joinPeptideChains">joinPeptideChains</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first</tt><tt class="py-op">,</tt> <tt class="py-param">second</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.joinPeptideChains-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.joinPeptideChains-expanded"><a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line"><tt class="py-docstring">        Join two peptide chains into a single one. The new chain occupies</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line"><tt class="py-docstring">        the position of the first chain, the second one is removed from</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line"><tt class="py-docstring">        the peptide chain list.</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line"><tt class="py-docstring">        @param first: the number of the first chain</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line"><tt class="py-docstring">        @type first: C{int}</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line"><tt class="py-docstring">        @param second: the number of the second chain</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line"><tt class="py-docstring">        @type second: C{int}</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure._joinChains()=extern.scientific_python.IO.PDB.Structure-class.html#_joinChains"><a title="extern.scientific_python.IO.PDB.Structure._joinChains" class="py-name" href="#" onclick="return doclink('link-476', '_joinChains', 'link-476');">_joinChains</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chain_constructor</tt><tt class="py-op">,</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt><tt class="py-op">,</tt> <tt class="py-name">first</tt><tt class="py-op">,</tt> <tt class="py-name">second</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">         </tt>
<a name="Structure.joinNucleotideChains"></a><div id="Structure.joinNucleotideChains-def"><a name="L1235"></a><tt class="py-lineno">1235</tt> <a class="py-toggle" href="#" id="Structure.joinNucleotideChains-toggle" onclick="return toggle('Structure.joinNucleotideChains');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#joinNucleotideChains">joinNucleotideChains</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">first</tt><tt class="py-op">,</tt> <tt class="py-param">second</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.joinNucleotideChains-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.joinNucleotideChains-expanded"><a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line"><tt class="py-docstring">        Join two nucleotide chains into a single one. The new chain occupies</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line"><tt class="py-docstring">        the position of the first chain, the second one is removed from</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line"><tt class="py-docstring">        the nucleotide chain list.</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line"><tt class="py-docstring">        @param first: the number of the first chain</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line"><tt class="py-docstring">        @type first: C{int}</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line"><tt class="py-docstring">        @param second: the number of the second chain</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line"><tt class="py-docstring">        @type second: C{int}</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="extern.scientific_python.IO.PDB.Structure._joinChains" class="py-name" href="#" onclick="return doclink('link-477', '_joinChains', 'link-476');">_joinChains</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chain_constructor</tt><tt class="py-op">,</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">,</tt> <tt class="py-name">first</tt><tt class="py-op">,</tt> <tt class="py-name">second</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line"> </tt>
<a name="Structure._joinChains"></a><div id="Structure._joinChains-def"><a name="L1249"></a><tt class="py-lineno">1249</tt> <a class="py-toggle" href="#" id="Structure._joinChains-toggle" onclick="return toggle('Structure._joinChains');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#_joinChains">_joinChains</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">constructor</tt><tt class="py-op">,</tt> <tt class="py-param">chain_list</tt><tt class="py-op">,</tt> <tt class="py-param">first</tt><tt class="py-op">,</tt> <tt class="py-param">second</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure._joinChains-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure._joinChains-expanded"><a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">        <tt class="py-name">chain1</tt> <tt class="py-op">=</tt> <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt class="py-name">first</tt><tt class="py-op">]</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">        <tt class="py-name">chain2</tt> <tt class="py-op">=</tt> <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt class="py-name">second</tt><tt class="py-op">]</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">        <tt class="py-name">total</tt> <tt class="py-op">=</tt> <tt class="py-name">constructor</tt><tt class="py-op">(</tt><tt class="py-name">chain1</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">+</tt><tt class="py-name">chain2</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">,</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">                            <tt class="py-name">chain1</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt><tt class="py-op">,</tt> <tt class="py-name">chain1</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">        <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt class="py-name">first</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">total</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">chain_list</tt><tt class="py-op">[</tt><tt class="py-name">second</tt><tt class="py-op">]</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">        <tt class="py-name">index</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt class="py-name">index</tt><tt class="py-op">(</tt><tt class="py-name">chain1</tt><tt class="py-op">)</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">[</tt><tt class="py-name">index</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">total</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">        <tt class="py-name">index</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt class="py-name">index</tt><tt class="py-op">(</tt><tt class="py-name">chain2</tt><tt class="py-op">)</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">[</tt><tt class="py-name">index</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line"> </tt>
<a name="Structure.addMolecule"></a><div id="Structure.addMolecule-def"><a name="L1261"></a><tt class="py-lineno">1261</tt> <a class="py-toggle" href="#" id="Structure.addMolecule-toggle" onclick="return toggle('Structure.addMolecule');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#addMolecule">addMolecule</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">molecule</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.addMolecule-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.addMolecule-expanded"><a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">            <tt class="py-name">molecule_list</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt id="link-478" class="py-name" targets="Module gui.user_functions.molecule=gui.user_functions.molecule-module.html,Module prompt.molecule=prompt.molecule-module.html,Module test_suite.unit_tests._generic_fns.molecule=test_suite.unit_tests._generic_fns.molecule-module.html"><a title="gui.user_functions.molecule
prompt.molecule
test_suite.unit_tests._generic_fns.molecule" class="py-name" href="#" onclick="return doclink('link-478', 'molecule', 'link-478');">molecule</a></tt><tt class="py-op">.</tt><tt id="link-479" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-479', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">            <tt class="py-name">molecule_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt id="link-480" class="py-name"><a title="gui.user_functions.molecule
prompt.molecule
test_suite.unit_tests._generic_fns.molecule" class="py-name" href="#" onclick="return doclink('link-480', 'molecule', 'link-478');">molecule</a></tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-481', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">molecule_list</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">        <tt class="py-name">molecule_list</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-482', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt id="link-483" class="py-name"><a title="gui.user_functions.molecule
prompt.molecule
test_suite.unit_tests._generic_fns.molecule" class="py-name" href="#" onclick="return doclink('link-483', 'molecule', 'link-478');">molecule</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-484', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt id="link-485" class="py-name"><a title="gui.user_functions.molecule
prompt.molecule
test_suite.unit_tests._generic_fns.molecule" class="py-name" href="#" onclick="return doclink('link-485', 'molecule', 'link-478');">molecule</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">     </tt>
<a name="Structure.deleteResidue"></a><div id="Structure.deleteResidue-def"><a name="L1270"></a><tt class="py-lineno">1270</tt> <a class="py-toggle" href="#" id="Structure.deleteResidue-toggle" onclick="return toggle('Structure.deleteResidue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#deleteResidue">deleteResidue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.deleteResidue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.deleteResidue-expanded"><a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt id="link-486" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-486', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt id="link-487" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-487', 'delete', 'link-323');">delete</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-488" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-488', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">mlist</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">                <tt class="py-name">mlist</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt id="link-489" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-489', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">mlist</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">                <tt id="link-490" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-490', 'delete', 'link-323');">delete</a></tt> <tt class="py-op">=</tt> <tt id="link-491" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-491', 'type', 'link-43');">type</a></tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-492" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-492', 'delete', 'link-323');">delete</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt><tt class="py-op">[</tt><tt id="link-493" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-493', 'delete', 'link-323');">delete</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line">        <tt id="link-494" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-494', 'delete', 'link-323');">delete</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">chain</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">:</tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line">                <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt id="link-495" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-495', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">                <tt id="link-496" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-496', 'delete', 'link-323');">delete</a></tt> <tt class="py-op">=</tt> <tt class="py-name">chain</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-497" class="py-name"><a title="generic_fns.align_tensor.delete
generic_fns.diffusion_tensor.delete
generic_fns.pipes.delete
generic_fns.relax_data.delete
generic_fns.spectrum.delete
generic_fns.structure.api_base.Base_struct_API.delete
generic_fns.structure.internal.Internal.delete
generic_fns.structure.main.delete
gui.analyses.auto_model_free.Auto_model_free.delete
gui.analyses.auto_noe.Auto_noe.delete
gui.analyses.auto_rx_base.Auto_rx.delete
gui.components.relax_data.Relax_data_list.delete
gui.components.spectrum.Spectra_list.delete
gui.user_functions.molecule.Molecule.delete
gui.user_functions.pipe.Pipe.delete
gui.user_functions.relax_data.Relax_data.delete
gui.user_functions.residue.Residue.delete
gui.user_functions.spectrum.Spectrum.delete
gui.user_functions.spin.Spin.delete
gui.user_functions.structure.Structure.delete
prompt.align_tensor.Align_tensor.delete
prompt.diffusion_tensor.Diffusion_tensor.delete
prompt.model_free.Model_free.delete
prompt.molecule.Molecule.delete
prompt.pcs.PCS.delete
prompt.pipe.Pipe.delete
prompt.rdc.RDC.delete
prompt.relax_data.Relax_data.delete
prompt.residue.Residue.delete
prompt.spectrum.Spectrum.delete
prompt.spin.Spin.delete
prompt.structure.Structure.delete
relax_io.delete" class="py-name" href="#" onclick="return doclink('link-497', 'delete', 'link-323');">delete</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt class="py-name">remove</tt><tt class="py-op">(</tt><tt id="link-498" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-498', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line"> </tt>
<a name="Structure.extractData"></a><div id="Structure.extractData-def"><a name="L1304"></a><tt class="py-lineno">1304</tt> <a class="py-toggle" href="#" id="Structure.extractData-toggle" onclick="return toggle('Structure.extractData');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#extractData">extractData</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.extractData-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.extractData-expanded"><a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">        <tt class="py-name">atom_data</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-499" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-499', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">,</tt> <tt class="py-string">'name'</tt><tt class="py-op">,</tt> <tt class="py-string">'position'</tt><tt class="py-op">,</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">                     <tt class="py-string">'occupancy'</tt><tt class="py-op">,</tt> <tt class="py-string">'temperature_factor'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">            <tt class="py-name">atom_data</tt><tt class="py-op">[</tt><tt id="link-500" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-500', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-501" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-501', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt id="link-502" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-502', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-503" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-503', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'alternate'</tt><tt class="py-op">,</tt> <tt class="py-string">'charge'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">            <tt id="link-504" class="py-name"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-504', 'value', 'link-292');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-505" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-505', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt id="link-506" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-506', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-507" class="py-name"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-507', 'value', 'link-292');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line">                <tt class="py-name">atom_data</tt><tt class="py-op">[</tt><tt id="link-508" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-508', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-509" class="py-name"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-509', 'value', 'link-292');">value</a></tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line">        <tt id="link-510" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-510', 'element', 'link-218');">element</a></tt> <tt class="py-op">=</tt> <tt id="link-511" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-511', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-512" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-512', 'element', 'link-218');">element</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">''</tt><tt class="py-op">:</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">                <tt class="py-name">string</tt><tt class="py-op">.</tt><tt class="py-name">atoi</tt><tt class="py-op">(</tt><tt id="link-513" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-513', 'element', 'link-218');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">                <tt class="py-name">atom_data</tt><tt class="py-op">[</tt><tt class="py-string">'element'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-514" class="py-name"><a title="gui.user_functions.spin.Spin.element
prompt.spin.Spin.element" class="py-name" href="#" onclick="return doclink('link-514', 'element', 'link-218');">element</a></tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">        <tt class="py-name">residue_data</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">:</tt> <tt id="link-515" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-515', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt><tt class="py-op">}</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">        <tt id="link-516" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-516', 'number', 'link-240');">number</a></tt> <tt class="py-op">=</tt> <tt id="link-517" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-517', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">        <tt class="py-name">insertion</tt> <tt class="py-op">=</tt> <tt id="link-518" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-518', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'insertion_code'</tt><tt class="py-op">]</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">insertion</tt> <tt class="py-op">==</tt> <tt class="py-string">''</tt><tt class="py-op">:</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">            <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-519" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-519', 'number', 'link-240');">number</a></tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">            <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-520" class="py-name" targets="Class extern.scientific_python.IO.PDB.ResidueNumber=extern.scientific_python.IO.PDB.ResidueNumber-class.html"><a title="extern.scientific_python.IO.PDB.ResidueNumber" class="py-name" href="#" onclick="return doclink('link-520', 'ResidueNumber', 'link-520');">ResidueNumber</a></tt><tt class="py-op">(</tt><tt id="link-521" class="py-name"><a title="gui.user_functions.spin.Spin.number
info.Clore90.number
info.Delaglio95.number
info.LipariSzabo82a.number
info.LipariSzabo82b.number
info.Ref.number
info.dAuvergneGooley03.number
info.dAuvergneGooley06.number
info.dAuvergneGooley07.number
info.dAuvergneGooley08a.number
info.dAuvergneGooley08b.number
prompt.residue.Residue.number
prompt.spin.Spin.number" class="py-name" href="#" onclick="return doclink('link-521', 'number', 'link-240');">number</a></tt><tt class="py-op">,</tt> <tt class="py-name">insertion</tt><tt class="py-op">)</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">        <tt class="py-name">chain_data</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-522" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-522', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">,</tt> <tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">            <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt id="link-523" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-523', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-524" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-524', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt id="link-525" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-525', 'name', 'link-216');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pdb_code</tt><tt class="py-op">:</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">            <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">atom_data</tt><tt class="py-op">,</tt> <tt class="py-name">residue_data</tt><tt class="py-op">,</tt> <tt class="py-name">chain_data</tt> </tt>
</div><a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line"> </tt>
<a name="Structure.newResidue"></a><div id="Structure.newResidue-def"><a name="L1333"></a><tt class="py-lineno">1333</tt> <a class="py-toggle" href="#" id="Structure.newResidue-toggle" onclick="return toggle('Structure.newResidue');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#newResidue">newResidue</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.newResidue-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.newResidue-expanded"><a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">        <tt id="link-526" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-526', 'name', 'link-216');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_name'</tt><tt class="py-op">]</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">        <tt class="py-name">residue_number</tt> <tt class="py-op">=</tt> <tt class="py-name">residue_data</tt><tt class="py-op">[</tt><tt class="py-string">'residue_number'</tt><tt class="py-op">]</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-527" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-527', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt id="link-528" class="py-name"><a title="extern.scientific_python.IO.PDB.amino_acids" class="py-name" href="#" onclick="return doclink('link-528', 'amino_acids', 'link-28');">amino_acids</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">            <tt id="link-529" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-529', 'residue', 'link-420');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-530" class="py-name" targets="Class extern.scientific_python.IO.PDB.AminoAcidResidue=extern.scientific_python.IO.PDB.AminoAcidResidue-class.html"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue" class="py-name" href="#" onclick="return doclink('link-530', 'AminoAcidResidue', 'link-530');">AminoAcidResidue</a></tt><tt class="py-op">(</tt><tt id="link-531" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-531', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">residue_number</tt><tt class="py-op">)</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-532" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-532', 'name', 'link-216');">name</a></tt> <tt class="py-keyword">in</tt> <tt id="link-533" class="py-name"><a title="extern.scientific_python.IO.PDB.nucleic_acids" class="py-name" href="#" onclick="return doclink('link-533', 'nucleic_acids', 'link-29');">nucleic_acids</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">            <tt id="link-534" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-534', 'residue', 'link-420');">residue</a></tt> <tt class="py-op">=</tt> <tt id="link-535" class="py-name" targets="Class extern.scientific_python.IO.PDB.NucleotideResidue=extern.scientific_python.IO.PDB.NucleotideResidue-class.html"><a title="extern.scientific_python.IO.PDB.NucleotideResidue" class="py-name" href="#" onclick="return doclink('link-535', 'NucleotideResidue', 'link-535');">NucleotideResidue</a></tt><tt class="py-op">(</tt><tt id="link-536" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-536', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">residue_number</tt><tt class="py-op">)</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">            <tt id="link-537" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-537', 'residue', 'link-420');">residue</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecule_constructor</tt><tt class="py-op">(</tt><tt id="link-538" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-538', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">residue_number</tt><tt class="py-op">)</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">.</tt><tt id="link-539" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-539', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt id="link-540" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-540', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-541" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-541', 'residue', 'link-420');">residue</a></tt> </tt>
</div><a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line"> </tt>
<a name="Structure.newChain"></a><div id="Structure.newChain-def"><a name="L1345"></a><tt class="py-lineno">1345</tt> <a class="py-toggle" href="#" id="Structure.newChain-toggle" onclick="return toggle('Structure.newChain');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#newChain">newChain</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">residue</tt><tt class="py-op">,</tt> <tt class="py-param">chain_data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.newChain-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.newChain-expanded"><a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-542" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-542', 'residue', 'link-420');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'is_amino_acid'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">            <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chain_constructor</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">                                                   <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt><tt class="py-op">.</tt><tt id="link-543" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-543', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt id="link-544" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-544', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-545" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-545', 'residue', 'link-420');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-string">'is_nucleotide'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">            <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chain_constructor</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">                                                      <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">                                                      <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">.</tt><tt id="link-546" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-546', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">.</tt><tt id="link-547" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-547', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">            <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">DummyChain</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'chain_id'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">                               <tt class="py-name">chain_data</tt><tt class="py-op">[</tt><tt class="py-string">'segment_id'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">chain</tt> </tt>
</div><a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line"> </tt>
<a name="Structure.parseFile"></a><div id="Structure.parseFile-def"><a name="L1362"></a><tt class="py-lineno">1362</tt> <a class="py-toggle" href="#" id="Structure.parseFile-toggle" onclick="return toggle('Structure.parseFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#parseFile">parseFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.parseFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.parseFile-expanded"><a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">        <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">        <tt id="link-548" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-548', 'residue', 'link-420');">residue</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">        <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">        <tt id="link-549" class="py-name" targets="Method bmrblib.base_classes.BaseSaveframe.read()=bmrblib.base_classes.BaseSaveframe-class.html#read,Method bmrblib.nmr_star_dict.NMR_STAR.read()=bmrblib.nmr_star_dict.NMR_STAR-class.html#read,Method bmrblib.pystarlib.File.File.read()=bmrblib.pystarlib.File.File-class.html#read,Method extern.scientific_python.IO.TextFile.TextFile.read()=extern.scientific_python.IO.TextFile.TextFile-class.html#read,Function generic_fns.bmrb.read()=generic_fns.bmrb-module.html#read,Function generic_fns.pcs.read()=generic_fns.pcs-module.html#read,Function generic_fns.pdc.read()=generic_fns.pdc-module.html#read,Function generic_fns.rdc.read()=generic_fns.rdc-module.html#read,Function generic_fns.relax_data.read()=generic_fns.relax_data-module.html#read,Function generic_fns.results.read()=generic_fns.results-module.html#read,Function generic_fns.sequence.read()=generic_fns.sequence-module.html#read,Function generic_fns.spectrum.read()=generic_fns.spectrum-module.html#read,Function generic_fns.value.read()=generic_fns.value-module.html#read,Method gui.user_functions.deselect.Deselect.read()=gui.user_functions.deselect.Deselect-class.html#read,Method gui.user_functions.relax_data.Relax_data.read()=gui.user_functions.relax_data.Relax_data-class.html#read,Method gui.user_functions.results.Results.read()=gui.user_functions.results.Results-class.html#read,Method gui.user_functions.select.Select.read()=gui.user_functions.select.Select-class.html#read,Method gui.user_functions.sequence.Sequence.read()=gui.user_functions.sequence.Sequence-class.html#read,Method prompt.bmrb.BMRB.read()=prompt.bmrb.BMRB-class.html#read,Method prompt.deselect.Deselect.read()=prompt.deselect.Deselect-class.html#read,Method prompt.pcs.PCS.read()=prompt.pcs.PCS-class.html#read,Method prompt.pdc.Pdc.read()=prompt.pdc.Pdc-class.html#read,Method prompt.rdc.RDC.read()=prompt.rdc.RDC-class.html#read,Method prompt.relax_data.Relax_data.read()=prompt.relax_data.Relax_data-class.html#read,Method prompt.results.Results.read()=prompt.results.Results-class.html#read,Method prompt.select.Select.read()=prompt.select.Select-class.html#read,Method prompt.sequence.Sequence.read()=prompt.sequence.Sequence-class.html#read,Method prompt.value.Value.read()=prompt.value.Value-class.html#read"><a title="bmrblib.base_classes.BaseSaveframe.read
bmrblib.nmr_star_dict.NMR_STAR.read
bmrblib.pystarlib.File.File.read
extern.scientific_python.IO.TextFile.TextFile.read
generic_fns.bmrb.read
generic_fns.pcs.read
generic_fns.pdc.read
generic_fns.rdc.read
generic_fns.relax_data.read
generic_fns.results.read
generic_fns.sequence.read
generic_fns.spectrum.read
generic_fns.value.read
gui.user_functions.deselect.Deselect.read
gui.user_functions.relax_data.Relax_data.read
gui.user_functions.results.Results.read
gui.user_functions.select.Select.read
gui.user_functions.sequence.Sequence.read
prompt.bmrb.BMRB.read
prompt.deselect.Deselect.read
prompt.pcs.PCS.read
prompt.pdc.Pdc.read
prompt.rdc.RDC.read
prompt.relax_data.Relax_data.read
prompt.results.Results.read
prompt.select.Select.read
prompt.sequence.Sequence.read
prompt.value.Value.read" class="py-name" href="#" onclick="return doclink('link-549', 'read', 'link-549');">read</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">            <tt id="link-550" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-550', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-551" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-551', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-552" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-552', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-553" class="py-name" targets="Method extern.scientific_python.IO.PDB.PDBFile.readLine()=extern.scientific_python.IO.PDB.PDBFile-class.html#readLine"><a title="extern.scientific_python.IO.PDB.PDBFile.readLine" class="py-name" href="#" onclick="return doclink('link-553', 'readLine', 'link-553');">readLine</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-554" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-554', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'END'</tt><tt class="py-op">:</tt> <tt class="py-keyword">break</tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-555" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-555', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HEADER'</tt><tt class="py-op">:</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pdb_code</tt> <tt class="py-op">=</tt> <tt id="link-556" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-556', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'pdb_code'</tt><tt class="py-op">]</tt> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-557" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-557', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'CRYST1'</tt><tt class="py-op">:</tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-558" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-558', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt id="link-559" class="py-name"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-559', 'value', 'link-292');">value</a></tt> <tt class="py-keyword">in</tt> <tt id="link-560" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-560', 'data', 'link-39');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">                    <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-561" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-561', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt id="link-562" class="py-name"><a title="generic_fns.value
gui.user_functions.value
prompt.value" class="py-name" href="#" onclick="return doclink('link-562', 'value', 'link-292');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">space_group</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">space_group</tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="relax_io.strip" class="py-name" href="#" onclick="return doclink('link-563', 'strip', 'link-42');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-564" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-564', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'SCALE'</tt><tt class="py-op">:</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">'_scale_matrix'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt><tt class="py-op">[</tt><tt id="link-565" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-565', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-566" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-566', 'data', 'link-39');">data</a></tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-567" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-567', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'3'</tt><tt class="py-op">:</tt> <tt class="py-comment"># last line read</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">                    <tt class="py-keyword">from</tt> <tt id="link-568" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-568', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-569', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-570" class="py-name"><a title="extern.scientific_python.Geometry" class="py-name" href="#" onclick="return doclink('link-570', 'Geometry', 'link-13');">Geometry</a></tt><tt class="py-op">.</tt><tt class="py-name">Transformation</tt> \ </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line">                         <tt class="py-keyword">import</tt> <tt class="py-name">Shear</tt><tt class="py-op">,</tt> <tt class="py-name">Translation</tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line">                    <tt class="py-name">l1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'1'</tt><tt class="py-op">]</tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line">                    <tt class="py-name">l2</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'2'</tt><tt class="py-op">]</tt> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line">                    <tt class="py-name">l3</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'3'</tt><tt class="py-op">]</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line">                    <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">N</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">[</tt><tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'s1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'s2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'s3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">                                 <tt class="py-op">[</tt><tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'s1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'s2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'s3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">                                 <tt class="py-op">[</tt><tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'s1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'s2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'s3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line">                    <tt class="py-name">u</tt> <tt class="py-op">=</tt> <tt class="py-name">Vector</tt><tt class="py-op">(</tt><tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">to_fractional</tt> <tt class="py-op">=</tt> <tt class="py-name">Translation</tt><tt class="py-op">(</tt><tt class="py-name">u</tt><tt class="py-op">)</tt><tt class="py-op">*</tt><tt class="py-name">Shear</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">from_fractional</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">to_fractional</tt><tt class="py-op">.</tt><tt class="py-name">inverse</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line">                    <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scale_matrix</tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-571" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-571', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'MTRIX'</tt><tt class="py-op">:</tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">'_ncs_matrix'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt><tt class="py-op">[</tt><tt id="link-572" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-572', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-573" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-573', 'data', 'link-39');">data</a></tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-574" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-574', 'type', 'link-43');">type</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'3'</tt><tt class="py-op">:</tt> <tt class="py-comment"># last line read</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line">                    <tt class="py-keyword">from</tt> <tt id="link-575" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-575', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-576" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-576', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-577" class="py-name"><a title="extern.scientific_python.Geometry" class="py-name" href="#" onclick="return doclink('link-577', 'Geometry', 'link-13');">Geometry</a></tt><tt class="py-op">.</tt><tt class="py-name">Transformation</tt> \ </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line">                         <tt class="py-keyword">import</tt> <tt class="py-name">Rotation</tt><tt class="py-op">,</tt> <tt class="py-name">Translation</tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line">                    <tt class="py-name">l1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'1'</tt><tt class="py-op">]</tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">                    <tt class="py-name">l2</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'2'</tt><tt class="py-op">]</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line">                    <tt class="py-name">l3</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt><tt class="py-op">[</tt><tt class="py-string">'3'</tt><tt class="py-op">]</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line">                    <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">N</tt><tt class="py-op">.</tt><tt class="py-name">array</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">[</tt><tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'m1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'m2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'m3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line">                                 <tt class="py-op">[</tt><tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'m1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'m2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'m3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line">                                 <tt class="py-op">[</tt><tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'m1'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'m2'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'m3'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line">                    <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt class="py-name">Vector</tt><tt class="py-op">(</tt><tt class="py-name">l1</tt><tt class="py-op">[</tt><tt class="py-string">'v'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l2</tt><tt class="py-op">[</tt><tt class="py-string">'v'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">l3</tt><tt class="py-op">[</tt><tt class="py-string">'v'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line">                    <tt class="py-name">tr</tt> <tt class="py-op">=</tt> <tt class="py-name">Translation</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">*</tt><tt class="py-name">Rotation</tt><tt class="py-op">(</tt><tt id="link-578" class="py-name"><a title="bmrblib.structure.tensor.Tensor" class="py-name" href="#" onclick="return doclink('link-578', 'Tensor', 'link-14');">Tensor</a></tt><tt class="py-op">(</tt><tt class="py-name">m</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line">                    <tt class="py-name">tr</tt><tt class="py-op">.</tt><tt class="py-name">given</tt> <tt class="py-op">=</tt> <tt id="link-579" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-579', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'given'</tt><tt class="py-op">]</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">                    <tt class="py-name">tr</tt><tt class="py-op">.</tt><tt class="py-name">serial</tt> <tt class="py-op">=</tt> <tt id="link-580" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-580', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial'</tt><tt class="py-op">]</tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ncs_transformations</tt><tt class="py-op">.</tt><tt id="link-581" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-581', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tr</tt><tt class="py-op">)</tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line">                    <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_ncs_matrix</tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-582" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-582', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'MODEL'</tt><tt class="py-op">:</tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line">                <tt id="link-583" class="py-name"><a title="bmrblib.base_classes.BaseSaveframe.read
bmrblib.nmr_star_dict.NMR_STAR.read
bmrblib.pystarlib.File.File.read
extern.scientific_python.IO.TextFile.TextFile.read
generic_fns.bmrb.read
generic_fns.pcs.read
generic_fns.pdc.read
generic_fns.rdc.read
generic_fns.relax_data.read
generic_fns.results.read
generic_fns.sequence.read
generic_fns.spectrum.read
generic_fns.value.read
gui.user_functions.deselect.Deselect.read
gui.user_functions.relax_data.Relax_data.read
gui.user_functions.results.Results.read
gui.user_functions.select.Select.read
gui.user_functions.sequence.Sequence.read
prompt.bmrb.BMRB.read
prompt.deselect.Deselect.read
prompt.pcs.PCS.read
prompt.pdc.Pdc.read
prompt.rdc.RDC.read
prompt.relax_data.Relax_data.read
prompt.results.Results.read
prompt.select.Select.read
prompt.sequence.Sequence.read
prompt.value.Value.read" class="py-name" href="#" onclick="return doclink('link-583', 'read', 'link-549');">read</a></tt> <tt class="py-op">=</tt> <tt id="link-584" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-584', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line">                    <tt id="link-585" class="py-name"><a title="bmrblib.base_classes.BaseSaveframe.read
bmrblib.nmr_star_dict.NMR_STAR.read
bmrblib.pystarlib.File.File.read
extern.scientific_python.IO.TextFile.TextFile.read
generic_fns.bmrb.read
generic_fns.pcs.read
generic_fns.pdc.read
generic_fns.rdc.read
generic_fns.relax_data.read
generic_fns.results.read
generic_fns.sequence.read
generic_fns.spectrum.read
generic_fns.value.read
gui.user_functions.deselect.Deselect.read
gui.user_functions.relax_data.Relax_data.read
gui.user_functions.results.Results.read
gui.user_functions.select.Select.read
gui.user_functions.sequence.Sequence.read
prompt.bmrb.BMRB.read
prompt.deselect.Deselect.read
prompt.pcs.PCS.read
prompt.pdc.Pdc.read
prompt.rdc.RDC.read
prompt.relax_data.Relax_data.read
prompt.results.Results.read
prompt.select.Select.read
prompt.sequence.Sequence.read
prompt.value.Value.read" class="py-name" href="#" onclick="return doclink('link-585', 'read', 'link-549');">read</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-586" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-586', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ENDMDL'</tt><tt class="py-op">:</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line">                <tt id="link-587" class="py-name"><a title="bmrblib.base_classes.BaseSaveframe.read
bmrblib.nmr_star_dict.NMR_STAR.read
bmrblib.pystarlib.File.File.read
extern.scientific_python.IO.TextFile.TextFile.read
generic_fns.bmrb.read
generic_fns.pcs.read
generic_fns.pdc.read
generic_fns.rdc.read
generic_fns.relax_data.read
generic_fns.results.read
generic_fns.sequence.read
generic_fns.spectrum.read
generic_fns.value.read
gui.user_functions.deselect.Deselect.read
gui.user_functions.relax_data.Relax_data.read
gui.user_functions.results.Results.read
gui.user_functions.select.Select.read
gui.user_functions.sequence.Sequence.read
prompt.bmrb.BMRB.read
prompt.deselect.Deselect.read
prompt.pcs.PCS.read
prompt.pdc.Pdc.read
prompt.rdc.RDC.read
prompt.relax_data.Relax_data.read
prompt.results.Results.read
prompt.select.Select.read
prompt.sequence.Sequence.read
prompt.value.Value.read" class="py-name" href="#" onclick="return doclink('link-587', 'read', 'link-549');">read</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-588" class="py-name"><a title="bmrblib.base_classes.BaseSaveframe.read
bmrblib.nmr_star_dict.NMR_STAR.read
bmrblib.pystarlib.File.File.read
extern.scientific_python.IO.TextFile.TextFile.read
generic_fns.bmrb.read
generic_fns.pcs.read
generic_fns.pdc.read
generic_fns.rdc.read
generic_fns.relax_data.read
generic_fns.results.read
generic_fns.sequence.read
generic_fns.spectrum.read
generic_fns.value.read
gui.user_functions.deselect.Deselect.read
gui.user_functions.relax_data.Relax_data.read
gui.user_functions.results.Results.read
gui.user_functions.select.Select.read
gui.user_functions.sequence.Sequence.read
prompt.bmrb.BMRB.read
prompt.deselect.Deselect.read
prompt.pcs.PCS.read
prompt.pdc.Pdc.read
prompt.rdc.RDC.read
prompt.relax_data.Relax_data.read
prompt.results.Results.read
prompt.select.Select.read
prompt.sequence.Sequence.read
prompt.value.Value.read" class="py-name" href="#" onclick="return doclink('link-588', 'read', 'link-549');">read</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-589" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-589', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ATOM'</tt> <tt class="py-keyword">or</tt> <tt id="link-590" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-590', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'HETATM'</tt><tt class="py-op">:</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">                    <tt class="py-name">alt</tt> <tt class="py-op">=</tt> <tt id="link-591" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-591', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'alternate'</tt><tt class="py-op">]</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">alt</tt> <tt class="py-op">==</tt> <tt class="py-string">''</tt> <tt class="py-keyword">or</tt> <tt class="py-name">alt</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alternate</tt><tt class="py-op">:</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">                        <tt class="py-name">atom_data</tt><tt class="py-op">,</tt> <tt class="py-name">residue_data</tt><tt class="py-op">,</tt> <tt class="py-name">chain_data</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">                                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-592" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.extractData()=extern.scientific_python.IO.PDB.Structure-class.html#extractData"><a title="extern.scientific_python.IO.PDB.Structure.extractData" class="py-name" href="#" onclick="return doclink('link-592', 'extractData', 'link-592');">extractData</a></tt><tt class="py-op">(</tt><tt id="link-593" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-593', 'data', 'link-39');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt id="link-594" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-594', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ATOM'</tt><tt class="py-op">:</tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">                            <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt id="link-595" class="py-name" targets="Method gui.interpreter.Interpreter.apply()=gui.interpreter.Interpreter-class.html#apply"><a title="gui.interpreter.Interpreter.apply" class="py-name" href="#" onclick="return doclink('link-595', 'apply', 'link-595');">apply</a></tt><tt class="py-op">(</tt><tt id="link-596" class="py-name" targets="Class extern.scientific_python.IO.PDB.Atom=extern.scientific_python.IO.PDB.Atom-class.html"><a title="extern.scientific_python.IO.PDB.Atom" class="py-name" href="#" onclick="return doclink('link-596', 'Atom', 'link-596');">Atom</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">atom_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">                            <tt class="py-name">atom</tt> <tt class="py-op">=</tt> <tt id="link-597" class="py-name"><a title="gui.interpreter.Interpreter.apply" class="py-name" href="#" onclick="return doclink('link-597', 'apply', 'link-595');">apply</a></tt><tt class="py-op">(</tt><tt id="link-598" class="py-name" targets="Class extern.scientific_python.IO.PDB.HetAtom=extern.scientific_python.IO.PDB.HetAtom-class.html"><a title="extern.scientific_python.IO.PDB.HetAtom" class="py-name" href="#" onclick="return doclink('link-598', 'HetAtom', 'link-598');">HetAtom</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">atom_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">                        <tt class="py-name">new_chain</tt> <tt class="py-op">=</tt> <tt class="py-name">chain</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">                                    <tt class="py-keyword">not</tt> <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt id="link-599" class="py-name" targets="Method extern.scientific_python.IO.PDB.Group.isCompatible()=extern.scientific_python.IO.PDB.Group-class.html#isCompatible,Method extern.scientific_python.IO.PDB.NucleotideChain.isCompatible()=extern.scientific_python.IO.PDB.NucleotideChain-class.html#isCompatible,Method extern.scientific_python.IO.PDB.NucleotideResidue.isCompatible()=extern.scientific_python.IO.PDB.NucleotideResidue-class.html#isCompatible,Method extern.scientific_python.IO.PDB.PeptideChain.isCompatible()=extern.scientific_python.IO.PDB.PeptideChain-class.html#isCompatible"><a title="extern.scientific_python.IO.PDB.Group.isCompatible
extern.scientific_python.IO.PDB.NucleotideChain.isCompatible
extern.scientific_python.IO.PDB.NucleotideResidue.isCompatible
extern.scientific_python.IO.PDB.PeptideChain.isCompatible" class="py-name" href="#" onclick="return doclink('link-599', 'isCompatible', 'link-599');">isCompatible</a></tt><tt class="py-op">(</tt><tt class="py-name">chain_data</tt><tt class="py-op">,</tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">                                                           <tt class="py-name">residue_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">                        <tt class="py-name">new_residue</tt> <tt class="py-op">=</tt> <tt class="py-name">new_chain</tt> <tt class="py-keyword">or</tt> <tt id="link-600" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-600', 'residue', 'link-420');">residue</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> \ </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">                                      <tt class="py-keyword">or</tt> <tt class="py-keyword">not</tt> <tt id="link-601" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-601', 'residue', 'link-420');">residue</a></tt><tt class="py-op">.</tt><tt id="link-602" class="py-name"><a title="extern.scientific_python.IO.PDB.Group.isCompatible
extern.scientific_python.IO.PDB.NucleotideChain.isCompatible
extern.scientific_python.IO.PDB.NucleotideResidue.isCompatible
extern.scientific_python.IO.PDB.PeptideChain.isCompatible" class="py-name" href="#" onclick="return doclink('link-602', 'isCompatible', 'link-599');">isCompatible</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">new_residue</tt> <tt class="py-keyword">and</tt> <tt class="py-name">chain</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">                           <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt id="link-603" class="py-name" targets="Method extern.scientific_python.IO.PDB.NucleotideChain.isTerminated()=extern.scientific_python.IO.PDB.NucleotideChain-class.html#isTerminated,Method extern.scientific_python.IO.PDB.PeptideChain.isTerminated()=extern.scientific_python.IO.PDB.PeptideChain-class.html#isTerminated"><a title="extern.scientific_python.IO.PDB.NucleotideChain.isTerminated
extern.scientific_python.IO.PDB.PeptideChain.isTerminated" class="py-name" href="#" onclick="return doclink('link-603', 'isTerminated', 'link-603');">isTerminated</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">                            <tt class="py-name">new_chain</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">new_residue</tt><tt class="py-op">:</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">                            <tt id="link-604" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-604', 'residue', 'link-420');">residue</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-605" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.newResidue()=extern.scientific_python.IO.PDB.Structure-class.html#newResidue"><a title="extern.scientific_python.IO.PDB.Structure.newResidue" class="py-name" href="#" onclick="return doclink('link-605', 'newResidue', 'link-605');">newResidue</a></tt><tt class="py-op">(</tt><tt class="py-name">residue_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">new_chain</tt><tt class="py-op">:</tt> </tt>
<a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line">                                <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-606" class="py-name" targets="Method extern.scientific_python.IO.PDB.Structure.newChain()=extern.scientific_python.IO.PDB.Structure-class.html#newChain"><a title="extern.scientific_python.IO.PDB.Structure.newChain" class="py-name" href="#" onclick="return doclink('link-606', 'newChain', 'link-606');">newChain</a></tt><tt class="py-op">(</tt><tt id="link-607" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-607', 'residue', 'link-420');">residue</a></tt><tt class="py-op">,</tt> <tt class="py-name">chain_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1440"></a><tt class="py-lineno">1440</tt>  <tt class="py-line">                            <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt id="link-608" class="py-name" targets="Method extern.scientific_python.IO.PDB.Chain.addResidue()=extern.scientific_python.IO.PDB.Chain-class.html#addResidue"><a title="extern.scientific_python.IO.PDB.Chain.addResidue" class="py-name" href="#" onclick="return doclink('link-608', 'addResidue', 'link-608');">addResidue</a></tt><tt class="py-op">(</tt><tt id="link-609" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-609', 'residue', 'link-420');">residue</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">                        <tt id="link-610" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-610', 'residue', 'link-420');">residue</a></tt><tt class="py-op">.</tt><tt id="link-611" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.addAtom
extern.scientific_python.IO.PDB.Group.addAtom
extern.scientific_python.IO.PDB.NucleotideResidue.addAtom" class="py-name" href="#" onclick="return doclink('link-611', 'addAtom', 'link-353');">addAtom</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-612" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-612', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'ANISOU'</tt><tt class="py-op">:</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line">                    <tt class="py-name">alt</tt> <tt class="py-op">=</tt> <tt id="link-613" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-613', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'alternate'</tt><tt class="py-op">]</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">alt</tt> <tt class="py-op">==</tt> <tt class="py-string">''</tt> <tt class="py-keyword">or</tt> <tt class="py-name">alt</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alternate</tt><tt class="py-op">:</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">                            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"ANISOU record before "</tt> <tt class="py-op">+</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">                                              <tt class="py-string">"ATOM record"</tt><tt class="py-op">)</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">                        <tt class="py-name">atom</tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-614" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-614', 'data', 'link-39');">data</a></tt><tt class="py-op">[</tt><tt class="py-string">'u'</tt><tt class="py-op">]</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt id="link-615" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-615', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'TERM'</tt><tt class="py-op">:</tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">chain</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">                        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"TERM record before chain"</tt><tt class="py-op">)</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">                    <tt class="py-name">chain</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line"> </tt>
<a name="Structure.findSpaceGroupTransformations"></a><div id="Structure.findSpaceGroupTransformations-def"><a name="L1454"></a><tt class="py-lineno">1454</tt> <a class="py-toggle" href="#" id="Structure.findSpaceGroupTransformations-toggle" onclick="return toggle('Structure.findSpaceGroupTransformations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#findSpaceGroupTransformations">findSpaceGroupTransformations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.findSpaceGroupTransformations-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.findSpaceGroupTransformations-expanded"><a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">space_group</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">to_fractional</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-616" class="py-name"><a title="extern" class="py-name" href="#" onclick="return doclink('link-616', 'extern', 'link-0');">extern</a></tt><tt class="py-op">.</tt><tt id="link-617" class="py-name"><a title="extern.scientific_python" class="py-name" href="#" onclick="return doclink('link-617', 'scientific_python', 'link-1');">scientific_python</a></tt><tt class="py-op">.</tt><tt id="link-618" class="py-name"><a title="extern.scientific_python.IO" class="py-name" href="#" onclick="return doclink('link-618', 'IO', 'link-2');">IO</a></tt><tt class="py-op">.</tt><tt id="link-619" class="py-name" targets="Module extern.scientific_python.IO.PDBSpaceGroups=extern.scientific_python.IO.PDBSpaceGroups-module.html"><a title="extern.scientific_python.IO.PDBSpaceGroups" class="py-name" href="#" onclick="return doclink('link-619', 'PDBSpaceGroups', 'link-619');">PDBSpaceGroups</a></tt> <tt class="py-keyword">import</tt> \ </tt>
<a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line">                 <tt id="link-620" class="py-name" targets="Function extern.scientific_python.IO.PDBSpaceGroups.getSpaceGroupTransformations()=extern.scientific_python.IO.PDBSpaceGroups-module.html#getSpaceGroupTransformations"><a title="extern.scientific_python.IO.PDBSpaceGroups.getSpaceGroupTransformations" class="py-name" href="#" onclick="return doclink('link-620', 'getSpaceGroupTransformations', 'link-620');">getSpaceGroupTransformations</a></tt> </tt>
<a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">                <tt class="py-name">trs</tt> <tt class="py-op">=</tt> <tt id="link-621" class="py-name"><a title="extern.scientific_python.IO.PDBSpaceGroups.getSpaceGroupTransformations" class="py-name" href="#" onclick="return doclink('link-621', 'getSpaceGroupTransformations', 'link-620');">getSpaceGroupTransformations</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">space_group</tt><tt class="py-op">)</tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">tr</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trs</tt><tt class="py-op">:</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line">                <tt class="py-name">tr</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">from_fractional</tt><tt class="py-op">*</tt><tt class="py-name">tr</tt><tt class="py-op">*</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">to_fractional</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cs_transformations</tt><tt class="py-op">.</tt><tt id="link-622" class="py-name"><a title="data.align_tensor.AlignTensorSimList.append
data.diff_tensor.DiffTensorSimList.append" class="py-name" href="#" onclick="return doclink('link-622', 'append', 'link-32');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tr</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"> </tt>
<a name="Structure.renumberAtoms"></a><div id="Structure.renumberAtoms-def"><a name="L1466"></a><tt class="py-lineno">1466</tt> <a class="py-toggle" href="#" id="Structure.renumberAtoms-toggle" onclick="return toggle('Structure.renumberAtoms');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#renumberAtoms">renumberAtoms</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.renumberAtoms-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.renumberAtoms-expanded"><a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"><tt class="py-docstring">        Renumber all atoms sequentially starting with 1</tt> </tt>
<a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line">        <tt class="py-name">n</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-623" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-623', 'residue', 'link-420');">residue</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">residues</tt><tt class="py-op">:</tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt id="link-624" class="py-name"><a title="gui.user_functions.residue
prompt.residue
test_suite.unit_tests._generic_fns.residue" class="py-name" href="#" onclick="return doclink('link-624', 'residue', 'link-420');">residue</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line">                <tt class="py-name">atom</tt><tt class="py-op">[</tt><tt class="py-string">'serial_number'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">n</tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line">                <tt class="py-name">n</tt> <tt class="py-op">=</tt> <tt class="py-name">n</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"> </tt>
<a name="Structure.__repr__"></a><div id="Structure.__repr__-def"><a name="L1476"></a><tt class="py-lineno">1476</tt> <a class="py-toggle" href="#" id="Structure.__repr__-toggle" onclick="return toggle('Structure.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.__repr__-expanded"><a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">+</tt> <tt class="py-string">"("</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">", model="</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">model</tt><tt class="py-op">)</tt> </tt>
<a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alternate</tt> <tt class="py-op">!=</tt> <tt class="py-string">'A'</tt><tt class="py-op">:</tt> </tt>
<a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">", alternate_code = "</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">alternate</tt><tt class="py-op">)</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"):\n"</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-625" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-625', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-string">"Peptide"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">peptide_chains</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line">                           <tt class="py-op">(</tt><tt class="py-string">"Nucleotide"</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">nucleotide_chains</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">list</tt><tt class="py-op">:</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line">                <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"  "</tt> <tt class="py-op">+</tt> <tt id="link-626" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-626', 'name', 'link-216');">name</a></tt> <tt class="py-op">+</tt> <tt class="py-string">" chain "</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt><tt class="py-op">:</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line">                    <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt class="py-name">segment_id</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt><tt class="py-op">:</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line">                    <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt class="py-name">chain_id</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line">                <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"of length "</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-627" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-627', 'name', 'link-216');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">molecules</tt><tt class="py-op">.</tt><tt class="py-name">items</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"  "</tt> <tt class="py-op">+</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt id="link-628" class="py-name"><a title="bmrblib.kinetics.auto_relaxation.AutoRelaxationSaveframe.name
bmrblib.kinetics.auto_relaxation_v3_1.AutoRelaxationSaveframe_v3_1.name
bmrblib.kinetics.heteronucl_NOEs_v2_1.HeteronuclNOESaveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v2_1.HeteronuclT1Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T1_relaxation_v3_1.HeteronuclT1Saveframe_v3_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v2_1.HeteronuclT2Saveframe_v2_1.name
bmrblib.kinetics.heteronucl_T2_relaxation_v3_1.HeteronuclT2Saveframe_v3_1.name
gui.user_functions.spin.Spin.name
prompt.molecule.Molecule.name
prompt.residue.Residue.name
prompt.spin.Spin.name" class="py-name" href="#" onclick="return doclink('link-628', 'name', 'link-216');">name</a></tt> <tt class="py-op">+</tt> <tt class="py-string">" molecule"</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line">                <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line">                <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> <tt class="py-op">+</tt> <tt class="py-string">"s\n"</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">s</tt> </tt>
</div><a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line"> </tt>
<a name="Structure.writeToFile"></a><div id="Structure.writeToFile-def"><a name="L1500"></a><tt class="py-lineno">1500</tt> <a class="py-toggle" href="#" id="Structure.writeToFile-toggle" onclick="return toggle('Structure.writeToFile');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="extern.scientific_python.IO.PDB.Structure-class.html#writeToFile">writeToFile</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Structure.writeToFile-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Structure.writeToFile-expanded"><a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line">        <tt class="py-docstring">"""</tt> </tt>
<a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line"><tt class="py-docstring">        Write everything to a file</tt> </tt>
<a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"><tt class="py-docstring">        @param file: a PDB file object or a filename</tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line"><tt class="py-docstring">        @type file: L{PDBFile} or C{str}</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line">        <tt id="link-629" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-629', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-630" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-630', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt id="link-631" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-631', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-632" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-632', 'type', 'link-43');">type</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line">            <tt id="link-633" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-633', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-634" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-634', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt id="link-635" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-635', 'file', 'link-36');">file</a></tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">)</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line">            <tt id="link-636" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-636', 'close', 'link-280');">close</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">o</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">objects</tt><tt class="py-op">:</tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line">            <tt class="py-name">o</tt><tt class="py-op">.</tt><tt id="link-637" class="py-name"><a title="extern.scientific_python.IO.PDB.AminoAcidResidue.writeToFile
extern.scientific_python.IO.PDB.Atom.writeToFile
extern.scientific_python.IO.PDB.Chain.writeToFile
extern.scientific_python.IO.PDB.Group.writeToFile
extern.scientific_python.IO.PDB.NucleotideResidue.writeToFile
extern.scientific_python.IO.PDB.Structure.writeToFile" class="py-name" href="#" onclick="return doclink('link-637', 'writeToFile', 'link-345');">writeToFile</a></tt><tt class="py-op">(</tt><tt id="link-638" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-638', 'file', 'link-36');">file</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-639" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-639', 'close', 'link-280');">close</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line">            <tt id="link-640" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-640', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-641" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-641', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line"> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'__main__'</tt><tt class="py-op">:</tt> </tt>
<a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line"> </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line">        <tt id="link-642" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-642', 'file', 'link-36');">file</a></tt> <tt class="py-op">=</tt> <tt id="link-643" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-643', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt class="py-string">'~/3lzt.pdb'</tt><tt class="py-op">)</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line">        <tt id="link-644" class="py-name"><a title="generic_fns.align_tensor.copy
generic_fns.diffusion_tensor.copy
generic_fns.pipes.copy
generic_fns.relax_data.copy
generic_fns.sequence.copy
generic_fns.value.copy
gui.user_functions.molecule.Molecule.copy
gui.user_functions.pipe.Pipe.copy
gui.user_functions.residue.Residue.copy
gui.user_functions.sequence.Sequence.copy
gui.user_functions.spin.Spin.copy
prompt.align_tensor.Align_tensor.copy
prompt.diffusion_tensor.Diffusion_tensor.copy
prompt.molecule.Molecule.copy
prompt.pcs.PCS.copy
prompt.pipe.Pipe.copy
prompt.rdc.RDC.copy
prompt.relax_data.Relax_data.copy
prompt.residue.Residue.copy
prompt.sequence.Sequence.copy
prompt.spin.Spin.copy
prompt.value.Value.copy" class="py-name" href="#" onclick="return doclink('link-644', 'copy', 'link-17');">copy</a></tt> <tt class="py-op">=</tt> <tt id="link-645" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile" class="py-name" href="#" onclick="return doclink('link-645', 'PDBFile', 'link-301');">PDBFile</a></tt><tt class="py-op">(</tt><tt class="py-string">'test.pdb'</tt><tt class="py-op">,</tt> <tt class="py-string">'w'</tt><tt class="py-op">,</tt> <tt class="py-string">'xplor'</tt><tt class="py-op">)</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line">            <tt id="link-646" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-646', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-647" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-647', 'data', 'link-39');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-648" class="py-name"><a title="dep_check.file" class="py-name" href="#" onclick="return doclink('link-648', 'file', 'link-36');">file</a></tt><tt class="py-op">.</tt><tt id="link-649" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.readLine" class="py-name" href="#" onclick="return doclink('link-649', 'readLine', 'link-553');">readLine</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-650" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-650', 'type', 'link-43');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'END'</tt><tt class="py-op">:</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line">            <tt id="link-651" class="py-name"><a title="generic_fns.align_tensor.copy
generic_fns.diffusion_tensor.copy
generic_fns.pipes.copy
generic_fns.relax_data.copy
generic_fns.sequence.copy
generic_fns.value.copy
gui.user_functions.molecule.Molecule.copy
gui.user_functions.pipe.Pipe.copy
gui.user_functions.residue.Residue.copy
gui.user_functions.sequence.Sequence.copy
gui.user_functions.spin.Spin.copy
prompt.align_tensor.Align_tensor.copy
prompt.diffusion_tensor.Diffusion_tensor.copy
prompt.molecule.Molecule.copy
prompt.pcs.PCS.copy
prompt.pipe.Pipe.copy
prompt.rdc.RDC.copy
prompt.relax_data.Relax_data.copy
prompt.residue.Residue.copy
prompt.sequence.Sequence.copy
prompt.spin.Spin.copy
prompt.value.Value.copy" class="py-name" href="#" onclick="return doclink('link-651', 'copy', 'link-17');">copy</a></tt><tt class="py-op">.</tt><tt id="link-652" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.writeLine" class="py-name" href="#" onclick="return doclink('link-652', 'writeLine', 'link-234');">writeLine</a></tt><tt class="py-op">(</tt><tt id="link-653" class="py-name"><a title="extern.scientific_python.IO.PDB.Atom.type
extern.scientific_python.IO.PDB.HetAtom.type
info.Bieri11.type
info.Clore90.type
info.Delaglio95.type
info.GoddardKneller.type
info.LipariSzabo82a.type
info.LipariSzabo82b.type
info.Ref.type
info.dAuvergne06.type
info.dAuvergneGooley03.type
info.dAuvergneGooley06.type
info.dAuvergneGooley07.type
info.dAuvergneGooley08a.type
info.dAuvergneGooley08b.type
prompt.molecule.Molecule.type" class="py-name" href="#" onclick="return doclink('link-653', 'type', 'link-43');">type</a></tt><tt class="py-op">,</tt> <tt id="link-654" class="py-name"><a title="data" class="py-name" href="#" onclick="return doclink('link-654', 'data', 'link-39');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line">        <tt id="link-655" class="py-name"><a title="generic_fns.align_tensor.copy
generic_fns.diffusion_tensor.copy
generic_fns.pipes.copy
generic_fns.relax_data.copy
generic_fns.sequence.copy
generic_fns.value.copy
gui.user_functions.molecule.Molecule.copy
gui.user_functions.pipe.Pipe.copy
gui.user_functions.residue.Residue.copy
gui.user_functions.sequence.Sequence.copy
gui.user_functions.spin.Spin.copy
prompt.align_tensor.Align_tensor.copy
prompt.diffusion_tensor.Diffusion_tensor.copy
prompt.molecule.Molecule.copy
prompt.pcs.PCS.copy
prompt.pipe.Pipe.copy
prompt.rdc.RDC.copy
prompt.relax_data.Relax_data.copy
prompt.residue.Residue.copy
prompt.sequence.Sequence.copy
prompt.spin.Spin.copy
prompt.value.Value.copy" class="py-name" href="#" onclick="return doclink('link-655', 'copy', 'link-17');">copy</a></tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="extern.scientific_python.IO.PDB.PDBFile.close
extern.scientific_python.IO.TextFile.TextFile.close
relax_io.DummyFileObject.close" class="py-name" href="#" onclick="return doclink('link-656', 'close', 'link-280');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line"> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line"> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt id="link-657" class="py-name" targets="Class extern.scientific_python.IO.PDB.Structure=extern.scientific_python.IO.PDB.Structure-class.html,Class gui.user_functions.structure.Structure=gui.user_functions.structure.Structure-class.html,Class prompt.structure.Structure=prompt.structure.Structure-class.html,Class test_suite.system_tests.structure.Structure=test_suite.system_tests.structure.Structure-class.html"><a title="extern.scientific_python.IO.PDB.Structure
gui.user_functions.structure.Structure
prompt.structure.Structure
test_suite.system_tests.structure.Structure" class="py-name" href="#" onclick="return doclink('link-657', 'Structure', 'link-657');">Structure</a></tt><tt class="py-op">(</tt><tt class="py-string">'~/Programs/MMTK/main/MMTK/Database/PDB/insulin.pdb'</tt><tt class="py-op">)</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt> <tt class="py-name">s</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://nmr-relax.com">relax</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Mon Nov 14 21:31:51 2011
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
